// vue-hooks-plus/dist/js/index.es.js
var Ru = Object.defineProperty;
var xu = (e, t, n) => t in e ? Ru(e, t, { enumerable: !0, configurable: !0, writable: !0, value: n }) : e[t] = n;
var H = (e, t, n) => (xu(e, typeof t != "symbol" ? t + "" : t, n), n);
import { ref as E, watchEffect as k, unref as w, watch as B, onScopeDispose as ft, computed as $, onUnmounted as _e, inject as Fu, reactive as dt, isRef as fe, toRefs as Jt, provide as Lu, shallowReadonly as fo, readonly as V, createApp as ku, defineComponent as Nu, markRaw as go, onMounted as gt, onBeforeUnmount as Uu, nextTick as $u, getCurrentScope as Bu } from "vue";
const po = (e, { manual: t, ready: n = !0, refreshDeps: r = [], refreshDepsAction: o }) => {
  const u = E(!1);
  return k(() => {
    !t && e.options.refreshDeps !== !0 && (u.value = w(n));
  }), r instanceof Array ? B(
    [u, ...r],
    ([s]) => {
      !s || !t && s && (o ? o() : e.refresh());
    },
    {
      deep: !0,
      immediate: !1
    }
  ) : B(u, (s) => {
    !t && s && (o ? o() : e.refresh());
  }), {
    name: "autoRunPlugin",
    onBefore: () => {
      if (!w(n))
        return {
          stopNow: !0
        };
    }
  };
};
po.onInit = ({ ready: e = !0, manual: t }) => ({
  loading: !t && w(e)
});
const we = /* @__PURE__ */ new Map(), Vu = (e, t, n) => {
  const r = we.get(e);
  r != null && r.timer && clearTimeout(r.timer);
  let o;
  t > -1 && (o = setTimeout(() => {
    we.delete(e);
  }, t)), we.set(e, {
    ...n,
    timer: o
  });
}, Mu = (e) => we.get(e), _g = (e) => {
  e ? (Array.isArray(e) ? e : [e]).forEach((n) => we.delete(n)) : we.clear();
}, rt = /* @__PURE__ */ new Map(), ju = (e) => rt.get(e), Hu = (e, t) => {
  rt.set(e, t), t.then((n) => (rt.delete(e), n)).catch((n) => {
    throw rt.delete(e), n;
  });
}, le = {}, zu = [], Ku = (e, t) => {
  le[e] && (le[e].forEach((n) => n(t)), zu.forEach((n) => n({
    type: e,
    data: t
  })));
}, Tt = (e, t) => (le[e] || (le[e] = []), le[e].push(t), function() {
  const r = le[e].indexOf(t);
  le[e].splice(r, 1);
}), qu = (e, {
  cacheKey: t,
  cacheTime: n = 5 * 60 * 1e3,
  staleTime: r = 0,
  setCache: o,
  getCache: u
}) => {
  const s = E(), a = E(), i = (f, l) => {
    o ? o(l) : Vu(f, n, l), Ku(f, l.data);
  }, c = (f, l = []) => u ? u(l) : Mu(f);
  return k(() => {
    if (!t)
      return;
    const f = c(t);
    f && Object.hasOwnProperty.call(f, "data") && (e.state.data = f.data, e.state.params = f.params, (r === -1 || new Date().getTime() - f.time <= r) && (e.state.loading = !1)), s.value = Tt(t, (l) => {
      e.setState({ data: l });
    });
  }), ft(() => {
    var f;
    (f = s.value) == null || f.call(s);
  }), t ? {
    name: "cachePlugin",
    onBefore: (f) => {
      const l = c(t, f);
      return !l || !Object.hasOwnProperty.call(l, "data") ? {} : r === -1 || new Date().getTime() - l.time <= r ? {
        loading: !1,
        data: l == null ? void 0 : l.data,
        returnNow: !0
      } : {
        data: l == null ? void 0 : l.data
      };
    },
    onRequest: (f, l) => {
      let g = ju(t);
      return g && g !== a.value ? { servicePromise: g } : (g = f(...l), a.value = g, Hu(t, g), { servicePromise: g });
    },
    onSuccess: (f, l) => {
      var g;
      t && ((g = s.value) == null || g.call(s), i(t, {
        data: f,
        params: l,
        time: new Date().getTime()
      }), s.value = Tt(t, (d) => {
        e.setState({ data: d });
      }));
    },
    onMutate: (f) => {
      var l;
      t && ((l = s.value) == null || l.call(s), i(t, {
        data: f,
        params: e.state.params,
        time: new Date().getTime()
      }), s.value = Tt(t, (g) => {
        e.setState({ data: g });
      }));
    }
  } : {};
};
function G(e) {
  var t = typeof e;
  return e != null && (t == "object" || t == "function");
}
var Gu = typeof global == "object" && global && global.Object === Object && global;
const _o = Gu;
var Wu = typeof self == "object" && self && self.Object === Object && self, Yu = _o || Wu || Function("return this")();
const Y = Yu;
var Xu = function() {
  return Y.Date.now();
};
const At = Xu;
var Zu = /\s/;
function Ju(e) {
  for (var t = e.length; t-- && Zu.test(e.charAt(t)); )
    ;
  return t;
}
var Qu = /^\s+/;
function es(e) {
  return e && e.slice(0, Ju(e) + 1).replace(Qu, "");
}
var ts = Y.Symbol;
const oe = ts;
var ho = Object.prototype, ns = ho.hasOwnProperty, rs = ho.toString, Fe = oe ? oe.toStringTag : void 0;
function os(e) {
  var t = ns.call(e, Fe), n = e[Fe];
  try {
    e[Fe] = void 0;
    var r = !0;
  } catch {
  }
  var o = rs.call(e);
  return r && (t ? e[Fe] = n : delete e[Fe]), o;
}
var us = Object.prototype, ss = us.toString;
function as(e) {
  return ss.call(e);
}
var is = "[object Null]", ls = "[object Undefined]", Sn = oe ? oe.toStringTag : void 0;
function he(e) {
  return e == null ? e === void 0 ? ls : is : Sn && Sn in Object(e) ? os(e) : as(e);
}
function ee(e) {
  return e != null && typeof e == "object";
}
var cs = "[object Symbol]";
function fs(e) {
  return typeof e == "symbol" || ee(e) && he(e) == cs;
}
var Cn = 0 / 0, ds = /^[-+]0x[0-9a-f]+$/i, gs = /^0b[01]+$/i, ps = /^0o[0-7]+$/i, _s = parseInt;
function Dn(e) {
  if (typeof e == "number")
    return e;
  if (fs(e))
    return Cn;
  if (G(e)) {
    var t = typeof e.valueOf == "function" ? e.valueOf() : e;
    e = G(t) ? t + "" : t;
  }
  if (typeof e != "string")
    return e === 0 ? e : +e;
  e = es(e);
  var n = gs.test(e);
  return n || ps.test(e) ? _s(e.slice(2), n ? 2 : 8) : ds.test(e) ? Cn : +e;
}
var hs = "Expected a function", vs = Math.max, ms = Math.min;
function Qt(e, t, n) {
  var r, o, u, s, a, i, c = 0, f = !1, l = !1, g = !0;
  if (typeof e != "function")
    throw new TypeError(hs);
  t = Dn(t) || 0, G(n) && (f = !!n.leading, l = "maxWait" in n, u = l ? vs(Dn(n.maxWait) || 0, t) : u, g = "trailing" in n ? !!n.trailing : g);
  function d(O) {
    var F = r, L = o;
    return r = o = void 0, c = O, s = e.apply(L, F), s;
  }
  function h(O) {
    return c = O, a = setTimeout(_, t), f ? d(O) : s;
  }
  function v(O) {
    var F = O - i, L = O - c, M = t - F;
    return l ? ms(M, u - L) : M;
  }
  function p(O) {
    var F = O - i, L = O - c;
    return i === void 0 || F >= t || F < 0 || l && L >= u;
  }
  function _() {
    var O = At();
    if (p(O))
      return y(O);
    a = setTimeout(_, v(O));
  }
  function y(O) {
    return a = void 0, g && r ? d(O) : (r = o = void 0, s);
  }
  function T() {
    a !== void 0 && clearTimeout(a), c = 0, r = i = o = a = void 0;
  }
  function S() {
    return a === void 0 ? s : y(At());
  }
  function A() {
    var O = At(), F = p(O);
    if (r = arguments, o = this, i = O, F) {
      if (a === void 0)
        return h(i);
      if (l)
        return clearTimeout(a), a = setTimeout(_, t), d(i);
    }
    return a === void 0 && (a = setTimeout(_, t)), s;
  }
  return A.cancel = T, A.flush = S, A;
}
const Es = (e, { debounceWait: t, debounceLeading: n, debounceTrailing: r, debounceMaxWait: o }) => {
  const u = E(), s = $(() => {
    const a = {}, i = w(n), c = w(r), f = w(o);
    return i !== void 0 && (a.leading = i), c !== void 0 && (a.trailing = c), f !== void 0 && (a.maxWait = f), a;
  });
  return k((a) => {
    if (w(t)) {
      const i = e.runAsync.bind(e);
      u.value = Qt(
        (c) => {
          c();
        },
        w(t),
        s.value
      ), e.runAsync = (...c) => new Promise((f, l) => {
        var g;
        (g = u.value) == null || g.call(u, () => {
          i(...c).then(f).catch(l);
        });
      }), a(() => {
        var c;
        (c = u.value) == null || c.cancel(), e.runAsync = i;
      });
    }
  }), w(t) ? {
    name: "debouncePlugin",
    onCancel: () => {
      var a;
      (a = u.value) == null || a.cancel();
    }
  } : {};
};
function ys(e) {
  const n = e.toString().match(/^function\s+([^\s(]+)/);
  return n ? n[1] : "";
}
function bs(e) {
  const t = e.toString(), n = /([a-zA-Z$_][a-zA-Z0-9$_]*)\s*\(/, r = t.match(n);
  return r ? r[1].trim() : "";
}
function Ts(e) {
  return e === "pending" ? 16747520 : e === "done" ? 6586111 : e === "error" ? 16724736 : e === "cancel" ? 10500409 : e === "mutate" ? 27647 : 4873398;
}
class As {
  constructor() {
    H(this, "requestInstances", /* @__PURE__ */ new Map());
    H(this, "listeners", []);
  }
  emit(t) {
    this.listeners.forEach((n) => n(t));
  }
  subscribe(t) {
    return this.listeners.push(t), () => {
      const n = this.listeners.indexOf(t);
      this.listeners.splice(n, 1);
    };
  }
  insert(t, n) {
    this.requestInstances.set(t, { ...n }), this.emit({
      key: t,
      ...n
    });
  }
  update(t, n) {
    this.has(t) && this.requestInstances.set(t, { ...this.requestInstances.get(t), ...n });
  }
  has(t) {
    return this.requestInstances.has(t);
  }
  reset(t) {
    if (this.requestInstances.has(t)) {
      const n = this.requestInstances.get(t);
      this.requestInstances.clear(), this.insert(t, n);
    } else
      this.requestInstances.clear();
  }
  getAll() {
    return this.requestInstances;
  }
}
const N = new As(), Os = (e, { ready: t = !0, debugKey: n, ...r }) => {
  const o = () => {
    if (n && !N.has(n)) {
      const s = e.serviceRef.value.toString().includes("function") ? ys(e.serviceRef.value.toString()) : bs(e.serviceRef.value.toString());
      N.insert(n, {
        instance: e,
        requestName: s,
        time: Date.now()
      });
    }
  }, u = $(
    () => Object.fromEntries(
      Object.entries({ ready: t, ...r }).map(([s, a]) => [s, w(a)])
    )
  );
  return k(() => {
    n && N.has(n) && N.emit({
      ...e,
      options: { ...e.options, ...u.value }
    });
  }), {
    name: "devtoolsPlugin",
    onBefore: (s) => {
      o(), n && N.has(n) && N.emit({
        ...e.state,
        key: n,
        params: s,
        loading: !0,
        time: Date.now(),
        type: "pending"
      });
    },
    onSuccess(s, a) {
      o(), n && N.has(n) && N.emit({
        ...e.state,
        key: n,
        data: s,
        params: a,
        loading: !1,
        time: Date.now(),
        type: "done"
      });
    },
    onCancel() {
      o(), n && N.has(n) && N.emit({
        ...e.state,
        key: n,
        loading: !1,
        time: Date.now(),
        type: "cancel"
      });
    },
    onError(s, a) {
      o(), n && N.has(n) && N.emit({
        ...e.state,
        key: n,
        params: a,
        loading: !1,
        error: s,
        time: Date.now(),
        type: "error"
      });
    },
    onMutate(s) {
      o(), n && N.has(n) && N.emit({
        ...e.state,
        key: n,
        data: s,
        loading: !1,
        time: Date.now(),
        type: "mutate"
      });
    }
  };
}, ws = (e, { loadingDelay: t }) => {
  const n = E(), r = () => {
    n.value && (clearTimeout(w(n.value)), n.value = void 0);
  };
  return {
    name: "loadingDelayPlugin",
    onFinally: () => {
      r();
      const o = w(t);
      o && (e.setState({
        loading: !0
      }), n.value = setTimeout(() => {
        e.setState({
          loading: !1
        });
      }, o));
    },
    onError: () => {
      r();
    }
  };
};
function en() {
  return !!(typeof window < "u" && window.document && window.document.createElement);
}
const Ss = !!(typeof window < "u" && window.document && window.document.createElement);
function tn() {
  return en() ? document.visibilityState !== "hidden" : !0;
}
const Ne = [];
function Cs(e) {
  return Ne.push(e), function() {
    const n = Ne.indexOf(e);
    Ne.splice(n, 1);
  };
}
if (en()) {
  const e = () => {
    if (!!tn())
      for (let t = 0; t < Ne.length; t++) {
        const n = Ne[t];
        n();
      }
  };
  window.addEventListener("visibilitychange", e, !1);
}
const Ds = (e, { pollingInterval: t, pollingWhenHidden: n = !0, pollingErrorRetryCount: r = -1 }) => {
  let o;
  const u = E(), s = E(0), a = () => {
    var i;
    o && clearTimeout(o), (i = u.value) == null || i.call(u);
  };
  return k(() => {
    w(t) || a();
  }), w(t) ? {
    name: "pollingPlugin",
    onBefore: () => {
      a();
    },
    onError: () => {
      s.value += 1;
    },
    onSuccess: () => {
      s.value = 0;
    },
    onFinally: () => {
      r === -1 || r !== -1 && s.value <= r ? o = setTimeout(() => {
        !n && !tn() ? u.value = Cs(() => {
          e.refresh();
        }) : e.refresh();
      }, w(t)) : s.value = 0;
    },
    onCancel: () => {
      a();
    }
  } : {};
};
function Is(e, t) {
  let n = !1;
  return (...r) => {
    n || (n = !0, e(...r), setTimeout(() => {
      n = !1;
    }, t));
  };
}
function Ps() {
  return en() && typeof navigator.onLine < "u" ? navigator.onLine : !0;
}
const Ue = [];
function Rs(e) {
  return Ue.push(e), function() {
    const n = Ue.indexOf(e);
    n > -1 && Ue.splice(n, 1);
  };
}
if (Ss) {
  const e = () => {
    if (!(!tn() || !Ps()))
      for (let t = 0; t < Ue.length; t++) {
        const n = Ue[t];
        n();
      }
  };
  window.addEventListener("visibilitychange", e, !1), window.addEventListener("focus", e, !1);
}
const xs = (e, { refreshOnWindowFocus: t, focusTimespan: n = 5e3 }) => {
  const r = E(), o = () => {
    var u;
    (u = r.value) == null || u.call(r);
  };
  return k((u) => {
    if (w(t)) {
      const s = Is(
        e.refresh.bind(e),
        w(n)
      );
      r.value = Rs(() => {
        s();
      });
    }
    u(() => {
      o();
    });
  }), ft(() => {
    o();
  }), {
    name: "refreshOnWindowFocusPlugin"
  };
}, Fs = (e, { retryInterval: t, retryCount: n }) => {
  const r = E(), o = E(0), u = E(!1);
  return n ? {
    name: "retryPlugin",
    onBefore: () => {
      u.value || (o.value = 0), u.value = !1, r.value && clearTimeout(r.value);
    },
    onSuccess: () => {
      o.value = 0;
    },
    onError: () => {
      if (o.value += 1, n === -1 || o.value <= n) {
        const s = t != null ? t : Math.min(1e3 * 2 ** o.value, 3e4);
        r.value = setTimeout(() => {
          u.value = !0, e.refresh();
        }, s);
      } else
        o.value = 0;
    },
    onCancel: () => {
      o.value = 0, r.value && clearTimeout(r.value);
    }
  } : {};
};
var Ls = Array.isArray;
const de = Ls;
function vo(e) {
  return e;
}
var ks = "[object AsyncFunction]", Ns = "[object Function]", Us = "[object GeneratorFunction]", $s = "[object Proxy]";
function nn(e) {
  if (!G(e))
    return !1;
  var t = he(e);
  return t == Ns || t == Us || t == ks || t == $s;
}
var Bs = Y["__core-js_shared__"];
const Ot = Bs;
var In = function() {
  var e = /[^.]+$/.exec(Ot && Ot.keys && Ot.keys.IE_PROTO || "");
  return e ? "Symbol(src)_1." + e : "";
}();
function Vs(e) {
  return !!In && In in e;
}
var Ms = Function.prototype, js = Ms.toString;
function ve(e) {
  if (e != null) {
    try {
      return js.call(e);
    } catch {
    }
    try {
      return e + "";
    } catch {
    }
  }
  return "";
}
var Hs = /[\\^$.*+?()[\]{}|]/g, zs = /^\[object .+?Constructor\]$/, Ks = Function.prototype, qs = Object.prototype, Gs = Ks.toString, Ws = qs.hasOwnProperty, Ys = RegExp(
  "^" + Gs.call(Ws).replace(Hs, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
);
function Xs(e) {
  if (!G(e) || Vs(e))
    return !1;
  var t = nn(e) ? Ys : zs;
  return t.test(ve(e));
}
function Zs(e, t) {
  return e == null ? void 0 : e[t];
}
function me(e, t) {
  var n = Zs(e, t);
  return Xs(n) ? n : void 0;
}
var Js = me(Y, "WeakMap");
const Ft = Js;
var Pn = Object.create, Qs = function() {
  function e() {
  }
  return function(t) {
    if (!G(t))
      return {};
    if (Pn)
      return Pn(t);
    e.prototype = t;
    var n = new e();
    return e.prototype = void 0, n;
  };
}();
const ea = Qs;
function ta(e, t, n) {
  switch (n.length) {
    case 0:
      return e.call(t);
    case 1:
      return e.call(t, n[0]);
    case 2:
      return e.call(t, n[0], n[1]);
    case 3:
      return e.call(t, n[0], n[1], n[2]);
  }
  return e.apply(t, n);
}
function mo(e, t) {
  var n = -1, r = e.length;
  for (t || (t = Array(r)); ++n < r; )
    t[n] = e[n];
  return t;
}
var na = 800, ra = 16, oa = Date.now;
function ua(e) {
  var t = 0, n = 0;
  return function() {
    var r = oa(), o = ra - (r - n);
    if (n = r, o > 0) {
      if (++t >= na)
        return arguments[0];
    } else
      t = 0;
    return e.apply(void 0, arguments);
  };
}
function sa(e) {
  return function() {
    return e;
  };
}
var aa = function() {
  try {
    var e = me(Object, "defineProperty");
    return e({}, "", {}), e;
  } catch {
  }
}();
const at = aa;
var ia = at ? function(e, t) {
  return at(e, "toString", {
    configurable: !0,
    enumerable: !1,
    value: sa(t),
    writable: !0
  });
} : vo;
const la = ia;
var ca = ua(la);
const fa = ca;
function da(e, t) {
  for (var n = -1, r = e == null ? 0 : e.length; ++n < r && t(e[n], n, e) !== !1; )
    ;
  return e;
}
var ga = 9007199254740991, pa = /^(?:0|[1-9]\d*)$/;
function Eo(e, t) {
  var n = typeof e;
  return t = t == null ? ga : t, !!t && (n == "number" || n != "symbol" && pa.test(e)) && e > -1 && e % 1 == 0 && e < t;
}
function rn(e, t, n) {
  t == "__proto__" && at ? at(e, t, {
    configurable: !0,
    enumerable: !0,
    value: n,
    writable: !0
  }) : e[t] = n;
}
function Ye(e, t) {
  return e === t || e !== e && t !== t;
}
var _a = Object.prototype, ha = _a.hasOwnProperty;
function yo(e, t, n) {
  var r = e[t];
  (!(ha.call(e, t) && Ye(r, n)) || n === void 0 && !(t in e)) && rn(e, t, n);
}
function Xe(e, t, n, r) {
  var o = !n;
  n || (n = {});
  for (var u = -1, s = t.length; ++u < s; ) {
    var a = t[u], i = r ? r(n[a], e[a], a, n, e) : void 0;
    i === void 0 && (i = e[a]), o ? rn(n, a, i) : yo(n, a, i);
  }
  return n;
}
var Rn = Math.max;
function va(e, t, n) {
  return t = Rn(t === void 0 ? e.length - 1 : t, 0), function() {
    for (var r = arguments, o = -1, u = Rn(r.length - t, 0), s = Array(u); ++o < u; )
      s[o] = r[t + o];
    o = -1;
    for (var a = Array(t + 1); ++o < t; )
      a[o] = r[o];
    return a[t] = n(s), ta(e, this, a);
  };
}
function ma(e, t) {
  return fa(va(e, t, vo), e + "");
}
var Ea = 9007199254740991;
function bo(e) {
  return typeof e == "number" && e > -1 && e % 1 == 0 && e <= Ea;
}
function pt(e) {
  return e != null && bo(e.length) && !nn(e);
}
function ya(e, t, n) {
  if (!G(n))
    return !1;
  var r = typeof t;
  return (r == "number" ? pt(n) && Eo(t, n.length) : r == "string" && t in n) ? Ye(n[t], e) : !1;
}
function ba(e) {
  return ma(function(t, n) {
    var r = -1, o = n.length, u = o > 1 ? n[o - 1] : void 0, s = o > 2 ? n[2] : void 0;
    for (u = e.length > 3 && typeof u == "function" ? (o--, u) : void 0, s && ya(n[0], n[1], s) && (u = o < 3 ? void 0 : u, o = 1), t = Object(t); ++r < o; ) {
      var a = n[r];
      a && e(t, a, r, u);
    }
    return t;
  });
}
var Ta = Object.prototype;
function on(e) {
  var t = e && e.constructor, n = typeof t == "function" && t.prototype || Ta;
  return e === n;
}
function Aa(e, t) {
  for (var n = -1, r = Array(e); ++n < e; )
    r[n] = t(n);
  return r;
}
var Oa = "[object Arguments]";
function xn(e) {
  return ee(e) && he(e) == Oa;
}
var To = Object.prototype, wa = To.hasOwnProperty, Sa = To.propertyIsEnumerable, Ca = xn(function() {
  return arguments;
}()) ? xn : function(e) {
  return ee(e) && wa.call(e, "callee") && !Sa.call(e, "callee");
};
const Lt = Ca;
function Da() {
  return !1;
}
var Ao = typeof exports == "object" && exports && !exports.nodeType && exports, Fn = Ao && typeof module == "object" && module && !module.nodeType && module, Ia = Fn && Fn.exports === Ao, Ln = Ia ? Y.Buffer : void 0, Pa = Ln ? Ln.isBuffer : void 0, Ra = Pa || Da;
const Ve = Ra;
var xa = "[object Arguments]", Fa = "[object Array]", La = "[object Boolean]", ka = "[object Date]", Na = "[object Error]", Ua = "[object Function]", $a = "[object Map]", Ba = "[object Number]", Va = "[object Object]", Ma = "[object RegExp]", ja = "[object Set]", Ha = "[object String]", za = "[object WeakMap]", Ka = "[object ArrayBuffer]", qa = "[object DataView]", Ga = "[object Float32Array]", Wa = "[object Float64Array]", Ya = "[object Int8Array]", Xa = "[object Int16Array]", Za = "[object Int32Array]", Ja = "[object Uint8Array]", Qa = "[object Uint8ClampedArray]", ei = "[object Uint16Array]", ti = "[object Uint32Array]", P = {};
P[Ga] = P[Wa] = P[Ya] = P[Xa] = P[Za] = P[Ja] = P[Qa] = P[ei] = P[ti] = !0;
P[xa] = P[Fa] = P[Ka] = P[La] = P[qa] = P[ka] = P[Na] = P[Ua] = P[$a] = P[Ba] = P[Va] = P[Ma] = P[ja] = P[Ha] = P[za] = !1;
function ni(e) {
  return ee(e) && bo(e.length) && !!P[he(e)];
}
function un(e) {
  return function(t) {
    return e(t);
  };
}
var Oo = typeof exports == "object" && exports && !exports.nodeType && exports, $e = Oo && typeof module == "object" && module && !module.nodeType && module, ri = $e && $e.exports === Oo, wt = ri && _o.process, oi = function() {
  try {
    var e = $e && $e.require && $e.require("util").types;
    return e || wt && wt.binding && wt.binding("util");
  } catch {
  }
}();
const Se = oi;
var kn = Se && Se.isTypedArray, ui = kn ? un(kn) : ni;
const sn = ui;
var si = Object.prototype, ai = si.hasOwnProperty;
function wo(e, t) {
  var n = de(e), r = !n && Lt(e), o = !n && !r && Ve(e), u = !n && !r && !o && sn(e), s = n || r || o || u, a = s ? Aa(e.length, String) : [], i = a.length;
  for (var c in e)
    (t || ai.call(e, c)) && !(s && (c == "length" || o && (c == "offset" || c == "parent") || u && (c == "buffer" || c == "byteLength" || c == "byteOffset") || Eo(c, i))) && a.push(c);
  return a;
}
function So(e, t) {
  return function(n) {
    return e(t(n));
  };
}
var ii = So(Object.keys, Object);
const li = ii;
var ci = Object.prototype, fi = ci.hasOwnProperty;
function di(e) {
  if (!on(e))
    return li(e);
  var t = [];
  for (var n in Object(e))
    fi.call(e, n) && n != "constructor" && t.push(n);
  return t;
}
function an(e) {
  return pt(e) ? wo(e) : di(e);
}
function gi(e) {
  var t = [];
  if (e != null)
    for (var n in Object(e))
      t.push(n);
  return t;
}
var pi = Object.prototype, _i = pi.hasOwnProperty;
function hi(e) {
  if (!G(e))
    return gi(e);
  var t = on(e), n = [];
  for (var r in e)
    r == "constructor" && (t || !_i.call(e, r)) || n.push(r);
  return n;
}
function Ze(e) {
  return pt(e) ? wo(e, !0) : hi(e);
}
var vi = me(Object, "create");
const Me = vi;
function mi() {
  this.__data__ = Me ? Me(null) : {}, this.size = 0;
}
function Ei(e) {
  var t = this.has(e) && delete this.__data__[e];
  return this.size -= t ? 1 : 0, t;
}
var yi = "__lodash_hash_undefined__", bi = Object.prototype, Ti = bi.hasOwnProperty;
function Ai(e) {
  var t = this.__data__;
  if (Me) {
    var n = t[e];
    return n === yi ? void 0 : n;
  }
  return Ti.call(t, e) ? t[e] : void 0;
}
var Oi = Object.prototype, wi = Oi.hasOwnProperty;
function Si(e) {
  var t = this.__data__;
  return Me ? t[e] !== void 0 : wi.call(t, e);
}
var Ci = "__lodash_hash_undefined__";
function Di(e, t) {
  var n = this.__data__;
  return this.size += this.has(e) ? 0 : 1, n[e] = Me && t === void 0 ? Ci : t, this;
}
function ge(e) {
  var t = -1, n = e == null ? 0 : e.length;
  for (this.clear(); ++t < n; ) {
    var r = e[t];
    this.set(r[0], r[1]);
  }
}
ge.prototype.clear = mi;
ge.prototype.delete = Ei;
ge.prototype.get = Ai;
ge.prototype.has = Si;
ge.prototype.set = Di;
function Ii() {
  this.__data__ = [], this.size = 0;
}
function _t(e, t) {
  for (var n = e.length; n--; )
    if (Ye(e[n][0], t))
      return n;
  return -1;
}
var Pi = Array.prototype, Ri = Pi.splice;
function xi(e) {
  var t = this.__data__, n = _t(t, e);
  if (n < 0)
    return !1;
  var r = t.length - 1;
  return n == r ? t.pop() : Ri.call(t, n, 1), --this.size, !0;
}
function Fi(e) {
  var t = this.__data__, n = _t(t, e);
  return n < 0 ? void 0 : t[n][1];
}
function Li(e) {
  return _t(this.__data__, e) > -1;
}
function ki(e, t) {
  var n = this.__data__, r = _t(n, e);
  return r < 0 ? (++this.size, n.push([e, t])) : n[r][1] = t, this;
}
function ne(e) {
  var t = -1, n = e == null ? 0 : e.length;
  for (this.clear(); ++t < n; ) {
    var r = e[t];
    this.set(r[0], r[1]);
  }
}
ne.prototype.clear = Ii;
ne.prototype.delete = xi;
ne.prototype.get = Fi;
ne.prototype.has = Li;
ne.prototype.set = ki;
var Ni = me(Y, "Map");
const je = Ni;
function Ui() {
  this.size = 0, this.__data__ = {
    hash: new ge(),
    map: new (je || ne)(),
    string: new ge()
  };
}
function $i(e) {
  var t = typeof e;
  return t == "string" || t == "number" || t == "symbol" || t == "boolean" ? e !== "__proto__" : e === null;
}
function ht(e, t) {
  var n = e.__data__;
  return $i(t) ? n[typeof t == "string" ? "string" : "hash"] : n.map;
}
function Bi(e) {
  var t = ht(this, e).delete(e);
  return this.size -= t ? 1 : 0, t;
}
function Vi(e) {
  return ht(this, e).get(e);
}
function Mi(e) {
  return ht(this, e).has(e);
}
function ji(e, t) {
  var n = ht(this, e), r = n.size;
  return n.set(e, t), this.size += n.size == r ? 0 : 1, this;
}
function Ee(e) {
  var t = -1, n = e == null ? 0 : e.length;
  for (this.clear(); ++t < n; ) {
    var r = e[t];
    this.set(r[0], r[1]);
  }
}
Ee.prototype.clear = Ui;
Ee.prototype.delete = Bi;
Ee.prototype.get = Vi;
Ee.prototype.has = Mi;
Ee.prototype.set = ji;
function Co(e, t) {
  for (var n = -1, r = t.length, o = e.length; ++n < r; )
    e[o + n] = t[n];
  return e;
}
var Hi = So(Object.getPrototypeOf, Object);
const ln = Hi;
var zi = "[object Object]", Ki = Function.prototype, qi = Object.prototype, Do = Ki.toString, Gi = qi.hasOwnProperty, Wi = Do.call(Object);
function Yi(e) {
  if (!ee(e) || he(e) != zi)
    return !1;
  var t = ln(e);
  if (t === null)
    return !0;
  var n = Gi.call(t, "constructor") && t.constructor;
  return typeof n == "function" && n instanceof n && Do.call(n) == Wi;
}
function Xi() {
  this.__data__ = new ne(), this.size = 0;
}
function Zi(e) {
  var t = this.__data__, n = t.delete(e);
  return this.size = t.size, n;
}
function Ji(e) {
  return this.__data__.get(e);
}
function Qi(e) {
  return this.__data__.has(e);
}
var el = 200;
function tl(e, t) {
  var n = this.__data__;
  if (n instanceof ne) {
    var r = n.__data__;
    if (!je || r.length < el - 1)
      return r.push([e, t]), this.size = ++n.size, this;
    n = this.__data__ = new Ee(r);
  }
  return n.set(e, t), this.size = n.size, this;
}
function Q(e) {
  var t = this.__data__ = new ne(e);
  this.size = t.size;
}
Q.prototype.clear = Xi;
Q.prototype.delete = Zi;
Q.prototype.get = Ji;
Q.prototype.has = Qi;
Q.prototype.set = tl;
function nl(e, t) {
  return e && Xe(t, an(t), e);
}
function rl(e, t) {
  return e && Xe(t, Ze(t), e);
}
var Io = typeof exports == "object" && exports && !exports.nodeType && exports, Nn = Io && typeof module == "object" && module && !module.nodeType && module, ol = Nn && Nn.exports === Io, Un = ol ? Y.Buffer : void 0, $n = Un ? Un.allocUnsafe : void 0;
function Po(e, t) {
  if (t)
    return e.slice();
  var n = e.length, r = $n ? $n(n) : new e.constructor(n);
  return e.copy(r), r;
}
function ul(e, t) {
  for (var n = -1, r = e == null ? 0 : e.length, o = 0, u = []; ++n < r; ) {
    var s = e[n];
    t(s, n, e) && (u[o++] = s);
  }
  return u;
}
function Ro() {
  return [];
}
var sl = Object.prototype, al = sl.propertyIsEnumerable, Bn = Object.getOwnPropertySymbols, il = Bn ? function(e) {
  return e == null ? [] : (e = Object(e), ul(Bn(e), function(t) {
    return al.call(e, t);
  }));
} : Ro;
const cn = il;
function ll(e, t) {
  return Xe(e, cn(e), t);
}
var cl = Object.getOwnPropertySymbols, fl = cl ? function(e) {
  for (var t = []; e; )
    Co(t, cn(e)), e = ln(e);
  return t;
} : Ro;
const xo = fl;
function dl(e, t) {
  return Xe(e, xo(e), t);
}
function Fo(e, t, n) {
  var r = t(e);
  return de(e) ? r : Co(r, n(e));
}
function kt(e) {
  return Fo(e, an, cn);
}
function gl(e) {
  return Fo(e, Ze, xo);
}
var pl = me(Y, "DataView");
const Nt = pl;
var _l = me(Y, "Promise");
const Ut = _l;
var hl = me(Y, "Set");
const $t = hl;
var Vn = "[object Map]", vl = "[object Object]", Mn = "[object Promise]", jn = "[object Set]", Hn = "[object WeakMap]", zn = "[object DataView]", ml = ve(Nt), El = ve(je), yl = ve(Ut), bl = ve($t), Tl = ve(Ft), ie = he;
(Nt && ie(new Nt(new ArrayBuffer(1))) != zn || je && ie(new je()) != Vn || Ut && ie(Ut.resolve()) != Mn || $t && ie(new $t()) != jn || Ft && ie(new Ft()) != Hn) && (ie = function(e) {
  var t = he(e), n = t == vl ? e.constructor : void 0, r = n ? ve(n) : "";
  if (r)
    switch (r) {
      case ml:
        return zn;
      case El:
        return Vn;
      case yl:
        return Mn;
      case bl:
        return jn;
      case Tl:
        return Hn;
    }
  return t;
});
const He = ie;
var Al = Object.prototype, Ol = Al.hasOwnProperty;
function wl(e) {
  var t = e.length, n = new e.constructor(t);
  return t && typeof e[0] == "string" && Ol.call(e, "index") && (n.index = e.index, n.input = e.input), n;
}
var Sl = Y.Uint8Array;
const it = Sl;
function fn(e) {
  var t = new e.constructor(e.byteLength);
  return new it(t).set(new it(e)), t;
}
function Cl(e, t) {
  var n = t ? fn(e.buffer) : e.buffer;
  return new e.constructor(n, e.byteOffset, e.byteLength);
}
var Dl = /\w*$/;
function Il(e) {
  var t = new e.constructor(e.source, Dl.exec(e));
  return t.lastIndex = e.lastIndex, t;
}
var Kn = oe ? oe.prototype : void 0, qn = Kn ? Kn.valueOf : void 0;
function Pl(e) {
  return qn ? Object(qn.call(e)) : {};
}
function Lo(e, t) {
  var n = t ? fn(e.buffer) : e.buffer;
  return new e.constructor(n, e.byteOffset, e.length);
}
var Rl = "[object Boolean]", xl = "[object Date]", Fl = "[object Map]", Ll = "[object Number]", kl = "[object RegExp]", Nl = "[object Set]", Ul = "[object String]", $l = "[object Symbol]", Bl = "[object ArrayBuffer]", Vl = "[object DataView]", Ml = "[object Float32Array]", jl = "[object Float64Array]", Hl = "[object Int8Array]", zl = "[object Int16Array]", Kl = "[object Int32Array]", ql = "[object Uint8Array]", Gl = "[object Uint8ClampedArray]", Wl = "[object Uint16Array]", Yl = "[object Uint32Array]";
function Xl(e, t, n) {
  var r = e.constructor;
  switch (t) {
    case Bl:
      return fn(e);
    case Rl:
    case xl:
      return new r(+e);
    case Vl:
      return Cl(e, n);
    case Ml:
    case jl:
    case Hl:
    case zl:
    case Kl:
    case ql:
    case Gl:
    case Wl:
    case Yl:
      return Lo(e, n);
    case Fl:
      return new r();
    case Ll:
    case Ul:
      return new r(e);
    case kl:
      return Il(e);
    case Nl:
      return new r();
    case $l:
      return Pl(e);
  }
}
function ko(e) {
  return typeof e.constructor == "function" && !on(e) ? ea(ln(e)) : {};
}
var Zl = "[object Map]";
function Jl(e) {
  return ee(e) && He(e) == Zl;
}
var Gn = Se && Se.isMap, Ql = Gn ? un(Gn) : Jl;
const ec = Ql;
var tc = "[object Set]";
function nc(e) {
  return ee(e) && He(e) == tc;
}
var Wn = Se && Se.isSet, rc = Wn ? un(Wn) : nc;
const oc = rc;
var uc = 1, sc = 2, ac = 4, No = "[object Arguments]", ic = "[object Array]", lc = "[object Boolean]", cc = "[object Date]", fc = "[object Error]", Uo = "[object Function]", dc = "[object GeneratorFunction]", gc = "[object Map]", pc = "[object Number]", $o = "[object Object]", _c = "[object RegExp]", hc = "[object Set]", vc = "[object String]", mc = "[object Symbol]", Ec = "[object WeakMap]", yc = "[object ArrayBuffer]", bc = "[object DataView]", Tc = "[object Float32Array]", Ac = "[object Float64Array]", Oc = "[object Int8Array]", wc = "[object Int16Array]", Sc = "[object Int32Array]", Cc = "[object Uint8Array]", Dc = "[object Uint8ClampedArray]", Ic = "[object Uint16Array]", Pc = "[object Uint32Array]", D = {};
D[No] = D[ic] = D[yc] = D[bc] = D[lc] = D[cc] = D[Tc] = D[Ac] = D[Oc] = D[wc] = D[Sc] = D[gc] = D[pc] = D[$o] = D[_c] = D[hc] = D[vc] = D[mc] = D[Cc] = D[Dc] = D[Ic] = D[Pc] = !0;
D[fc] = D[Uo] = D[Ec] = !1;
function ot(e, t, n, r, o, u) {
  var s, a = t & uc, i = t & sc, c = t & ac;
  if (n && (s = o ? n(e, r, o, u) : n(e)), s !== void 0)
    return s;
  if (!G(e))
    return e;
  var f = de(e);
  if (f) {
    if (s = wl(e), !a)
      return mo(e, s);
  } else {
    var l = He(e), g = l == Uo || l == dc;
    if (Ve(e))
      return Po(e, a);
    if (l == $o || l == No || g && !o) {
      if (s = i || g ? {} : ko(e), !a)
        return i ? dl(e, rl(s, e)) : ll(e, nl(s, e));
    } else {
      if (!D[l])
        return o ? e : {};
      s = Xl(e, l, a);
    }
  }
  u || (u = new Q());
  var d = u.get(e);
  if (d)
    return d;
  u.set(e, s), oc(e) ? e.forEach(function(p) {
    s.add(ot(p, t, n, p, e, u));
  }) : ec(e) && e.forEach(function(p, _) {
    s.set(_, ot(p, t, n, _, e, u));
  });
  var h = c ? i ? gl : kt : i ? Ze : an, v = f ? void 0 : h(e);
  return da(v || e, function(p, _) {
    v && (_ = p, p = e[_]), yo(s, _, ot(p, t, n, _, e, u));
  }), s;
}
var Rc = 1, xc = 4;
function Yn(e) {
  return ot(e, Rc | xc);
}
var Fc = "__lodash_hash_undefined__";
function Lc(e) {
  return this.__data__.set(e, Fc), this;
}
function kc(e) {
  return this.__data__.has(e);
}
function lt(e) {
  var t = -1, n = e == null ? 0 : e.length;
  for (this.__data__ = new Ee(); ++t < n; )
    this.add(e[t]);
}
lt.prototype.add = lt.prototype.push = Lc;
lt.prototype.has = kc;
function Nc(e, t) {
  for (var n = -1, r = e == null ? 0 : e.length; ++n < r; )
    if (t(e[n], n, e))
      return !0;
  return !1;
}
function Uc(e, t) {
  return e.has(t);
}
var $c = 1, Bc = 2;
function Bo(e, t, n, r, o, u) {
  var s = n & $c, a = e.length, i = t.length;
  if (a != i && !(s && i > a))
    return !1;
  var c = u.get(e), f = u.get(t);
  if (c && f)
    return c == t && f == e;
  var l = -1, g = !0, d = n & Bc ? new lt() : void 0;
  for (u.set(e, t), u.set(t, e); ++l < a; ) {
    var h = e[l], v = t[l];
    if (r)
      var p = s ? r(v, h, l, t, e, u) : r(h, v, l, e, t, u);
    if (p !== void 0) {
      if (p)
        continue;
      g = !1;
      break;
    }
    if (d) {
      if (!Nc(t, function(_, y) {
        if (!Uc(d, y) && (h === _ || o(h, _, n, r, u)))
          return d.push(y);
      })) {
        g = !1;
        break;
      }
    } else if (!(h === v || o(h, v, n, r, u))) {
      g = !1;
      break;
    }
  }
  return u.delete(e), u.delete(t), g;
}
function Vc(e) {
  var t = -1, n = Array(e.size);
  return e.forEach(function(r, o) {
    n[++t] = [o, r];
  }), n;
}
function Mc(e) {
  var t = -1, n = Array(e.size);
  return e.forEach(function(r) {
    n[++t] = r;
  }), n;
}
var jc = 1, Hc = 2, zc = "[object Boolean]", Kc = "[object Date]", qc = "[object Error]", Gc = "[object Map]", Wc = "[object Number]", Yc = "[object RegExp]", Xc = "[object Set]", Zc = "[object String]", Jc = "[object Symbol]", Qc = "[object ArrayBuffer]", ef = "[object DataView]", Xn = oe ? oe.prototype : void 0, St = Xn ? Xn.valueOf : void 0;
function tf(e, t, n, r, o, u, s) {
  switch (n) {
    case ef:
      if (e.byteLength != t.byteLength || e.byteOffset != t.byteOffset)
        return !1;
      e = e.buffer, t = t.buffer;
    case Qc:
      return !(e.byteLength != t.byteLength || !u(new it(e), new it(t)));
    case zc:
    case Kc:
    case Wc:
      return Ye(+e, +t);
    case qc:
      return e.name == t.name && e.message == t.message;
    case Yc:
    case Zc:
      return e == t + "";
    case Gc:
      var a = Vc;
    case Xc:
      var i = r & jc;
      if (a || (a = Mc), e.size != t.size && !i)
        return !1;
      var c = s.get(e);
      if (c)
        return c == t;
      r |= Hc, s.set(e, t);
      var f = Bo(a(e), a(t), r, o, u, s);
      return s.delete(e), f;
    case Jc:
      if (St)
        return St.call(e) == St.call(t);
  }
  return !1;
}
var nf = 1, rf = Object.prototype, of = rf.hasOwnProperty;
function uf(e, t, n, r, o, u) {
  var s = n & nf, a = kt(e), i = a.length, c = kt(t), f = c.length;
  if (i != f && !s)
    return !1;
  for (var l = i; l--; ) {
    var g = a[l];
    if (!(s ? g in t : of.call(t, g)))
      return !1;
  }
  var d = u.get(e), h = u.get(t);
  if (d && h)
    return d == t && h == e;
  var v = !0;
  u.set(e, t), u.set(t, e);
  for (var p = s; ++l < i; ) {
    g = a[l];
    var _ = e[g], y = t[g];
    if (r)
      var T = s ? r(y, _, g, t, e, u) : r(_, y, g, e, t, u);
    if (!(T === void 0 ? _ === y || o(_, y, n, r, u) : T)) {
      v = !1;
      break;
    }
    p || (p = g == "constructor");
  }
  if (v && !p) {
    var S = e.constructor, A = t.constructor;
    S != A && "constructor" in e && "constructor" in t && !(typeof S == "function" && S instanceof S && typeof A == "function" && A instanceof A) && (v = !1);
  }
  return u.delete(e), u.delete(t), v;
}
var sf = 1, Zn = "[object Arguments]", Jn = "[object Array]", et = "[object Object]", af = Object.prototype, Qn = af.hasOwnProperty;
function lf(e, t, n, r, o, u) {
  var s = de(e), a = de(t), i = s ? Jn : He(e), c = a ? Jn : He(t);
  i = i == Zn ? et : i, c = c == Zn ? et : c;
  var f = i == et, l = c == et, g = i == c;
  if (g && Ve(e)) {
    if (!Ve(t))
      return !1;
    s = !0, f = !1;
  }
  if (g && !f)
    return u || (u = new Q()), s || sn(e) ? Bo(e, t, n, r, o, u) : tf(e, t, i, n, r, o, u);
  if (!(n & sf)) {
    var d = f && Qn.call(e, "__wrapped__"), h = l && Qn.call(t, "__wrapped__");
    if (d || h) {
      var v = d ? e.value() : e, p = h ? t.value() : t;
      return u || (u = new Q()), o(v, p, n, r, u);
    }
  }
  return g ? (u || (u = new Q()), uf(e, t, n, r, o, u)) : !1;
}
function Vo(e, t, n, r, o) {
  return e === t ? !0 : e == null || t == null || !ee(e) && !ee(t) ? e !== e && t !== t : lf(e, t, n, r, Vo, o);
}
function cf(e) {
  return function(t, n, r) {
    for (var o = -1, u = Object(t), s = r(t), a = s.length; a--; ) {
      var i = s[e ? a : ++o];
      if (n(u[i], i, u) === !1)
        break;
    }
    return t;
  };
}
var ff = cf();
const df = ff;
function Bt(e, t, n) {
  (n !== void 0 && !Ye(e[t], n) || n === void 0 && !(t in e)) && rn(e, t, n);
}
function gf(e) {
  return ee(e) && pt(e);
}
function Vt(e, t) {
  if (!(t === "constructor" && typeof e[t] == "function") && t != "__proto__")
    return e[t];
}
function pf(e) {
  return Xe(e, Ze(e));
}
function _f(e, t, n, r, o, u, s) {
  var a = Vt(e, n), i = Vt(t, n), c = s.get(i);
  if (c) {
    Bt(e, n, c);
    return;
  }
  var f = u ? u(a, i, n + "", e, t, s) : void 0, l = f === void 0;
  if (l) {
    var g = de(i), d = !g && Ve(i), h = !g && !d && sn(i);
    f = i, g || d || h ? de(a) ? f = a : gf(a) ? f = mo(a) : d ? (l = !1, f = Po(i, !0)) : h ? (l = !1, f = Lo(i, !0)) : f = [] : Yi(i) || Lt(i) ? (f = a, Lt(a) ? f = pf(a) : (!G(a) || nn(a)) && (f = ko(i))) : l = !1;
  }
  l && (s.set(i, f), o(f, i, r, u, s), s.delete(i)), Bt(e, n, f);
}
function Mo(e, t, n, r, o) {
  e !== t && df(t, function(u, s) {
    if (o || (o = new Q()), G(u))
      _f(e, t, s, n, Mo, r, o);
    else {
      var a = r ? r(Vt(e, s), u, s + "", e, t, o) : void 0;
      a === void 0 && (a = u), Bt(e, s, a);
    }
  }, Ze);
}
function hf(e, t) {
  return Vo(e, t);
}
var vf = ba(function(e, t, n) {
  Mo(e, t, n);
});
const mf = vf;
var Ef = "Expected a function";
function jo(e, t, n) {
  var r = !0, o = !0;
  if (typeof e != "function")
    throw new TypeError(Ef);
  return G(n) && (r = "leading" in n ? !!n.leading : r, o = "trailing" in n ? !!n.trailing : o), Qt(e, t, {
    leading: r,
    maxWait: t,
    trailing: o
  });
}
const yf = (e, { throttleWait: t, throttleLeading: n, throttleTrailing: r }) => {
  let o = null;
  const u = $(() => {
    const c = {};
    return w(n) !== void 0 && (c.leading = w(n)), w(r) !== void 0 && (c.trailing = w(r)), c;
  }), s = e.runAsync.bind(e), a = E(), i = $(() => w(t));
  return B([i, u], (c) => {
    o && (o.cancel(), e.runAsync = s);
    const [f, l] = c, g = jo(
      (d) => {
        d();
      },
      w(f),
      l
    );
    o = g, a.value = g, e.runAsync = (...d) => new Promise((h, v) => {
      var p;
      (p = a.value) == null || p.call(a, () => {
        s(...d).then(h).catch(v);
      });
    });
  }, {
    immediate: !0
  }), w(t) ? (_e(() => {
    var c;
    (c = a.value) == null || c.cancel();
  }), {
    name: "throttlePlugin",
    onCancel: () => {
      var c;
      (c = a.value) == null || c.cancel();
    }
  }) : {};
}, er = (e) => typeof e == "function", bf = (e) => typeof e == "boolean";
class Tf {
  constructor(t, n, r, o = {}) {
    H(this, "pluginImpls");
    H(this, "count", 0);
    H(this, "state", {
      loading: !1,
      params: void 0,
      data: void 0,
      error: void 0
    });
    H(this, "previousValidData");
    this.serviceRef = t, this.options = n, this.setUpdateData = r, this.initState = o, this.state = {
      ...this.state,
      loading: !n.manual,
      ...o
    };
  }
  setState(t = {}) {
    this.state = {
      ...this.state,
      ...t
    }, this.setUpdateData(this.state);
  }
  setData(t, n) {
    console.warn("Please use 'setFetchState' instead of 'setData'"), n instanceof Array ? n.forEach((r) => {
      this.state[r] = t, this.setUpdateData(t, r);
    }) : (this.state[n] = t, this.setUpdateData(t, n));
  }
  setFetchState(t, n) {
    n instanceof Array ? n.forEach((r) => {
      this.state[r] = t, this.setUpdateData(t, r);
    }) : (this.state[n] = t, this.setUpdateData(t, n));
  }
  runPluginHandler(t, ...n) {
    var o, u, s;
    const r = (s = (u = (o = this.pluginImpls) == null ? void 0 : o.map((a) => {
      var i;
      return (i = a[t]) == null ? void 0 : i.call(a, ...n);
    })) != null ? u : []) == null ? void 0 : s.filter(Boolean);
    return Object.assign({}, ...r);
  }
  async runAsync(...t) {
    var s, a, i, c, f, l, g, d, h, v, p;
    this.count += 1;
    const n = this.count, { stopNow: r = !1, returnNow: o = !1, ...u } = this.runPluginHandler(
      "onBefore",
      t
    );
    if (r)
      return new Promise(() => {
      });
    if (this.setState({
      loading: !0,
      params: t,
      ...u
    }), o)
      return Promise.resolve(u.data);
    try {
      (a = (s = this.options).onBefore) == null || a.call(s, t);
    } catch (_) {
      return this.setState({
        error: _,
        loading: !1
      }), (c = (i = this.options).onError) == null || c.call(i, _, t), this.runPluginHandler("onError", _, t), new Promise(() => {
      });
    }
    try {
      let { servicePromise: _ } = this.runPluginHandler("onRequest", this.serviceRef.value, t);
      const y = (S) => {
        var F, L, M, R;
        if (n !== this.count)
          return new Promise(() => {
          });
        const A = this.options.formatResult ? this.options.formatResult(S) : S, O = {
          data: S
        };
        return this.setState({
          data: A,
          error: void 0,
          loading: !1
        }), (L = (F = this.options).onSuccess) == null || L.call(F, A, t), this.runPluginHandler("onSuccess", A, t, O), this.previousValidData = A, (R = (M = this.options).onFinally) == null || R.call(M, t, A, void 0), n === this.count && this.runPluginHandler("onFinally", t, A, void 0), A;
      };
      _ || (_ = this.serviceRef.value(...t));
      const T = await _;
      return y(T);
    } catch (_) {
      if (n !== this.count)
        return new Promise(() => {
        });
      throw this.setState({
        error: _,
        loading: !1
      }), (l = (f = this.options).onError) == null || l.call(f, _, t), this.runPluginHandler("onError", _, t), (er((g = this.options) == null ? void 0 : g.rollbackOnError) && ((d = this.options) == null ? void 0 : d.rollbackOnError(t)) || bf((h = this.options) == null ? void 0 : h.rollbackOnError) && this.options.rollbackOnError) && this.setState({
        data: this.previousValidData
      }), (p = (v = this.options).onFinally) == null || p.call(v, t, void 0, _), n === this.count && this.runPluginHandler("onFinally", t, void 0, _), _;
    }
  }
  run(...t) {
    this.runAsync(...t).catch((n) => {
      this.options.onError || console.error(n);
    });
  }
  cancel() {
    this.count += 1, this.setState({
      loading: !1
    }), this.runPluginHandler("onCancel");
  }
  refresh() {
    this.run(...this.state.params || []);
  }
  refreshAsync() {
    return this.runAsync(...this.state.params || []);
  }
  mutate(t) {
    const n = er(t) ? t(this.state.data) : t;
    this.runPluginHandler("onMutate", n), this.setState({
      data: n
    });
  }
}
const Ho = Symbol(
  "USEREQUEST_GLOBAL_OPTIONS_PROVIDE_KEY"
);
function Af(e) {
  return Object.keys(e).filter((n) => ["data", "loading", "params", "error"].includes(n)).length === 4;
}
function Of(e, t = {}, n = []) {
  const r = Fu(
    Ho,
    {}
  ), { initialData: o = void 0, manual: u = !1, ready: s = !0, ...a } = {
    ...r != null ? r : {},
    ...t != null ? t : {}
  }, i = {
    manual: u,
    ready: s,
    initialData: o,
    ...a
  }, c = E(e), f = dt({
    data: o,
    loading: !1,
    params: void 0,
    error: void 0
  }), l = (v, p) => {
    p ? f[p] = v : Af(v) && (f.data = v.data, f.loading = v.loading, f.error = v.error, f.params = v.params);
  }, g = n.map((v) => {
    var p;
    return (p = v == null ? void 0 : v.onInit) == null ? void 0 : p.call(v, i);
  }).filter(Boolean), d = new Tf(
    c,
    i,
    l,
    Object.assign({}, ...g, f)
  );
  d.options = i, d.pluginImpls = n.map((v) => v(d, i));
  const h = $(() => fe(s) ? s.value : s);
  if (k(() => {
    if (!u) {
      const v = d.state.params || t.defaultParams || [];
      h.value && d.options.refreshDeps === !0 && !!c.value && d.run(...v);
    }
  }), !u && d.options.refreshDeps !== !0) {
    const v = d.state.params || t.defaultParams || [];
    w(s) && d.run(...v);
  }
  return ft(() => {
    d.cancel();
  }), {
    ...Jt(f),
    cancel: d.cancel.bind(d),
    refresh: d.refresh.bind(d),
    refreshAsync: d.refreshAsync.bind(d),
    run: d.run.bind(d),
    runAsync: d.runAsync.bind(d),
    mutate: d.mutate.bind(d)
  };
}
const wf = (e, t) => function(r, o = {}, u = []) {
  let s = e;
  const a = t || [];
  for (let i = a.length; i--; )
    s = a[i](s);
  return s(r, o, u);
};
function zo(e, t, n) {
  const r = [
    process.env.NODE_ENV === "development" ? Os : null,
    Es,
    ws,
    Ds,
    xs,
    yf,
    po,
    qu,
    Fs
  ].filter(Boolean);
  return wf(Of, t == null ? void 0 : t.use)(
    e,
    t,
    [...n || [], ...r]
  );
}
function hg(e) {
  Lu(Ho, e);
}
var Sf = Object.create, Ko = Object.defineProperty, Cf = Object.getOwnPropertyDescriptor, dn = Object.getOwnPropertyNames, Df = Object.getPrototypeOf, If = Object.prototype.hasOwnProperty, Pf = (e, t) => function() {
  return e && (t = (0, e[dn(e)[0]])(e = 0)), t;
}, Rf = (e, t) => function() {
  return t || (0, e[dn(e)[0]])((t = { exports: {} }).exports, t), t.exports;
}, xf = (e, t, n, r) => {
  if (t && typeof t == "object" || typeof t == "function")
    for (let o of dn(t))
      !If.call(e, o) && o !== n && Ko(e, o, { get: () => t[o], enumerable: !(r = Cf(t, o)) || r.enumerable });
  return e;
}, Ff = (e, t, n) => (n = e != null ? Sf(Df(e)) : {}, xf(
  t || !e || !e.__esModule ? Ko(n, "default", { value: e, enumerable: !0 }) : n,
  e
)), Je = Pf({
  "../../node_modules/.pnpm/tsup@8.3.5_@microsoft+api-extractor@7.48.1_@types+node@22.10.5__jiti@2.4.2_postcss@8.4.49_tsx_s7k37zks4wtn7x2grzma6lrsfa/node_modules/tsup/assets/esm_shims.js"() {
  }
}), Lf = Rf({
  "../../node_modules/.pnpm/rfdc@1.4.1/node_modules/rfdc/index.js"(e, t) {
    Je(), t.exports = r;
    function n(u) {
      return u instanceof Buffer ? Buffer.from(u) : new u.constructor(u.buffer.slice(), u.byteOffset, u.length);
    }
    function r(u) {
      if (u = u || {}, u.circles)
        return o(u);
      const s = /* @__PURE__ */ new Map();
      if (s.set(Date, (l) => new Date(l)), s.set(Map, (l, g) => new Map(i(Array.from(l), g))), s.set(Set, (l, g) => new Set(i(Array.from(l), g))), u.constructorHandlers)
        for (const l of u.constructorHandlers)
          s.set(l[0], l[1]);
      let a = null;
      return u.proto ? f : c;
      function i(l, g) {
        const d = Object.keys(l), h = new Array(d.length);
        for (let v = 0; v < d.length; v++) {
          const p = d[v], _ = l[p];
          typeof _ != "object" || _ === null ? h[p] = _ : _.constructor !== Object && (a = s.get(_.constructor)) ? h[p] = a(_, g) : ArrayBuffer.isView(_) ? h[p] = n(_) : h[p] = g(_);
        }
        return h;
      }
      function c(l) {
        if (typeof l != "object" || l === null)
          return l;
        if (Array.isArray(l))
          return i(l, c);
        if (l.constructor !== Object && (a = s.get(l.constructor)))
          return a(l, c);
        const g = {};
        for (const d in l) {
          if (Object.hasOwnProperty.call(l, d) === !1)
            continue;
          const h = l[d];
          typeof h != "object" || h === null ? g[d] = h : h.constructor !== Object && (a = s.get(h.constructor)) ? g[d] = a(h, c) : ArrayBuffer.isView(h) ? g[d] = n(h) : g[d] = c(h);
        }
        return g;
      }
      function f(l) {
        if (typeof l != "object" || l === null)
          return l;
        if (Array.isArray(l))
          return i(l, f);
        if (l.constructor !== Object && (a = s.get(l.constructor)))
          return a(l, f);
        const g = {};
        for (const d in l) {
          const h = l[d];
          typeof h != "object" || h === null ? g[d] = h : h.constructor !== Object && (a = s.get(h.constructor)) ? g[d] = a(h, f) : ArrayBuffer.isView(h) ? g[d] = n(h) : g[d] = f(h);
        }
        return g;
      }
    }
    function o(u) {
      const s = [], a = [], i = /* @__PURE__ */ new Map();
      if (i.set(Date, (d) => new Date(d)), i.set(Map, (d, h) => new Map(f(Array.from(d), h))), i.set(Set, (d, h) => new Set(f(Array.from(d), h))), u.constructorHandlers)
        for (const d of u.constructorHandlers)
          i.set(d[0], d[1]);
      let c = null;
      return u.proto ? g : l;
      function f(d, h) {
        const v = Object.keys(d), p = new Array(v.length);
        for (let _ = 0; _ < v.length; _++) {
          const y = v[_], T = d[y];
          if (typeof T != "object" || T === null)
            p[y] = T;
          else if (T.constructor !== Object && (c = i.get(T.constructor)))
            p[y] = c(T, h);
          else if (ArrayBuffer.isView(T))
            p[y] = n(T);
          else {
            const S = s.indexOf(T);
            S !== -1 ? p[y] = a[S] : p[y] = h(T);
          }
        }
        return p;
      }
      function l(d) {
        if (typeof d != "object" || d === null)
          return d;
        if (Array.isArray(d))
          return f(d, l);
        if (d.constructor !== Object && (c = i.get(d.constructor)))
          return c(d, l);
        const h = {};
        s.push(d), a.push(h);
        for (const v in d) {
          if (Object.hasOwnProperty.call(d, v) === !1)
            continue;
          const p = d[v];
          if (typeof p != "object" || p === null)
            h[v] = p;
          else if (p.constructor !== Object && (c = i.get(p.constructor)))
            h[v] = c(p, l);
          else if (ArrayBuffer.isView(p))
            h[v] = n(p);
          else {
            const _ = s.indexOf(p);
            _ !== -1 ? h[v] = a[_] : h[v] = l(p);
          }
        }
        return s.pop(), a.pop(), h;
      }
      function g(d) {
        if (typeof d != "object" || d === null)
          return d;
        if (Array.isArray(d))
          return f(d, g);
        if (d.constructor !== Object && (c = i.get(d.constructor)))
          return c(d, g);
        const h = {};
        s.push(d), a.push(h);
        for (const v in d) {
          const p = d[v];
          if (typeof p != "object" || p === null)
            h[v] = p;
          else if (p.constructor !== Object && (c = i.get(p.constructor)))
            h[v] = c(p, g);
          else if (ArrayBuffer.isView(p))
            h[v] = n(p);
          else {
            const _ = s.indexOf(p);
            _ !== -1 ? h[v] = a[_] : h[v] = g(p);
          }
        }
        return s.pop(), a.pop(), h;
      }
    }
  }
});
Je();
Je();
Je();
var qo = typeof navigator < "u", b = typeof window < "u" ? window : typeof globalThis < "u" ? globalThis : typeof global < "u" ? global : {};
typeof b.chrome < "u" && b.chrome.devtools;
qo && (b.self, b.top);
var tr;
typeof navigator < "u" && ((tr = navigator.userAgent) == null || tr.toLowerCase().includes("electron"));
Je();
var kf = Ff(Lf(), 1), Nf = /(?:^|[-_/])(\w)/g;
function Uf(e, t) {
  return t ? t.toUpperCase() : "";
}
function $f(e) {
  return e && `${e}`.replace(Nf, Uf);
}
function Bf(e, t) {
  let n = e.replace(/^[a-z]:/i, "").replace(/\\/g, "/");
  n.endsWith(`index${t}`) && (n = n.replace(`/index${t}`, t));
  const r = n.lastIndexOf("/"), o = n.substring(r + 1);
  if (t) {
    const u = o.lastIndexOf(t);
    return o.substring(0, u);
  }
  return "";
}
var nr = (0, kf.default)({ circles: !0 });
const Vf = {
  trailing: !0
};
function Ce(e, t = 25, n = {}) {
  if (n = { ...Vf, ...n }, !Number.isFinite(t))
    throw new TypeError("Expected `wait` to be a finite number");
  let r, o, u = [], s, a;
  const i = (c, f) => (s = Mf(e, c, f), s.finally(() => {
    if (s = null, n.trailing && a && !o) {
      const l = i(c, a);
      return a = null, l;
    }
  }), s);
  return function(...c) {
    return s ? (n.trailing && (a = c), s) : new Promise((f) => {
      const l = !o && n.leading;
      clearTimeout(o), o = setTimeout(() => {
        o = null;
        const g = n.leading ? r : i(this, c);
        for (const d of u)
          d(g);
        u = [];
      }, t), l ? (r = i(this, c), f(r)) : u.push(f);
    });
  };
}
async function Mf(e, t, n) {
  return await e.apply(t, n);
}
function Mt(e, t = {}, n) {
  for (const r in e) {
    const o = e[r], u = n ? `${n}:${r}` : r;
    typeof o == "object" && o !== null ? Mt(o, t, u) : typeof o == "function" && (t[u] = o);
  }
  return t;
}
const jf = { run: (e) => e() }, Hf = () => jf, Go = typeof console.createTask < "u" ? console.createTask : Hf;
function zf(e, t) {
  const n = t.shift(), r = Go(n);
  return e.reduce(
    (o, u) => o.then(() => r.run(() => u(...t))),
    Promise.resolve()
  );
}
function Kf(e, t) {
  const n = t.shift(), r = Go(n);
  return Promise.all(e.map((o) => r.run(() => o(...t))));
}
function Ct(e, t) {
  for (const n of [...e])
    n(t);
}
class qf {
  constructor() {
    this._hooks = {}, this._before = void 0, this._after = void 0, this._deprecatedMessages = void 0, this._deprecatedHooks = {}, this.hook = this.hook.bind(this), this.callHook = this.callHook.bind(this), this.callHookWith = this.callHookWith.bind(this);
  }
  hook(t, n, r = {}) {
    if (!t || typeof n != "function")
      return () => {
      };
    const o = t;
    let u;
    for (; this._deprecatedHooks[t]; )
      u = this._deprecatedHooks[t], t = u.to;
    if (u && !r.allowDeprecated) {
      let s = u.message;
      s || (s = `${o} hook has been deprecated` + (u.to ? `, please use ${u.to}` : "")), this._deprecatedMessages || (this._deprecatedMessages = /* @__PURE__ */ new Set()), this._deprecatedMessages.has(s) || (console.warn(s), this._deprecatedMessages.add(s));
    }
    if (!n.name)
      try {
        Object.defineProperty(n, "name", {
          get: () => "_" + t.replace(/\W+/g, "_") + "_hook_cb",
          configurable: !0
        });
      } catch {
      }
    return this._hooks[t] = this._hooks[t] || [], this._hooks[t].push(n), () => {
      n && (this.removeHook(t, n), n = void 0);
    };
  }
  hookOnce(t, n) {
    let r, o = (...u) => (typeof r == "function" && r(), r = void 0, o = void 0, n(...u));
    return r = this.hook(t, o), r;
  }
  removeHook(t, n) {
    if (this._hooks[t]) {
      const r = this._hooks[t].indexOf(n);
      r !== -1 && this._hooks[t].splice(r, 1), this._hooks[t].length === 0 && delete this._hooks[t];
    }
  }
  deprecateHook(t, n) {
    this._deprecatedHooks[t] = typeof n == "string" ? { to: n } : n;
    const r = this._hooks[t] || [];
    delete this._hooks[t];
    for (const o of r)
      this.hook(t, o);
  }
  deprecateHooks(t) {
    Object.assign(this._deprecatedHooks, t);
    for (const n in t)
      this.deprecateHook(n, t[n]);
  }
  addHooks(t) {
    const n = Mt(t), r = Object.keys(n).map(
      (o) => this.hook(o, n[o])
    );
    return () => {
      for (const o of r.splice(0, r.length))
        o();
    };
  }
  removeHooks(t) {
    const n = Mt(t);
    for (const r in n)
      this.removeHook(r, n[r]);
  }
  removeAllHooks() {
    for (const t in this._hooks)
      delete this._hooks[t];
  }
  callHook(t, ...n) {
    return n.unshift(t), this.callHookWith(zf, t, ...n);
  }
  callHookParallel(t, ...n) {
    return n.unshift(t), this.callHookWith(Kf, t, ...n);
  }
  callHookWith(t, n, ...r) {
    const o = this._before || this._after ? { name: n, args: r, context: {} } : void 0;
    this._before && Ct(this._before, o);
    const u = t(
      n in this._hooks ? [...this._hooks[n]] : [],
      r
    );
    return u instanceof Promise ? u.finally(() => {
      this._after && o && Ct(this._after, o);
    }) : (this._after && o && Ct(this._after, o), u);
  }
  beforeEach(t) {
    return this._before = this._before || [], this._before.push(t), () => {
      if (this._before !== void 0) {
        const n = this._before.indexOf(t);
        n !== -1 && this._before.splice(n, 1);
      }
    };
  }
  afterEach(t) {
    return this._after = this._after || [], this._after.push(t), () => {
      if (this._after !== void 0) {
        const n = this._after.indexOf(t);
        n !== -1 && this._after.splice(n, 1);
      }
    };
  }
}
function Wo() {
  return new qf();
}
var Gf = Object.create, Yo = Object.defineProperty, Wf = Object.getOwnPropertyDescriptor, gn = Object.getOwnPropertyNames, Yf = Object.getPrototypeOf, Xf = Object.prototype.hasOwnProperty, Zf = (e, t) => function() {
  return e && (t = (0, e[gn(e)[0]])(e = 0)), t;
}, Xo = (e, t) => function() {
  return t || (0, e[gn(e)[0]])((t = { exports: {} }).exports, t), t.exports;
}, Jf = (e, t, n, r) => {
  if (t && typeof t == "object" || typeof t == "function")
    for (let o of gn(t))
      !Xf.call(e, o) && o !== n && Yo(e, o, { get: () => t[o], enumerable: !(r = Wf(t, o)) || r.enumerable });
  return e;
}, Qf = (e, t, n) => (n = e != null ? Gf(Yf(e)) : {}, Jf(
  t || !e || !e.__esModule ? Yo(n, "default", { value: e, enumerable: !0 }) : n,
  e
)), m = Zf({
  "../../node_modules/.pnpm/tsup@8.3.5_@microsoft+api-extractor@7.48.1_@types+node@22.10.5__jiti@2.4.2_postcss@8.4.49_tsx_s7k37zks4wtn7x2grzma6lrsfa/node_modules/tsup/assets/esm_shims.js"() {
  }
}), ed = Xo({
  "../../node_modules/.pnpm/speakingurl@14.0.1/node_modules/speakingurl/lib/speakingurl.js"(e, t) {
    m(), function(n) {
      var r = {
        \u00C0: "A",
        \u00C1: "A",
        \u00C2: "A",
        \u00C3: "A",
        \u00C4: "Ae",
        \u00C5: "A",
        \u00C6: "AE",
        \u00C7: "C",
        \u00C8: "E",
        \u00C9: "E",
        \u00CA: "E",
        \u00CB: "E",
        \u00CC: "I",
        \u00CD: "I",
        \u00CE: "I",
        \u00CF: "I",
        \u00D0: "D",
        \u00D1: "N",
        \u00D2: "O",
        \u00D3: "O",
        \u00D4: "O",
        \u00D5: "O",
        \u00D6: "Oe",
        \u0150: "O",
        \u00D8: "O",
        \u00D9: "U",
        \u00DA: "U",
        \u00DB: "U",
        \u00DC: "Ue",
        \u0170: "U",
        \u00DD: "Y",
        \u00DE: "TH",
        \u00DF: "ss",
        \u00E0: "a",
        \u00E1: "a",
        \u00E2: "a",
        \u00E3: "a",
        \u00E4: "ae",
        \u00E5: "a",
        \u00E6: "ae",
        \u00E7: "c",
        \u00E8: "e",
        \u00E9: "e",
        \u00EA: "e",
        \u00EB: "e",
        \u00EC: "i",
        \u00ED: "i",
        \u00EE: "i",
        \u00EF: "i",
        \u00F0: "d",
        \u00F1: "n",
        \u00F2: "o",
        \u00F3: "o",
        \u00F4: "o",
        \u00F5: "o",
        \u00F6: "oe",
        \u0151: "o",
        \u00F8: "o",
        \u00F9: "u",
        \u00FA: "u",
        \u00FB: "u",
        \u00FC: "ue",
        \u0171: "u",
        \u00FD: "y",
        \u00FE: "th",
        \u00FF: "y",
        "\u1E9E": "SS",
        \u0627: "a",
        \u0623: "a",
        \u0625: "i",
        \u0622: "aa",
        \u0624: "u",
        \u0626: "e",
        \u0621: "a",
        \u0628: "b",
        \u062A: "t",
        \u062B: "th",
        \u062C: "j",
        \u062D: "h",
        \u062E: "kh",
        \u062F: "d",
        \u0630: "th",
        \u0631: "r",
        \u0632: "z",
        \u0633: "s",
        \u0634: "sh",
        \u0635: "s",
        \u0636: "dh",
        \u0637: "t",
        \u0638: "z",
        \u0639: "a",
        \u063A: "gh",
        \u0641: "f",
        \u0642: "q",
        \u0643: "k",
        \u0644: "l",
        \u0645: "m",
        \u0646: "n",
        \u0647: "h",
        \u0648: "w",
        \u064A: "y",
        \u0649: "a",
        \u0629: "h",
        \uFEFB: "la",
        \uFEF7: "laa",
        \uFEF9: "lai",
        \uFEF5: "laa",
        \u06AF: "g",
        \u0686: "ch",
        \u067E: "p",
        \u0698: "zh",
        \u06A9: "k",
        \u06CC: "y",
        "\u064E": "a",
        "\u064B": "an",
        "\u0650": "e",
        "\u064D": "en",
        "\u064F": "u",
        "\u064C": "on",
        "\u0652": "",
        "\u0660": "0",
        "\u0661": "1",
        "\u0662": "2",
        "\u0663": "3",
        "\u0664": "4",
        "\u0665": "5",
        "\u0666": "6",
        "\u0667": "7",
        "\u0668": "8",
        "\u0669": "9",
        "\u06F0": "0",
        "\u06F1": "1",
        "\u06F2": "2",
        "\u06F3": "3",
        "\u06F4": "4",
        "\u06F5": "5",
        "\u06F6": "6",
        "\u06F7": "7",
        "\u06F8": "8",
        "\u06F9": "9",
        \u1000: "k",
        \u1001: "kh",
        \u1002: "g",
        \u1003: "ga",
        \u1004: "ng",
        \u1005: "s",
        \u1006: "sa",
        \u1007: "z",
        "\u1005\u103B": "za",
        \u100A: "ny",
        \u100B: "t",
        \u100C: "ta",
        \u100D: "d",
        \u100E: "da",
        \u100F: "na",
        \u1010: "t",
        \u1011: "ta",
        \u1012: "d",
        \u1013: "da",
        \u1014: "n",
        \u1015: "p",
        \u1016: "pa",
        \u1017: "b",
        \u1018: "ba",
        \u1019: "m",
        \u101A: "y",
        \u101B: "ya",
        \u101C: "l",
        \u101D: "w",
        \u101E: "th",
        \u101F: "h",
        \u1020: "la",
        \u1021: "a",
        "\u103C": "y",
        "\u103B": "ya",
        "\u103D": "w",
        "\u103C\u103D": "yw",
        "\u103B\u103D": "ywa",
        "\u103E": "h",
        \u1027: "e",
        "\u104F": "-e",
        \u1023: "i",
        \u1024: "-i",
        \u1009: "u",
        \u1026: "-u",
        \u1029: "aw",
        "\u101E\u103C\u1031\u102C": "aw",
        \u102A: "aw",
        "\u1040": "0",
        "\u1041": "1",
        "\u1042": "2",
        "\u1043": "3",
        "\u1044": "4",
        "\u1045": "5",
        "\u1046": "6",
        "\u1047": "7",
        "\u1048": "8",
        "\u1049": "9",
        "\u1039": "",
        "\u1037": "",
        "\u1038": "",
        \u010D: "c",
        \u010F: "d",
        \u011B: "e",
        \u0148: "n",
        \u0159: "r",
        \u0161: "s",
        \u0165: "t",
        \u016F: "u",
        \u017E: "z",
        \u010C: "C",
        \u010E: "D",
        \u011A: "E",
        \u0147: "N",
        \u0158: "R",
        \u0160: "S",
        \u0164: "T",
        \u016E: "U",
        \u017D: "Z",
        \u0780: "h",
        \u0781: "sh",
        \u0782: "n",
        \u0783: "r",
        \u0784: "b",
        \u0785: "lh",
        \u0786: "k",
        \u0787: "a",
        \u0788: "v",
        \u0789: "m",
        \u078A: "f",
        \u078B: "dh",
        \u078C: "th",
        \u078D: "l",
        \u078E: "g",
        \u078F: "gn",
        \u0790: "s",
        \u0791: "d",
        \u0792: "z",
        \u0793: "t",
        \u0794: "y",
        \u0795: "p",
        \u0796: "j",
        \u0797: "ch",
        \u0798: "tt",
        \u0799: "hh",
        \u079A: "kh",
        \u079B: "th",
        \u079C: "z",
        \u079D: "sh",
        \u079E: "s",
        \u079F: "d",
        \u07A0: "t",
        \u07A1: "z",
        \u07A2: "a",
        \u07A3: "gh",
        \u07A4: "q",
        \u07A5: "w",
        "\u07A6": "a",
        "\u07A7": "aa",
        "\u07A8": "i",
        "\u07A9": "ee",
        "\u07AA": "u",
        "\u07AB": "oo",
        "\u07AC": "e",
        "\u07AD": "ey",
        "\u07AE": "o",
        "\u07AF": "oa",
        "\u07B0": "",
        \u10D0: "a",
        \u10D1: "b",
        \u10D2: "g",
        \u10D3: "d",
        \u10D4: "e",
        \u10D5: "v",
        \u10D6: "z",
        \u10D7: "t",
        \u10D8: "i",
        \u10D9: "k",
        \u10DA: "l",
        \u10DB: "m",
        \u10DC: "n",
        \u10DD: "o",
        \u10DE: "p",
        \u10DF: "zh",
        \u10E0: "r",
        \u10E1: "s",
        \u10E2: "t",
        \u10E3: "u",
        \u10E4: "p",
        \u10E5: "k",
        \u10E6: "gh",
        \u10E7: "q",
        \u10E8: "sh",
        \u10E9: "ch",
        \u10EA: "ts",
        \u10EB: "dz",
        \u10EC: "ts",
        \u10ED: "ch",
        \u10EE: "kh",
        \u10EF: "j",
        \u10F0: "h",
        \u03B1: "a",
        \u03B2: "v",
        \u03B3: "g",
        \u03B4: "d",
        \u03B5: "e",
        \u03B6: "z",
        \u03B7: "i",
        \u03B8: "th",
        \u03B9: "i",
        \u03BA: "k",
        \u03BB: "l",
        \u03BC: "m",
        \u03BD: "n",
        \u03BE: "ks",
        \u03BF: "o",
        \u03C0: "p",
        \u03C1: "r",
        \u03C3: "s",
        \u03C4: "t",
        \u03C5: "y",
        \u03C6: "f",
        \u03C7: "x",
        \u03C8: "ps",
        \u03C9: "o",
        \u03AC: "a",
        \u03AD: "e",
        \u03AF: "i",
        \u03CC: "o",
        \u03CD: "y",
        \u03AE: "i",
        \u03CE: "o",
        \u03C2: "s",
        \u03CA: "i",
        \u03B0: "y",
        \u03CB: "y",
        \u0390: "i",
        \u0391: "A",
        \u0392: "B",
        \u0393: "G",
        \u0394: "D",
        \u0395: "E",
        \u0396: "Z",
        \u0397: "I",
        \u0398: "TH",
        \u0399: "I",
        \u039A: "K",
        \u039B: "L",
        \u039C: "M",
        \u039D: "N",
        \u039E: "KS",
        \u039F: "O",
        \u03A0: "P",
        \u03A1: "R",
        \u03A3: "S",
        \u03A4: "T",
        \u03A5: "Y",
        \u03A6: "F",
        \u03A7: "X",
        \u03A8: "PS",
        \u03A9: "O",
        \u0386: "A",
        \u0388: "E",
        \u038A: "I",
        \u038C: "O",
        \u038E: "Y",
        \u0389: "I",
        \u038F: "O",
        \u03AA: "I",
        \u03AB: "Y",
        \u0101: "a",
        \u0113: "e",
        \u0123: "g",
        \u012B: "i",
        \u0137: "k",
        \u013C: "l",
        \u0146: "n",
        \u016B: "u",
        \u0100: "A",
        \u0112: "E",
        \u0122: "G",
        \u012A: "I",
        \u0136: "k",
        \u013B: "L",
        \u0145: "N",
        \u016A: "U",
        \u040C: "Kj",
        \u045C: "kj",
        \u0409: "Lj",
        \u0459: "lj",
        \u040A: "Nj",
        \u045A: "nj",
        \u0422\u0441: "Ts",
        \u0442\u0441: "ts",
        \u0105: "a",
        \u0107: "c",
        \u0119: "e",
        \u0142: "l",
        \u0144: "n",
        \u015B: "s",
        \u017A: "z",
        \u017C: "z",
        \u0104: "A",
        \u0106: "C",
        \u0118: "E",
        \u0141: "L",
        \u0143: "N",
        \u015A: "S",
        \u0179: "Z",
        \u017B: "Z",
        \u0404: "Ye",
        \u0406: "I",
        \u0407: "Yi",
        \u0490: "G",
        \u0454: "ye",
        \u0456: "i",
        \u0457: "yi",
        \u0491: "g",
        \u0103: "a",
        \u0102: "A",
        \u0219: "s",
        \u0218: "S",
        \u021B: "t",
        \u021A: "T",
        \u0163: "t",
        \u0162: "T",
        \u0430: "a",
        \u0431: "b",
        \u0432: "v",
        \u0433: "g",
        \u0434: "d",
        \u0435: "e",
        \u0451: "yo",
        \u0436: "zh",
        \u0437: "z",
        \u0438: "i",
        \u0439: "i",
        \u043A: "k",
        \u043B: "l",
        \u043C: "m",
        \u043D: "n",
        \u043E: "o",
        \u043F: "p",
        \u0440: "r",
        \u0441: "s",
        \u0442: "t",
        \u0443: "u",
        \u0444: "f",
        \u0445: "kh",
        \u0446: "c",
        \u0447: "ch",
        \u0448: "sh",
        \u0449: "sh",
        \u044A: "",
        \u044B: "y",
        \u044C: "",
        \u044D: "e",
        \u044E: "yu",
        \u044F: "ya",
        \u0410: "A",
        \u0411: "B",
        \u0412: "V",
        \u0413: "G",
        \u0414: "D",
        \u0415: "E",
        \u0401: "Yo",
        \u0416: "Zh",
        \u0417: "Z",
        \u0418: "I",
        \u0419: "I",
        \u041A: "K",
        \u041B: "L",
        \u041C: "M",
        \u041D: "N",
        \u041E: "O",
        \u041F: "P",
        \u0420: "R",
        \u0421: "S",
        \u0422: "T",
        \u0423: "U",
        \u0424: "F",
        \u0425: "Kh",
        \u0426: "C",
        \u0427: "Ch",
        \u0428: "Sh",
        \u0429: "Sh",
        \u042A: "",
        \u042B: "Y",
        \u042C: "",
        \u042D: "E",
        \u042E: "Yu",
        \u042F: "Ya",
        \u0452: "dj",
        \u0458: "j",
        \u045B: "c",
        \u045F: "dz",
        \u0402: "Dj",
        \u0408: "j",
        \u040B: "C",
        \u040F: "Dz",
        \u013E: "l",
        \u013A: "l",
        \u0155: "r",
        \u013D: "L",
        \u0139: "L",
        \u0154: "R",
        \u015F: "s",
        \u015E: "S",
        \u0131: "i",
        \u0130: "I",
        \u011F: "g",
        \u011E: "G",
        \u1EA3: "a",
        \u1EA2: "A",
        \u1EB3: "a",
        \u1EB2: "A",
        \u1EA9: "a",
        \u1EA8: "A",
        \u0111: "d",
        \u0110: "D",
        \u1EB9: "e",
        \u1EB8: "E",
        \u1EBD: "e",
        \u1EBC: "E",
        \u1EBB: "e",
        \u1EBA: "E",
        \u1EBF: "e",
        \u1EBE: "E",
        \u1EC1: "e",
        \u1EC0: "E",
        \u1EC7: "e",
        \u1EC6: "E",
        \u1EC5: "e",
        \u1EC4: "E",
        \u1EC3: "e",
        \u1EC2: "E",
        \u1ECF: "o",
        \u1ECD: "o",
        \u1ECC: "o",
        \u1ED1: "o",
        \u1ED0: "O",
        \u1ED3: "o",
        \u1ED2: "O",
        \u1ED5: "o",
        \u1ED4: "O",
        \u1ED9: "o",
        \u1ED8: "O",
        \u1ED7: "o",
        \u1ED6: "O",
        \u01A1: "o",
        \u01A0: "O",
        \u1EDB: "o",
        \u1EDA: "O",
        \u1EDD: "o",
        \u1EDC: "O",
        \u1EE3: "o",
        \u1EE2: "O",
        \u1EE1: "o",
        \u1EE0: "O",
        \u1EDE: "o",
        \u1EDF: "o",
        \u1ECB: "i",
        \u1ECA: "I",
        \u0129: "i",
        \u0128: "I",
        \u1EC9: "i",
        \u1EC8: "i",
        \u1EE7: "u",
        \u1EE6: "U",
        \u1EE5: "u",
        \u1EE4: "U",
        \u0169: "u",
        \u0168: "U",
        \u01B0: "u",
        \u01AF: "U",
        \u1EE9: "u",
        \u1EE8: "U",
        \u1EEB: "u",
        \u1EEA: "U",
        \u1EF1: "u",
        \u1EF0: "U",
        \u1EEF: "u",
        \u1EEE: "U",
        \u1EED: "u",
        \u1EEC: "\u01B0",
        \u1EF7: "y",
        \u1EF6: "y",
        \u1EF3: "y",
        \u1EF2: "Y",
        \u1EF5: "y",
        \u1EF4: "Y",
        \u1EF9: "y",
        \u1EF8: "Y",
        \u1EA1: "a",
        \u1EA0: "A",
        \u1EA5: "a",
        \u1EA4: "A",
        \u1EA7: "a",
        \u1EA6: "A",
        \u1EAD: "a",
        \u1EAC: "A",
        \u1EAB: "a",
        \u1EAA: "A",
        \u1EAF: "a",
        \u1EAE: "A",
        \u1EB1: "a",
        \u1EB0: "A",
        \u1EB7: "a",
        \u1EB6: "A",
        \u1EB5: "a",
        \u1EB4: "A",
        "\u24EA": "0",
        "\u2460": "1",
        "\u2461": "2",
        "\u2462": "3",
        "\u2463": "4",
        "\u2464": "5",
        "\u2465": "6",
        "\u2466": "7",
        "\u2467": "8",
        "\u2468": "9",
        "\u2469": "10",
        "\u246A": "11",
        "\u246B": "12",
        "\u246C": "13",
        "\u246D": "14",
        "\u246E": "15",
        "\u246F": "16",
        "\u2470": "17",
        "\u2471": "18",
        "\u2472": "18",
        "\u2473": "18",
        "\u24F5": "1",
        "\u24F6": "2",
        "\u24F7": "3",
        "\u24F8": "4",
        "\u24F9": "5",
        "\u24FA": "6",
        "\u24FB": "7",
        "\u24FC": "8",
        "\u24FD": "9",
        "\u24FE": "10",
        "\u24FF": "0",
        "\u24EB": "11",
        "\u24EC": "12",
        "\u24ED": "13",
        "\u24EE": "14",
        "\u24EF": "15",
        "\u24F0": "16",
        "\u24F1": "17",
        "\u24F2": "18",
        "\u24F3": "19",
        "\u24F4": "20",
        "\u24B6": "A",
        "\u24B7": "B",
        "\u24B8": "C",
        "\u24B9": "D",
        "\u24BA": "E",
        "\u24BB": "F",
        "\u24BC": "G",
        "\u24BD": "H",
        "\u24BE": "I",
        "\u24BF": "J",
        "\u24C0": "K",
        "\u24C1": "L",
        "\u24C2": "M",
        "\u24C3": "N",
        "\u24C4": "O",
        "\u24C5": "P",
        "\u24C6": "Q",
        "\u24C7": "R",
        "\u24C8": "S",
        "\u24C9": "T",
        "\u24CA": "U",
        "\u24CB": "V",
        "\u24CC": "W",
        "\u24CD": "X",
        "\u24CE": "Y",
        "\u24CF": "Z",
        "\u24D0": "a",
        "\u24D1": "b",
        "\u24D2": "c",
        "\u24D3": "d",
        "\u24D4": "e",
        "\u24D5": "f",
        "\u24D6": "g",
        "\u24D7": "h",
        "\u24D8": "i",
        "\u24D9": "j",
        "\u24DA": "k",
        "\u24DB": "l",
        "\u24DC": "m",
        "\u24DD": "n",
        "\u24DE": "o",
        "\u24DF": "p",
        "\u24E0": "q",
        "\u24E1": "r",
        "\u24E2": "s",
        "\u24E3": "t",
        "\u24E4": "u",
        "\u24E6": "v",
        "\u24E5": "w",
        "\u24E7": "x",
        "\u24E8": "y",
        "\u24E9": "z",
        "\u201C": '"',
        "\u201D": '"',
        "\u2018": "'",
        "\u2019": "'",
        "\u2202": "d",
        \u0192: "f",
        "\u2122": "(TM)",
        "\xA9": "(C)",
        \u0153: "oe",
        \u0152: "OE",
        "\xAE": "(R)",
        "\u2020": "+",
        "\u2120": "(SM)",
        "\u2026": "...",
        "\u02DA": "o",
        \u00BA: "o",
        \u00AA: "a",
        "\u2022": "*",
        "\u104A": ",",
        "\u104B": ".",
        $: "USD",
        "\u20AC": "EUR",
        "\u20A2": "BRN",
        "\u20A3": "FRF",
        "\xA3": "GBP",
        "\u20A4": "ITL",
        "\u20A6": "NGN",
        "\u20A7": "ESP",
        "\u20A9": "KRW",
        "\u20AA": "ILS",
        "\u20AB": "VND",
        "\u20AD": "LAK",
        "\u20AE": "MNT",
        "\u20AF": "GRD",
        "\u20B1": "ARS",
        "\u20B2": "PYG",
        "\u20B3": "ARA",
        "\u20B4": "UAH",
        "\u20B5": "GHS",
        "\xA2": "cent",
        "\xA5": "CNY",
        \u5143: "CNY",
        \u5186: "YEN",
        "\uFDFC": "IRR",
        "\u20A0": "EWE",
        "\u0E3F": "THB",
        "\u20A8": "INR",
        "\u20B9": "INR",
        "\u20B0": "PF",
        "\u20BA": "TRY",
        "\u060B": "AFN",
        "\u20BC": "AZN",
        \u043B\u0432: "BGN",
        "\u17DB": "KHR",
        "\u20A1": "CRC",
        "\u20B8": "KZT",
        \u0434\u0435\u043D: "MKD",
        z\u0142: "PLN",
        "\u20BD": "RUB",
        "\u20BE": "GEL"
      }, o = [
        "\u103A",
        "\u07B0"
      ], u = {
        "\u102C": "a",
        "\u102B": "a",
        "\u1031": "e",
        "\u1032": "e",
        "\u102D": "i",
        "\u102E": "i",
        "\u102D\u102F": "o",
        "\u102F": "u",
        "\u1030": "u",
        "\u1031\u102B\u1004\u103A": "aung",
        "\u1031\u102C": "aw",
        "\u1031\u102C\u103A": "aw",
        "\u1031\u102B": "aw",
        "\u1031\u102B\u103A": "aw",
        "\u103A": "\u103A",
        "\u1000\u103A": "et",
        "\u102D\u102F\u1000\u103A": "aik",
        "\u1031\u102C\u1000\u103A": "auk",
        "\u1004\u103A": "in",
        "\u102D\u102F\u1004\u103A": "aing",
        "\u1031\u102C\u1004\u103A": "aung",
        "\u1005\u103A": "it",
        "\u100A\u103A": "i",
        "\u1010\u103A": "at",
        "\u102D\u1010\u103A": "eik",
        "\u102F\u1010\u103A": "ok",
        "\u103D\u1010\u103A": "ut",
        "\u1031\u1010\u103A": "it",
        "\u1012\u103A": "d",
        "\u102D\u102F\u1012\u103A": "ok",
        "\u102F\u1012\u103A": "ait",
        "\u1014\u103A": "an",
        "\u102C\u1014\u103A": "an",
        "\u102D\u1014\u103A": "ein",
        "\u102F\u1014\u103A": "on",
        "\u103D\u1014\u103A": "un",
        "\u1015\u103A": "at",
        "\u102D\u1015\u103A": "eik",
        "\u102F\u1015\u103A": "ok",
        "\u103D\u1015\u103A": "ut",
        "\u1014\u103A\u102F\u1015\u103A": "nub",
        "\u1019\u103A": "an",
        "\u102D\u1019\u103A": "ein",
        "\u102F\u1019\u103A": "on",
        "\u103D\u1019\u103A": "un",
        "\u101A\u103A": "e",
        "\u102D\u102F\u101C\u103A": "ol",
        "\u1009\u103A": "in",
        "\u1036": "an",
        "\u102D\u1036": "ein",
        "\u102F\u1036": "on",
        "\u07A6\u0787\u07B0": "ah",
        "\u07A6\u0781\u07B0": "ah"
      }, s = {
        en: {},
        az: {
          \u00E7: "c",
          \u0259: "e",
          \u011F: "g",
          \u0131: "i",
          \u00F6: "o",
          \u015F: "s",
          \u00FC: "u",
          \u00C7: "C",
          \u018F: "E",
          \u011E: "G",
          \u0130: "I",
          \u00D6: "O",
          \u015E: "S",
          \u00DC: "U"
        },
        cs: {
          \u010D: "c",
          \u010F: "d",
          \u011B: "e",
          \u0148: "n",
          \u0159: "r",
          \u0161: "s",
          \u0165: "t",
          \u016F: "u",
          \u017E: "z",
          \u010C: "C",
          \u010E: "D",
          \u011A: "E",
          \u0147: "N",
          \u0158: "R",
          \u0160: "S",
          \u0164: "T",
          \u016E: "U",
          \u017D: "Z"
        },
        fi: {
          \u00E4: "a",
          \u00C4: "A",
          \u00F6: "o",
          \u00D6: "O"
        },
        hu: {
          \u00E4: "a",
          \u00C4: "A",
          \u00F6: "o",
          \u00D6: "O",
          \u00FC: "u",
          \u00DC: "U",
          \u0171: "u",
          \u0170: "U"
        },
        lt: {
          \u0105: "a",
          \u010D: "c",
          \u0119: "e",
          \u0117: "e",
          \u012F: "i",
          \u0161: "s",
          \u0173: "u",
          \u016B: "u",
          \u017E: "z",
          \u0104: "A",
          \u010C: "C",
          \u0118: "E",
          \u0116: "E",
          \u012E: "I",
          \u0160: "S",
          \u0172: "U",
          \u016A: "U"
        },
        lv: {
          \u0101: "a",
          \u010D: "c",
          \u0113: "e",
          \u0123: "g",
          \u012B: "i",
          \u0137: "k",
          \u013C: "l",
          \u0146: "n",
          \u0161: "s",
          \u016B: "u",
          \u017E: "z",
          \u0100: "A",
          \u010C: "C",
          \u0112: "E",
          \u0122: "G",
          \u012A: "i",
          \u0136: "k",
          \u013B: "L",
          \u0145: "N",
          \u0160: "S",
          \u016A: "u",
          \u017D: "Z"
        },
        pl: {
          \u0105: "a",
          \u0107: "c",
          \u0119: "e",
          \u0142: "l",
          \u0144: "n",
          \u00F3: "o",
          \u015B: "s",
          \u017A: "z",
          \u017C: "z",
          \u0104: "A",
          \u0106: "C",
          \u0118: "e",
          \u0141: "L",
          \u0143: "N",
          \u00D3: "O",
          \u015A: "S",
          \u0179: "Z",
          \u017B: "Z"
        },
        sv: {
          \u00E4: "a",
          \u00C4: "A",
          \u00F6: "o",
          \u00D6: "O"
        },
        sk: {
          \u00E4: "a",
          \u00C4: "A"
        },
        sr: {
          \u0459: "lj",
          \u045A: "nj",
          \u0409: "Lj",
          \u040A: "Nj",
          \u0111: "dj",
          \u0110: "Dj"
        },
        tr: {
          \u00DC: "U",
          \u00D6: "O",
          \u00FC: "u",
          \u00F6: "o"
        }
      }, a = {
        ar: {
          "\u2206": "delta",
          "\u221E": "la-nihaya",
          "\u2665": "hob",
          "&": "wa",
          "|": "aw",
          "<": "aqal-men",
          ">": "akbar-men",
          "\u2211": "majmou",
          "\xA4": "omla"
        },
        az: {},
        ca: {
          "\u2206": "delta",
          "\u221E": "infinit",
          "\u2665": "amor",
          "&": "i",
          "|": "o",
          "<": "menys que",
          ">": "mes que",
          "\u2211": "suma dels",
          "\xA4": "moneda"
        },
        cs: {
          "\u2206": "delta",
          "\u221E": "nekonecno",
          "\u2665": "laska",
          "&": "a",
          "|": "nebo",
          "<": "mensi nez",
          ">": "vetsi nez",
          "\u2211": "soucet",
          "\xA4": "mena"
        },
        de: {
          "\u2206": "delta",
          "\u221E": "unendlich",
          "\u2665": "Liebe",
          "&": "und",
          "|": "oder",
          "<": "kleiner als",
          ">": "groesser als",
          "\u2211": "Summe von",
          "\xA4": "Waehrung"
        },
        dv: {
          "\u2206": "delta",
          "\u221E": "kolunulaa",
          "\u2665": "loabi",
          "&": "aai",
          "|": "noonee",
          "<": "ah vure kuda",
          ">": "ah vure bodu",
          "\u2211": "jumula",
          "\xA4": "faisaa"
        },
        en: {
          "\u2206": "delta",
          "\u221E": "infinity",
          "\u2665": "love",
          "&": "and",
          "|": "or",
          "<": "less than",
          ">": "greater than",
          "\u2211": "sum",
          "\xA4": "currency"
        },
        es: {
          "\u2206": "delta",
          "\u221E": "infinito",
          "\u2665": "amor",
          "&": "y",
          "|": "u",
          "<": "menos que",
          ">": "mas que",
          "\u2211": "suma de los",
          "\xA4": "moneda"
        },
        fa: {
          "\u2206": "delta",
          "\u221E": "bi-nahayat",
          "\u2665": "eshgh",
          "&": "va",
          "|": "ya",
          "<": "kamtar-az",
          ">": "bishtar-az",
          "\u2211": "majmooe",
          "\xA4": "vahed"
        },
        fi: {
          "\u2206": "delta",
          "\u221E": "aarettomyys",
          "\u2665": "rakkaus",
          "&": "ja",
          "|": "tai",
          "<": "pienempi kuin",
          ">": "suurempi kuin",
          "\u2211": "summa",
          "\xA4": "valuutta"
        },
        fr: {
          "\u2206": "delta",
          "\u221E": "infiniment",
          "\u2665": "Amour",
          "&": "et",
          "|": "ou",
          "<": "moins que",
          ">": "superieure a",
          "\u2211": "somme des",
          "\xA4": "monnaie"
        },
        ge: {
          "\u2206": "delta",
          "\u221E": "usasruloba",
          "\u2665": "siqvaruli",
          "&": "da",
          "|": "an",
          "<": "naklebi",
          ">": "meti",
          "\u2211": "jami",
          "\xA4": "valuta"
        },
        gr: {},
        hu: {
          "\u2206": "delta",
          "\u221E": "vegtelen",
          "\u2665": "szerelem",
          "&": "es",
          "|": "vagy",
          "<": "kisebb mint",
          ">": "nagyobb mint",
          "\u2211": "szumma",
          "\xA4": "penznem"
        },
        it: {
          "\u2206": "delta",
          "\u221E": "infinito",
          "\u2665": "amore",
          "&": "e",
          "|": "o",
          "<": "minore di",
          ">": "maggiore di",
          "\u2211": "somma",
          "\xA4": "moneta"
        },
        lt: {
          "\u2206": "delta",
          "\u221E": "begalybe",
          "\u2665": "meile",
          "&": "ir",
          "|": "ar",
          "<": "maziau nei",
          ">": "daugiau nei",
          "\u2211": "suma",
          "\xA4": "valiuta"
        },
        lv: {
          "\u2206": "delta",
          "\u221E": "bezgaliba",
          "\u2665": "milestiba",
          "&": "un",
          "|": "vai",
          "<": "mazak neka",
          ">": "lielaks neka",
          "\u2211": "summa",
          "\xA4": "valuta"
        },
        my: {
          "\u2206": "kwahkhyaet",
          "\u221E": "asaonasme",
          "\u2665": "akhyait",
          "&": "nhin",
          "|": "tho",
          "<": "ngethaw",
          ">": "kyithaw",
          "\u2211": "paungld",
          "\xA4": "ngwekye"
        },
        mk: {},
        nl: {
          "\u2206": "delta",
          "\u221E": "oneindig",
          "\u2665": "liefde",
          "&": "en",
          "|": "of",
          "<": "kleiner dan",
          ">": "groter dan",
          "\u2211": "som",
          "\xA4": "valuta"
        },
        pl: {
          "\u2206": "delta",
          "\u221E": "nieskonczonosc",
          "\u2665": "milosc",
          "&": "i",
          "|": "lub",
          "<": "mniejsze niz",
          ">": "wieksze niz",
          "\u2211": "suma",
          "\xA4": "waluta"
        },
        pt: {
          "\u2206": "delta",
          "\u221E": "infinito",
          "\u2665": "amor",
          "&": "e",
          "|": "ou",
          "<": "menor que",
          ">": "maior que",
          "\u2211": "soma",
          "\xA4": "moeda"
        },
        ro: {
          "\u2206": "delta",
          "\u221E": "infinit",
          "\u2665": "dragoste",
          "&": "si",
          "|": "sau",
          "<": "mai mic ca",
          ">": "mai mare ca",
          "\u2211": "suma",
          "\xA4": "valuta"
        },
        ru: {
          "\u2206": "delta",
          "\u221E": "beskonechno",
          "\u2665": "lubov",
          "&": "i",
          "|": "ili",
          "<": "menshe",
          ">": "bolshe",
          "\u2211": "summa",
          "\xA4": "valjuta"
        },
        sk: {
          "\u2206": "delta",
          "\u221E": "nekonecno",
          "\u2665": "laska",
          "&": "a",
          "|": "alebo",
          "<": "menej ako",
          ">": "viac ako",
          "\u2211": "sucet",
          "\xA4": "mena"
        },
        sr: {},
        tr: {
          "\u2206": "delta",
          "\u221E": "sonsuzluk",
          "\u2665": "ask",
          "&": "ve",
          "|": "veya",
          "<": "kucuktur",
          ">": "buyuktur",
          "\u2211": "toplam",
          "\xA4": "para birimi"
        },
        uk: {
          "\u2206": "delta",
          "\u221E": "bezkinechnist",
          "\u2665": "lubov",
          "&": "i",
          "|": "abo",
          "<": "menshe",
          ">": "bilshe",
          "\u2211": "suma",
          "\xA4": "valjuta"
        },
        vn: {
          "\u2206": "delta",
          "\u221E": "vo cuc",
          "\u2665": "yeu",
          "&": "va",
          "|": "hoac",
          "<": "nho hon",
          ">": "lon hon",
          "\u2211": "tong",
          "\xA4": "tien te"
        }
      }, i = [";", "?", ":", "@", "&", "=", "+", "$", ",", "/"].join(""), c = [";", "?", ":", "@", "&", "=", "+", "$", ","].join(""), f = [".", "!", "~", "*", "'", "(", ")"].join(""), l = function(p, _) {
        var y = "-", T = "", S = "", A = !0, O = {}, F, L, M, R, q, be, ae, Te, On, te, C, Qe, X, Ae, re = "";
        if (typeof p != "string")
          return "";
        if (typeof _ == "string" && (y = _), ae = a.en, Te = s.en, typeof _ == "object") {
          F = _.maintainCase || !1, O = _.custom && typeof _.custom == "object" ? _.custom : O, M = +_.truncate > 1 && _.truncate || !1, R = _.uric || !1, q = _.uricNoSlash || !1, be = _.mark || !1, A = !(_.symbols === !1 || _.lang === !1), y = _.separator || y, R && (re += i), q && (re += c), be && (re += f), ae = _.lang && a[_.lang] && A ? a[_.lang] : A ? a.en : {}, Te = _.lang && s[_.lang] ? s[_.lang] : _.lang === !1 || _.lang === !0 ? {} : s.en, _.titleCase && typeof _.titleCase.length == "number" && Array.prototype.toString.call(_.titleCase) ? (_.titleCase.forEach(function(W) {
            O[W + ""] = W + "";
          }), L = !0) : L = !!_.titleCase, _.custom && typeof _.custom.length == "number" && Array.prototype.toString.call(_.custom) && _.custom.forEach(function(W) {
            O[W + ""] = W + "";
          }), Object.keys(O).forEach(function(W) {
            var xe;
            W.length > 1 ? xe = new RegExp("\\b" + d(W) + "\\b", "gi") : xe = new RegExp(d(W), "gi"), p = p.replace(xe, O[W]);
          });
          for (C in O)
            re += C;
        }
        for (re += y, re = d(re), p = p.replace(/(^\s+|\s+$)/g, ""), X = !1, Ae = !1, te = 0, Qe = p.length; te < Qe; te++)
          C = p[te], h(C, O) ? X = !1 : Te[C] ? (C = X && Te[C].match(/[A-Za-z0-9]/) ? " " + Te[C] : Te[C], X = !1) : C in r ? (te + 1 < Qe && o.indexOf(p[te + 1]) >= 0 ? (S += C, C = "") : Ae === !0 ? (C = u[S] + r[C], S = "") : C = X && r[C].match(/[A-Za-z0-9]/) ? " " + r[C] : r[C], X = !1, Ae = !1) : C in u ? (S += C, C = "", te === Qe - 1 && (C = u[S]), Ae = !0) : ae[C] && !(R && i.indexOf(C) !== -1) && !(q && c.indexOf(C) !== -1) ? (C = X || T.substr(-1).match(/[A-Za-z0-9]/) ? y + ae[C] : ae[C], C += p[te + 1] !== void 0 && p[te + 1].match(/[A-Za-z0-9]/) ? y : "", X = !0) : (Ae === !0 ? (C = u[S] + C, S = "", Ae = !1) : X && (/[A-Za-z0-9]/.test(C) || T.substr(-1).match(/A-Za-z0-9]/)) && (C = " " + C), X = !1), T += C.replace(new RegExp("[^\\w\\s" + re + "_-]", "g"), y);
        return L && (T = T.replace(/(\w)(\S*)/g, function(W, xe, wn) {
          var bt = xe.toUpperCase() + (wn !== null ? wn : "");
          return Object.keys(O).indexOf(bt.toLowerCase()) < 0 ? bt : bt.toLowerCase();
        })), T = T.replace(/\s+/g, y).replace(new RegExp("\\" + y + "+", "g"), y).replace(new RegExp("(^\\" + y + "+|\\" + y + "+$)", "g"), ""), M && T.length > M && (On = T.charAt(M) === y, T = T.slice(0, M), On || (T = T.slice(0, T.lastIndexOf(y)))), !F && !L && (T = T.toLowerCase()), T;
      }, g = function(p) {
        return function(y) {
          return l(y, p);
        };
      }, d = function(p) {
        return p.replace(/[-\\^$*+?.()|[\]{}\/]/g, "\\$&");
      }, h = function(v, p) {
        for (var _ in p)
          if (p[_] === v)
            return !0;
      };
      if (typeof t < "u" && t.exports)
        t.exports = l, t.exports.createSlug = g;
      else if (typeof define < "u" && define.amd)
        define([], function() {
          return l;
        });
      else
        try {
          if (n.getSlug || n.createSlug)
            throw "speakingurl: globals exists /(getSlug|createSlug)/";
          n.getSlug = l, n.createSlug = g;
        } catch {
        }
    }(e);
  }
}), td = Xo({
  "../../node_modules/.pnpm/speakingurl@14.0.1/node_modules/speakingurl/index.js"(e, t) {
    m(), t.exports = ed();
  }
});
m();
m();
m();
m();
m();
m();
m();
m();
function nd(e) {
  var t;
  const n = e.name || e._componentTag || e.__VUE_DEVTOOLS_COMPONENT_GUSSED_NAME__ || e.__name;
  return n === "index" && ((t = e.__file) == null ? void 0 : t.endsWith("index.vue")) ? "" : n;
}
function rd(e) {
  const t = e.__file;
  if (t)
    return $f(Bf(t, ".vue"));
}
function rr(e, t) {
  return e.type.__VUE_DEVTOOLS_COMPONENT_GUSSED_NAME__ = t, t;
}
function pn(e) {
  if (e.__VUE_DEVTOOLS_NEXT_APP_RECORD__)
    return e.__VUE_DEVTOOLS_NEXT_APP_RECORD__;
  if (e.root)
    return e.appContext.app.__VUE_DEVTOOLS_NEXT_APP_RECORD__;
}
function Zo(e) {
  var t, n;
  const r = (t = e.subTree) == null ? void 0 : t.type, o = pn(e);
  return o ? ((n = o == null ? void 0 : o.types) == null ? void 0 : n.Fragment) === r : !1;
}
function vt(e) {
  var t, n, r;
  const o = nd((e == null ? void 0 : e.type) || {});
  if (o)
    return o;
  if ((e == null ? void 0 : e.root) === e)
    return "Root";
  for (const s in (n = (t = e.parent) == null ? void 0 : t.type) == null ? void 0 : n.components)
    if (e.parent.type.components[s] === (e == null ? void 0 : e.type))
      return rr(e, s);
  for (const s in (r = e.appContext) == null ? void 0 : r.components)
    if (e.appContext.components[s] === (e == null ? void 0 : e.type))
      return rr(e, s);
  const u = rd((e == null ? void 0 : e.type) || {});
  return u || "Anonymous Component";
}
function od(e) {
  var t, n, r;
  const o = (r = (n = (t = e == null ? void 0 : e.appContext) == null ? void 0 : t.app) == null ? void 0 : n.__VUE_DEVTOOLS_NEXT_APP_RECORD_ID__) != null ? r : 0, u = e === (e == null ? void 0 : e.root) ? "root" : e.uid;
  return `${o}:${u}`;
}
function jt(e, t) {
  return t = t || `${e.id}:root`, e.instanceMap.get(t) || e.instanceMap.get(":root");
}
function ud() {
  const e = {
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    get width() {
      return e.right - e.left;
    },
    get height() {
      return e.bottom - e.top;
    }
  };
  return e;
}
var tt;
function sd(e) {
  return tt || (tt = document.createRange()), tt.selectNode(e), tt.getBoundingClientRect();
}
function ad(e) {
  const t = ud();
  if (!e.children)
    return t;
  for (let n = 0, r = e.children.length; n < r; n++) {
    const o = e.children[n];
    let u;
    if (o.component)
      u = pe(o.component);
    else if (o.el) {
      const s = o.el;
      s.nodeType === 1 || s.getBoundingClientRect ? u = s.getBoundingClientRect() : s.nodeType === 3 && s.data.trim() && (u = sd(s));
    }
    u && id(t, u);
  }
  return t;
}
function id(e, t) {
  return (!e.top || t.top < e.top) && (e.top = t.top), (!e.bottom || t.bottom > e.bottom) && (e.bottom = t.bottom), (!e.left || t.left < e.left) && (e.left = t.left), (!e.right || t.right > e.right) && (e.right = t.right), e;
}
var or = {
  top: 0,
  left: 0,
  right: 0,
  bottom: 0,
  width: 0,
  height: 0
};
function pe(e) {
  const t = e.subTree.el;
  return typeof window > "u" ? or : Zo(e) ? ad(e.subTree) : (t == null ? void 0 : t.nodeType) === 1 ? t == null ? void 0 : t.getBoundingClientRect() : e.subTree.component ? pe(e.subTree.component) : or;
}
m();
function _n(e) {
  return Zo(e) ? ld(e.subTree) : e.subTree ? [e.subTree.el] : [];
}
function ld(e) {
  if (!e.children)
    return [];
  const t = [];
  return e.children.forEach((n) => {
    n.component ? t.push(..._n(n.component)) : n != null && n.el && t.push(n.el);
  }), t;
}
var Jo = "__vue-devtools-component-inspector__", Qo = "__vue-devtools-component-inspector__card__", eu = "__vue-devtools-component-inspector__name__", tu = "__vue-devtools-component-inspector__indicator__", nu = {
  display: "block",
  zIndex: 2147483640,
  position: "fixed",
  backgroundColor: "#42b88325",
  border: "1px solid #42b88350",
  borderRadius: "5px",
  transition: "all 0.1s ease-in",
  pointerEvents: "none"
}, cd = {
  fontFamily: "Arial, Helvetica, sans-serif",
  padding: "5px 8px",
  borderRadius: "4px",
  textAlign: "left",
  position: "absolute",
  left: 0,
  color: "#e9e9e9",
  fontSize: "14px",
  fontWeight: 600,
  lineHeight: "24px",
  backgroundColor: "#42b883",
  boxShadow: "0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px -1px rgba(0, 0, 0, 0.1)"
}, fd = {
  display: "inline-block",
  fontWeight: 400,
  fontStyle: "normal",
  fontSize: "12px",
  opacity: 0.7
};
function Pe() {
  return document.getElementById(Jo);
}
function dd() {
  return document.getElementById(Qo);
}
function gd() {
  return document.getElementById(tu);
}
function pd() {
  return document.getElementById(eu);
}
function hn(e) {
  return {
    left: `${Math.round(e.left * 100) / 100}px`,
    top: `${Math.round(e.top * 100) / 100}px`,
    width: `${Math.round(e.width * 100) / 100}px`,
    height: `${Math.round(e.height * 100) / 100}px`
  };
}
function vn(e) {
  var t;
  const n = document.createElement("div");
  n.id = (t = e.elementId) != null ? t : Jo, Object.assign(n.style, {
    ...nu,
    ...hn(e.bounds),
    ...e.style
  });
  const r = document.createElement("span");
  r.id = Qo, Object.assign(r.style, {
    ...cd,
    top: e.bounds.top < 35 ? 0 : "-35px"
  });
  const o = document.createElement("span");
  o.id = eu, o.innerHTML = `&lt;${e.name}&gt;&nbsp;&nbsp;`;
  const u = document.createElement("i");
  return u.id = tu, u.innerHTML = `${Math.round(e.bounds.width * 100) / 100} x ${Math.round(e.bounds.height * 100) / 100}`, Object.assign(u.style, fd), r.appendChild(o), r.appendChild(u), n.appendChild(r), document.body.appendChild(n), n;
}
function mn(e) {
  const t = Pe(), n = dd(), r = pd(), o = gd();
  t && (Object.assign(t.style, {
    ...nu,
    ...hn(e.bounds)
  }), Object.assign(n.style, {
    top: e.bounds.top < 35 ? 0 : "-35px"
  }), r.innerHTML = `&lt;${e.name}&gt;&nbsp;&nbsp;`, o.innerHTML = `${Math.round(e.bounds.width * 100) / 100} x ${Math.round(e.bounds.height * 100) / 100}`);
}
function _d(e) {
  const t = pe(e);
  if (!t.width && !t.height)
    return;
  const n = vt(e);
  Pe() ? mn({ bounds: t, name: n }) : vn({ bounds: t, name: n });
}
function ru() {
  const e = Pe();
  e && (e.style.display = "none");
}
var Ht = null;
function zt(e) {
  const t = e.target;
  if (t) {
    const n = t.__vueParentComponent;
    if (n && (Ht = n, n.vnode.el)) {
      const o = pe(n), u = vt(n);
      Pe() ? mn({ bounds: o, name: u }) : vn({ bounds: o, name: u });
    }
  }
}
function hd(e, t) {
  if (e.preventDefault(), e.stopPropagation(), Ht) {
    const n = od(Ht);
    t(n);
  }
}
var ct = null;
function vd() {
  ru(), window.removeEventListener("mouseover", zt), window.removeEventListener("click", ct, !0), ct = null;
}
function md() {
  return window.addEventListener("mouseover", zt), new Promise((e) => {
    function t(n) {
      n.preventDefault(), n.stopPropagation(), hd(n, (r) => {
        window.removeEventListener("click", t, !0), ct = null, window.removeEventListener("mouseover", zt);
        const o = Pe();
        o && (o.style.display = "none"), e(JSON.stringify({ id: r }));
      });
    }
    ct = t, window.addEventListener("click", t, !0);
  });
}
function Ed(e) {
  const t = jt(j.value, e.id);
  if (t) {
    const [n] = _n(t);
    if (typeof n.scrollIntoView == "function")
      n.scrollIntoView({
        behavior: "smooth"
      });
    else {
      const r = pe(t), o = document.createElement("div"), u = {
        ...hn(r),
        position: "absolute"
      };
      Object.assign(o.style, u), document.body.appendChild(o), o.scrollIntoView({
        behavior: "smooth"
      }), setTimeout(() => {
        document.body.removeChild(o);
      }, 2e3);
    }
    setTimeout(() => {
      const r = pe(t);
      if (r.width || r.height) {
        const o = vt(t), u = Pe();
        u ? mn({ ...e, name: o, bounds: r }) : vn({ ...e, name: o, bounds: r }), setTimeout(() => {
          u && (u.style.display = "none");
        }, 1500);
      }
    }, 1200);
  }
}
m();
var ur, sr;
(sr = (ur = b).__VUE_DEVTOOLS_COMPONENT_INSPECTOR_ENABLED__) != null || (ur.__VUE_DEVTOOLS_COMPONENT_INSPECTOR_ENABLED__ = !0);
function yd(e) {
  let t = 0;
  const n = setInterval(() => {
    b.__VUE_INSPECTOR__ && (clearInterval(n), t += 30, e()), t >= 5e3 && clearInterval(n);
  }, 30);
}
function bd() {
  const e = b.__VUE_INSPECTOR__, t = e.openInEditor;
  e.openInEditor = async (...n) => {
    e.disable(), t(...n);
  };
}
function Td() {
  return new Promise((e) => {
    function t() {
      bd(), e(b.__VUE_INSPECTOR__);
    }
    b.__VUE_INSPECTOR__ ? t() : yd(() => {
      t();
    });
  });
}
m();
m();
function Ad(e) {
  return !!(e && e.__v_isReadonly);
}
function ou(e) {
  return Ad(e) ? ou(e.__v_raw) : !!(e && e.__v_isReactive);
}
function Dt(e) {
  return !!(e && e.__v_isRef === !0);
}
function ke(e) {
  const t = e && e.__v_raw;
  return t ? ke(t) : e;
}
var uu = class {
  constructor() {
    this.refEditor = new Od();
  }
  set(e, t, n, r) {
    const o = Array.isArray(t) ? t : t.split(".");
    for (; o.length > 1; ) {
      const a = o.shift();
      e instanceof Map && (e = e.get(a)), e instanceof Set ? e = Array.from(e.values())[a] : e = e[a], this.refEditor.isRef(e) && (e = this.refEditor.get(e));
    }
    const u = o[0], s = this.refEditor.get(e)[u];
    r ? r(e, u, n) : this.refEditor.isRef(s) ? this.refEditor.set(s, n) : e[u] = n;
  }
  get(e, t) {
    const n = Array.isArray(t) ? t : t.split(".");
    for (let r = 0; r < n.length; r++)
      if (e instanceof Map ? e = e.get(n[r]) : e = e[n[r]], this.refEditor.isRef(e) && (e = this.refEditor.get(e)), !e)
        return;
    return e;
  }
  has(e, t, n = !1) {
    if (typeof e > "u")
      return !1;
    const r = Array.isArray(t) ? t.slice() : t.split("."), o = n ? 2 : 1;
    for (; e && r.length > o; ) {
      const u = r.shift();
      e = e[u], this.refEditor.isRef(e) && (e = this.refEditor.get(e));
    }
    return e != null && Object.prototype.hasOwnProperty.call(e, r[0]);
  }
  createDefaultSetCallback(e) {
    return (t, n, r) => {
      if ((e.remove || e.newKey) && (Array.isArray(t) ? t.splice(n, 1) : ke(t) instanceof Map ? t.delete(n) : ke(t) instanceof Set ? t.delete(Array.from(t.values())[n]) : Reflect.deleteProperty(t, n)), !e.remove) {
        const o = t[e.newKey || n];
        this.refEditor.isRef(o) ? this.refEditor.set(o, r) : ke(t) instanceof Map ? t.set(e.newKey || n, r) : ke(t) instanceof Set ? t.add(r) : t[e.newKey || n] = r;
      }
    };
  }
}, Od = class {
  set(e, t) {
    if (Dt(e))
      e.value = t;
    else {
      if (e instanceof Set && Array.isArray(t)) {
        e.clear(), t.forEach((o) => e.add(o));
        return;
      }
      const n = Object.keys(t);
      if (e instanceof Map) {
        const o = new Set(e.keys());
        n.forEach((u) => {
          e.set(u, Reflect.get(t, u)), o.delete(u);
        }), o.forEach((u) => e.delete(u));
        return;
      }
      const r = new Set(Object.keys(e));
      n.forEach((o) => {
        Reflect.set(e, o, Reflect.get(t, o)), r.delete(o);
      }), r.forEach((o) => Reflect.deleteProperty(e, o));
    }
  }
  get(e) {
    return Dt(e) ? e.value : e;
  }
  isRef(e) {
    return Dt(e) || ou(e);
  }
};
new uu();
m();
m();
m();
var wd = "__VUE_DEVTOOLS_KIT_TIMELINE_LAYERS_STATE__";
function Sd() {
  if (!qo || typeof localStorage > "u" || localStorage === null)
    return {
      recordingState: !1,
      mouseEventEnabled: !1,
      keyboardEventEnabled: !1,
      componentEventEnabled: !1,
      performanceEventEnabled: !1,
      selected: ""
    };
  const e = localStorage.getItem(wd);
  return e ? JSON.parse(e) : {
    recordingState: !1,
    mouseEventEnabled: !1,
    keyboardEventEnabled: !1,
    componentEventEnabled: !1,
    performanceEventEnabled: !1,
    selected: ""
  };
}
m();
m();
m();
var ar, ir;
(ir = (ar = b).__VUE_DEVTOOLS_KIT_TIMELINE_LAYERS) != null || (ar.__VUE_DEVTOOLS_KIT_TIMELINE_LAYERS = []);
var Cd = new Proxy(b.__VUE_DEVTOOLS_KIT_TIMELINE_LAYERS, {
  get(e, t, n) {
    return Reflect.get(e, t, n);
  }
});
function Dd(e, t) {
  U.timelineLayersState[t.id] = !1, Cd.push({
    ...e,
    descriptorId: t.id,
    appRecord: pn(t.app)
  });
}
var lr, cr;
(cr = (lr = b).__VUE_DEVTOOLS_KIT_INSPECTOR__) != null || (lr.__VUE_DEVTOOLS_KIT_INSPECTOR__ = []);
var En = new Proxy(b.__VUE_DEVTOOLS_KIT_INSPECTOR__, {
  get(e, t, n) {
    return Reflect.get(e, t, n);
  }
}), su = Ce(() => {
  Re.hooks.callHook("sendInspectorToClient", au());
});
function Id(e, t) {
  var n, r;
  En.push({
    options: e,
    descriptor: t,
    treeFilterPlaceholder: (n = e.treeFilterPlaceholder) != null ? n : "Search tree...",
    stateFilterPlaceholder: (r = e.stateFilterPlaceholder) != null ? r : "Search state...",
    treeFilter: "",
    selectedNodeId: "",
    appRecord: pn(t.app)
  }), su();
}
function au() {
  return En.filter((e) => e.descriptor.app === j.value.app).filter((e) => e.descriptor.id !== "components").map((e) => {
    var t;
    const n = e.descriptor, r = e.options;
    return {
      id: r.id,
      label: r.label,
      logo: n.logo,
      icon: `custom-ic-baseline-${(t = r == null ? void 0 : r.icon) == null ? void 0 : t.replace(/_/g, "-")}`,
      packageName: n.packageName,
      homepage: n.homepage,
      pluginId: n.id
    };
  });
}
function ut(e, t) {
  return En.find((n) => n.options.id === e && (t ? n.descriptor.app === t : !0));
}
function Pd() {
  const e = Wo();
  e.hook("addInspector", ({ inspector: r, plugin: o }) => {
    Id(r, o.descriptor);
  });
  const t = Ce(async ({ inspectorId: r, plugin: o }) => {
    var u;
    if (!r || !((u = o == null ? void 0 : o.descriptor) != null && u.app) || U.highPerfModeEnabled)
      return;
    const s = ut(r, o.descriptor.app), a = {
      app: o.descriptor.app,
      inspectorId: r,
      filter: (s == null ? void 0 : s.treeFilter) || "",
      rootNodes: []
    };
    await new Promise((i) => {
      e.callHookWith(async (c) => {
        await Promise.all(c.map((f) => f(a))), i();
      }, "getInspectorTree");
    }), e.callHookWith(async (i) => {
      await Promise.all(i.map((c) => c({
        inspectorId: r,
        rootNodes: a.rootNodes
      })));
    }, "sendInspectorTreeToClient");
  }, 120);
  e.hook("sendInspectorTree", t);
  const n = Ce(async ({ inspectorId: r, plugin: o }) => {
    var u;
    if (!r || !((u = o == null ? void 0 : o.descriptor) != null && u.app) || U.highPerfModeEnabled)
      return;
    const s = ut(r, o.descriptor.app), a = {
      app: o.descriptor.app,
      inspectorId: r,
      nodeId: (s == null ? void 0 : s.selectedNodeId) || "",
      state: null
    }, i = {
      currentTab: `custom-inspector:${r}`
    };
    a.nodeId && await new Promise((c) => {
      e.callHookWith(async (f) => {
        await Promise.all(f.map((l) => l(a, i))), c();
      }, "getInspectorState");
    }), e.callHookWith(async (c) => {
      await Promise.all(c.map((f) => f({
        inspectorId: r,
        nodeId: a.nodeId,
        state: a.state
      })));
    }, "sendInspectorStateToClient");
  }, 120);
  return e.hook("sendInspectorState", n), e.hook("customInspectorSelectNode", ({ inspectorId: r, nodeId: o, plugin: u }) => {
    const s = ut(r, u.descriptor.app);
    !s || (s.selectedNodeId = o);
  }), e.hook("timelineLayerAdded", ({ options: r, plugin: o }) => {
    Dd(r, o.descriptor);
  }), e.hook("timelineEventAdded", ({ options: r, plugin: o }) => {
    var u;
    const s = ["performance", "component-event", "keyboard", "mouse"];
    U.highPerfModeEnabled || !((u = U.timelineLayersState) != null && u[o.descriptor.id]) && !s.includes(r.layerId) || e.callHookWith(async (a) => {
      await Promise.all(a.map((i) => i(r)));
    }, "sendTimelineEventToClient");
  }), e.hook("getComponentInstances", async ({ app: r }) => {
    const o = r.__VUE_DEVTOOLS_NEXT_APP_RECORD__;
    if (!o)
      return null;
    const u = o.id.toString();
    return [...o.instanceMap].filter(([a]) => a.split(":")[0] === u).map(([, a]) => a);
  }), e.hook("getComponentBounds", async ({ instance: r }) => pe(r)), e.hook("getComponentName", ({ instance: r }) => vt(r)), e.hook("componentHighlight", ({ uid: r }) => {
    const o = j.value.instanceMap.get(r);
    o && _d(o);
  }), e.hook("componentUnhighlight", () => {
    ru();
  }), e;
}
var fr, dr;
(dr = (fr = b).__VUE_DEVTOOLS_KIT_APP_RECORDS__) != null || (fr.__VUE_DEVTOOLS_KIT_APP_RECORDS__ = []);
var gr, pr;
(pr = (gr = b).__VUE_DEVTOOLS_KIT_ACTIVE_APP_RECORD__) != null || (gr.__VUE_DEVTOOLS_KIT_ACTIVE_APP_RECORD__ = {});
var _r, hr;
(hr = (_r = b).__VUE_DEVTOOLS_KIT_ACTIVE_APP_RECORD_ID__) != null || (_r.__VUE_DEVTOOLS_KIT_ACTIVE_APP_RECORD_ID__ = "");
var vr, mr;
(mr = (vr = b).__VUE_DEVTOOLS_KIT_CUSTOM_TABS__) != null || (vr.__VUE_DEVTOOLS_KIT_CUSTOM_TABS__ = []);
var Er, yr;
(yr = (Er = b).__VUE_DEVTOOLS_KIT_CUSTOM_COMMANDS__) != null || (Er.__VUE_DEVTOOLS_KIT_CUSTOM_COMMANDS__ = []);
var ce = "__VUE_DEVTOOLS_KIT_GLOBAL_STATE__";
function Rd() {
  return {
    connected: !1,
    clientConnected: !1,
    vitePluginDetected: !0,
    appRecords: [],
    activeAppRecordId: "",
    tabs: [],
    commands: [],
    highPerfModeEnabled: !0,
    devtoolsClientDetected: {},
    perfUniqueGroupId: 0,
    timelineLayersState: Sd()
  };
}
var br, Tr;
(Tr = (br = b)[ce]) != null || (br[ce] = Rd());
var xd = Ce((e) => {
  Re.hooks.callHook("devtoolsStateUpdated", { state: e });
});
Ce((e, t) => {
  Re.hooks.callHook("devtoolsConnectedUpdated", { state: e, oldState: t });
});
var mt = new Proxy(b.__VUE_DEVTOOLS_KIT_APP_RECORDS__, {
  get(e, t, n) {
    return t === "value" ? b.__VUE_DEVTOOLS_KIT_APP_RECORDS__ : b.__VUE_DEVTOOLS_KIT_APP_RECORDS__[t];
  }
}), j = new Proxy(b.__VUE_DEVTOOLS_KIT_ACTIVE_APP_RECORD__, {
  get(e, t, n) {
    return t === "value" ? b.__VUE_DEVTOOLS_KIT_ACTIVE_APP_RECORD__ : t === "id" ? b.__VUE_DEVTOOLS_KIT_ACTIVE_APP_RECORD_ID__ : b.__VUE_DEVTOOLS_KIT_ACTIVE_APP_RECORD__[t];
  }
});
function iu() {
  xd({
    ...b[ce],
    appRecords: mt.value,
    activeAppRecordId: j.id,
    tabs: b.__VUE_DEVTOOLS_KIT_CUSTOM_TABS__,
    commands: b.__VUE_DEVTOOLS_KIT_CUSTOM_COMMANDS__
  });
}
function Fd(e) {
  b.__VUE_DEVTOOLS_KIT_ACTIVE_APP_RECORD__ = e, iu();
}
function Ld(e) {
  b.__VUE_DEVTOOLS_KIT_ACTIVE_APP_RECORD_ID__ = e, iu();
}
var U = new Proxy(b[ce], {
  get(e, t) {
    return t === "appRecords" ? mt : t === "activeAppRecordId" ? j.id : t === "tabs" ? b.__VUE_DEVTOOLS_KIT_CUSTOM_TABS__ : t === "commands" ? b.__VUE_DEVTOOLS_KIT_CUSTOM_COMMANDS__ : b[ce][t];
  },
  deleteProperty(e, t) {
    return delete e[t], !0;
  },
  set(e, t, n) {
    return { ...b[ce] }, e[t] = n, b[ce][t] = n, !0;
  }
});
function kd(e = {}) {
  var t, n, r;
  const { file: o, host: u, baseUrl: s = window.location.origin, line: a = 0, column: i = 0 } = e;
  if (o) {
    if (u === "chrome-extension") {
      const c = o.replace(/\\/g, "\\\\"), f = (n = (t = window.VUE_DEVTOOLS_CONFIG) == null ? void 0 : t.openInEditorHost) != null ? n : "/";
      fetch(`${f}__open-in-editor?file=${encodeURI(o)}`).then((l) => {
        if (!l.ok) {
          const g = `Opening component ${c} failed`;
          console.log(`%c${g}`, "color:red");
        }
      });
    } else if (U.vitePluginDetected) {
      const c = (r = b.__VUE_DEVTOOLS_OPEN_IN_EDITOR_BASE_URL__) != null ? r : s;
      b.__VUE_INSPECTOR__.openInEditor(c, o, a, i);
    }
  }
}
m();
m();
m();
m();
m();
var Ar, Or;
(Or = (Ar = b).__VUE_DEVTOOLS_KIT_PLUGIN_BUFFER__) != null || (Ar.__VUE_DEVTOOLS_KIT_PLUGIN_BUFFER__ = []);
var yn = new Proxy(b.__VUE_DEVTOOLS_KIT_PLUGIN_BUFFER__, {
  get(e, t, n) {
    return Reflect.get(e, t, n);
  }
});
function Kt(e) {
  const t = {};
  return Object.keys(e).forEach((n) => {
    t[n] = e[n].defaultValue;
  }), t;
}
function bn(e) {
  return `__VUE_DEVTOOLS_NEXT_PLUGIN_SETTINGS__${e}__`;
}
function Nd(e) {
  var t, n, r;
  const o = (n = (t = yn.find((u) => {
    var s;
    return u[0].id === e && !!((s = u[0]) != null && s.settings);
  })) == null ? void 0 : t[0]) != null ? n : null;
  return (r = o == null ? void 0 : o.settings) != null ? r : null;
}
function lu(e, t) {
  var n, r, o;
  const u = bn(e);
  if (u) {
    const s = localStorage.getItem(u);
    if (s)
      return JSON.parse(s);
  }
  if (e) {
    const s = (r = (n = yn.find((a) => a[0].id === e)) == null ? void 0 : n[0]) != null ? r : null;
    return Kt((o = s == null ? void 0 : s.settings) != null ? o : {});
  }
  return Kt(t);
}
function Ud(e, t) {
  const n = bn(e);
  localStorage.getItem(n) || localStorage.setItem(n, JSON.stringify(Kt(t)));
}
function $d(e, t, n) {
  const r = bn(e), o = localStorage.getItem(r), u = JSON.parse(o || "{}"), s = {
    ...u,
    [t]: n
  };
  localStorage.setItem(r, JSON.stringify(s)), Re.hooks.callHookWith((a) => {
    a.forEach((i) => i({
      pluginId: e,
      key: t,
      oldValue: u[t],
      newValue: n,
      settings: s
    }));
  }, "setPluginSettings");
}
m();
m();
m();
m();
m();
m();
m();
m();
m();
m();
m();
var wr, Sr, K = (Sr = (wr = b).__VUE_DEVTOOLS_HOOK) != null ? Sr : wr.__VUE_DEVTOOLS_HOOK = Wo(), Bd = {
  vueAppInit(e) {
    K.hook("app:init", e);
  },
  vueAppUnmount(e) {
    K.hook("app:unmount", e);
  },
  vueAppConnected(e) {
    K.hook("app:connected", e);
  },
  componentAdded(e) {
    return K.hook("component:added", e);
  },
  componentEmit(e) {
    return K.hook("component:emit", e);
  },
  componentUpdated(e) {
    return K.hook("component:updated", e);
  },
  componentRemoved(e) {
    return K.hook("component:removed", e);
  },
  setupDevtoolsPlugin(e) {
    K.hook("devtools-plugin:setup", e);
  },
  perfStart(e) {
    return K.hook("perf:start", e);
  },
  perfEnd(e) {
    return K.hook("perf:end", e);
  }
}, cu = {
  on: Bd,
  setupDevToolsPlugin(e, t) {
    return K.callHook("devtools-plugin:setup", e, t);
  }
}, Vd = class {
  constructor({ plugin: e, ctx: t }) {
    this.hooks = t.hooks, this.plugin = e;
  }
  get on() {
    return {
      visitComponentTree: (e) => {
        this.hooks.hook("visitComponentTree", e);
      },
      inspectComponent: (e) => {
        this.hooks.hook("inspectComponent", e);
      },
      editComponentState: (e) => {
        this.hooks.hook("editComponentState", e);
      },
      getInspectorTree: (e) => {
        this.hooks.hook("getInspectorTree", e);
      },
      getInspectorState: (e) => {
        this.hooks.hook("getInspectorState", e);
      },
      editInspectorState: (e) => {
        this.hooks.hook("editInspectorState", e);
      },
      inspectTimelineEvent: (e) => {
        this.hooks.hook("inspectTimelineEvent", e);
      },
      timelineCleared: (e) => {
        this.hooks.hook("timelineCleared", e);
      },
      setPluginSettings: (e) => {
        this.hooks.hook("setPluginSettings", e);
      }
    };
  }
  notifyComponentUpdate(e) {
    var t;
    if (U.highPerfModeEnabled)
      return;
    const n = au().find((r) => r.packageName === this.plugin.descriptor.packageName);
    if (n != null && n.id) {
      if (e) {
        const r = [
          e.appContext.app,
          e.uid,
          (t = e.parent) == null ? void 0 : t.uid,
          e
        ];
        K.callHook("component:updated", ...r);
      } else
        K.callHook("component:updated");
      this.hooks.callHook("sendInspectorState", { inspectorId: n.id, plugin: this.plugin });
    }
  }
  addInspector(e) {
    this.hooks.callHook("addInspector", { inspector: e, plugin: this.plugin }), this.plugin.descriptor.settings && Ud(e.id, this.plugin.descriptor.settings);
  }
  sendInspectorTree(e) {
    U.highPerfModeEnabled || this.hooks.callHook("sendInspectorTree", { inspectorId: e, plugin: this.plugin });
  }
  sendInspectorState(e) {
    U.highPerfModeEnabled || this.hooks.callHook("sendInspectorState", { inspectorId: e, plugin: this.plugin });
  }
  selectInspectorNode(e, t) {
    this.hooks.callHook("customInspectorSelectNode", { inspectorId: e, nodeId: t, plugin: this.plugin });
  }
  visitComponentTree(e) {
    return this.hooks.callHook("visitComponentTree", e);
  }
  now() {
    return U.highPerfModeEnabled ? 0 : Date.now();
  }
  addTimelineLayer(e) {
    this.hooks.callHook("timelineLayerAdded", { options: e, plugin: this.plugin });
  }
  addTimelineEvent(e) {
    U.highPerfModeEnabled || this.hooks.callHook("timelineEventAdded", { options: e, plugin: this.plugin });
  }
  getSettings(e) {
    return lu(e != null ? e : this.plugin.descriptor.id, this.plugin.descriptor.settings);
  }
  getComponentInstances(e) {
    return this.hooks.callHook("getComponentInstances", { app: e });
  }
  getComponentBounds(e) {
    return this.hooks.callHook("getComponentBounds", { instance: e });
  }
  getComponentName(e) {
    return this.hooks.callHook("getComponentName", { instance: e });
  }
  highlightElement(e) {
    const t = e.__VUE_DEVTOOLS_NEXT_UID__;
    return this.hooks.callHook("componentHighlight", { uid: t });
  }
  unhighlightElement() {
    return this.hooks.callHook("componentUnhighlight");
  }
}, Md = Vd;
m();
m();
m();
m();
var jd = "__vue_devtool_undefined__", Hd = "__vue_devtool_infinity__", zd = "__vue_devtool_negative_infinity__", Kd = "__vue_devtool_nan__";
m();
m();
var qd = {
  [jd]: "undefined",
  [Kd]: "NaN",
  [Hd]: "Infinity",
  [zd]: "-Infinity"
};
Object.entries(qd).reduce((e, [t, n]) => (e[n] = t, e), {});
m();
m();
m();
m();
m();
var Cr, Dr;
(Dr = (Cr = b).__VUE_DEVTOOLS_KIT__REGISTERED_PLUGIN_APPS__) != null || (Cr.__VUE_DEVTOOLS_KIT__REGISTERED_PLUGIN_APPS__ = /* @__PURE__ */ new Set());
function Gd(e, t) {
  return cu.setupDevToolsPlugin(e, t);
}
function Wd(e, t) {
  const [n, r] = e;
  if (n.app !== t)
    return;
  const o = new Md({
    plugin: {
      setupFn: r,
      descriptor: n
    },
    ctx: Re
  });
  n.packageName === "vuex" && o.on.editInspectorState((u) => {
    o.sendInspectorState(u.inspectorId);
  }), r(o);
}
function fu(e, t) {
  b.__VUE_DEVTOOLS_KIT__REGISTERED_PLUGIN_APPS__.has(e) || U.highPerfModeEnabled && !(t != null && t.inspectingComponent) || (b.__VUE_DEVTOOLS_KIT__REGISTERED_PLUGIN_APPS__.add(e), yn.forEach((n) => {
    Wd(n, e);
  }));
}
m();
m();
var ze = "__VUE_DEVTOOLS_ROUTER__", De = "__VUE_DEVTOOLS_ROUTER_INFO__", Ir, Pr;
(Pr = (Ir = b)[De]) != null || (Ir[De] = {
  currentRoute: null,
  routes: []
});
var Rr, xr;
(xr = (Rr = b)[ze]) != null || (Rr[ze] = {});
new Proxy(b[De], {
  get(e, t) {
    return b[De][t];
  }
});
new Proxy(b[ze], {
  get(e, t) {
    if (t === "value")
      return b[ze];
  }
});
function Yd(e) {
  const t = /* @__PURE__ */ new Map();
  return ((e == null ? void 0 : e.getRoutes()) || []).filter((n) => !t.has(n.path) && t.set(n.path, 1));
}
function Tn(e) {
  return e.map((t) => {
    let { path: n, name: r, children: o, meta: u } = t;
    return o != null && o.length && (o = Tn(o)), {
      path: n,
      name: r,
      children: o,
      meta: u
    };
  });
}
function Xd(e) {
  if (e) {
    const { fullPath: t, hash: n, href: r, path: o, name: u, matched: s, params: a, query: i } = e;
    return {
      fullPath: t,
      hash: n,
      href: r,
      path: o,
      name: u,
      params: a,
      query: i,
      matched: Tn(s)
    };
  }
  return e;
}
function Zd(e, t) {
  function n() {
    var r;
    const o = (r = e.app) == null ? void 0 : r.config.globalProperties.$router, u = Xd(o == null ? void 0 : o.currentRoute.value), s = Tn(Yd(o)), a = console.warn;
    console.warn = () => {
    }, b[De] = {
      currentRoute: u ? nr(u) : {},
      routes: nr(s)
    }, b[ze] = o, console.warn = a;
  }
  n(), cu.on.componentUpdated(Ce(() => {
    var r;
    ((r = t.value) == null ? void 0 : r.app) === e.app && (n(), !U.highPerfModeEnabled && Re.hooks.callHook("routerInfoUpdated", { state: b[De] }));
  }, 200));
}
function Jd(e) {
  return {
    async getInspectorTree(t) {
      const n = {
        ...t,
        app: j.value.app,
        rootNodes: []
      };
      return await new Promise((r) => {
        e.callHookWith(async (o) => {
          await Promise.all(o.map((u) => u(n))), r();
        }, "getInspectorTree");
      }), n.rootNodes;
    },
    async getInspectorState(t) {
      const n = {
        ...t,
        app: j.value.app,
        state: null
      }, r = {
        currentTab: `custom-inspector:${t.inspectorId}`
      };
      return await new Promise((o) => {
        e.callHookWith(async (u) => {
          await Promise.all(u.map((s) => s(n, r))), o();
        }, "getInspectorState");
      }), n.state;
    },
    editInspectorState(t) {
      const n = new uu(), r = {
        ...t,
        app: j.value.app,
        set: (o, u = t.path, s = t.state.value, a) => {
          n.set(o, u, s, a || n.createDefaultSetCallback(t.state));
        }
      };
      e.callHookWith((o) => {
        o.forEach((u) => u(r));
      }, "editInspectorState");
    },
    sendInspectorState(t) {
      const n = ut(t);
      e.callHook("sendInspectorState", { inspectorId: t, plugin: {
        descriptor: n.descriptor,
        setupFn: () => ({})
      } });
    },
    inspectComponentInspector() {
      return md();
    },
    cancelInspectComponentInspector() {
      return vd();
    },
    getComponentRenderCode(t) {
      const n = jt(j.value, t);
      if (n)
        return (n == null ? void 0 : n.type) instanceof Function ? n.type.toString() : n.render.toString();
    },
    scrollToComponent(t) {
      return Ed({ id: t });
    },
    openInEditor: kd,
    getVueInspector: Td,
    toggleApp(t, n) {
      const r = mt.value.find((o) => o.id === t);
      r && (Ld(t), Fd(r), Zd(r, j), su(), fu(r.app, n));
    },
    inspectDOM(t) {
      const n = jt(j.value, t);
      if (n) {
        const [r] = _n(n);
        r && (b.__VUE_DEVTOOLS_INSPECT_DOM_TARGET__ = r);
      }
    },
    updatePluginSettings(t, n, r) {
      $d(t, n, r);
    },
    getPluginSettings(t) {
      return {
        options: Nd(t),
        values: lu(t)
      };
    }
  };
}
m();
var Fr, Lr;
(Lr = (Fr = b).__VUE_DEVTOOLS_ENV__) != null || (Fr.__VUE_DEVTOOLS_ENV__ = {
  vitePluginDetected: !1
});
var kr = Pd(), Nr, Ur;
(Ur = (Nr = b).__VUE_DEVTOOLS_KIT_CONTEXT__) != null || (Nr.__VUE_DEVTOOLS_KIT_CONTEXT__ = {
  hooks: kr,
  get state() {
    return {
      ...U,
      activeAppRecordId: j.id,
      activeAppRecord: j.value,
      appRecords: mt.value
    };
  },
  api: Jd(kr)
});
var Re = b.__VUE_DEVTOOLS_KIT_CONTEXT__;
m();
Qf(td(), 1);
var $r, Br;
(Br = ($r = b).__VUE_DEVTOOLS_NEXT_APP_RECORD_INFO__) != null || ($r.__VUE_DEVTOOLS_NEXT_APP_RECORD_INFO__ = {
  id: 0,
  appIds: /* @__PURE__ */ new Set()
});
m();
function Qd(e) {
  U.highPerfModeEnabled = e != null ? e : !U.highPerfModeEnabled, !e && j.value && fu(j.value.app);
}
m();
m();
m();
function e0(e) {
  U.devtoolsClientDetected = {
    ...U.devtoolsClientDetected,
    ...e
  };
  const t = Object.values(U.devtoolsClientDetected).some(Boolean);
  Qd(!t);
}
var Vr, Mr;
(Mr = (Vr = b).__VUE_DEVTOOLS_UPDATE_CLIENT_DETECTED__) != null || (Vr.__VUE_DEVTOOLS_UPDATE_CLIENT_DETECTED__ = e0);
m();
m();
m();
m();
m();
m();
m();
var t0 = class {
  constructor() {
    this.keyToValue = /* @__PURE__ */ new Map(), this.valueToKey = /* @__PURE__ */ new Map();
  }
  set(e, t) {
    this.keyToValue.set(e, t), this.valueToKey.set(t, e);
  }
  getByKey(e) {
    return this.keyToValue.get(e);
  }
  getByValue(e) {
    return this.valueToKey.get(e);
  }
  clear() {
    this.keyToValue.clear(), this.valueToKey.clear();
  }
}, du = class {
  constructor(e) {
    this.generateIdentifier = e, this.kv = new t0();
  }
  register(e, t) {
    this.kv.getByValue(e) || (t || (t = this.generateIdentifier(e)), this.kv.set(t, e));
  }
  clear() {
    this.kv.clear();
  }
  getIdentifier(e) {
    return this.kv.getByValue(e);
  }
  getValue(e) {
    return this.kv.getByKey(e);
  }
}, n0 = class extends du {
  constructor() {
    super((e) => e.name), this.classToAllowedProps = /* @__PURE__ */ new Map();
  }
  register(e, t) {
    typeof t == "object" ? (t.allowProps && this.classToAllowedProps.set(e, t.allowProps), super.register(e, t.identifier)) : super.register(e, t);
  }
  getAllowedProps(e) {
    return this.classToAllowedProps.get(e);
  }
};
m();
m();
function r0(e) {
  if ("values" in Object)
    return Object.values(e);
  const t = [];
  for (const n in e)
    e.hasOwnProperty(n) && t.push(e[n]);
  return t;
}
function o0(e, t) {
  const n = r0(e);
  if ("find" in n)
    return n.find(t);
  const r = n;
  for (let o = 0; o < r.length; o++) {
    const u = r[o];
    if (t(u))
      return u;
  }
}
function Ie(e, t) {
  Object.entries(e).forEach(([n, r]) => t(r, n));
}
function st(e, t) {
  return e.indexOf(t) !== -1;
}
function jr(e, t) {
  for (let n = 0; n < e.length; n++) {
    const r = e[n];
    if (t(r))
      return r;
  }
}
var u0 = class {
  constructor() {
    this.transfomers = {};
  }
  register(e) {
    this.transfomers[e.name] = e;
  }
  findApplicable(e) {
    return o0(this.transfomers, (t) => t.isApplicable(e));
  }
  findByName(e) {
    return this.transfomers[e];
  }
};
m();
m();
var s0 = (e) => Object.prototype.toString.call(e).slice(8, -1), gu = (e) => typeof e > "u", a0 = (e) => e === null, Ke = (e) => typeof e != "object" || e === null || e === Object.prototype ? !1 : Object.getPrototypeOf(e) === null ? !0 : Object.getPrototypeOf(e) === Object.prototype, qt = (e) => Ke(e) && Object.keys(e).length === 0, ue = (e) => Array.isArray(e), i0 = (e) => typeof e == "string", l0 = (e) => typeof e == "number" && !isNaN(e), c0 = (e) => typeof e == "boolean", f0 = (e) => e instanceof RegExp, qe = (e) => e instanceof Map, Ge = (e) => e instanceof Set, pu = (e) => s0(e) === "Symbol", d0 = (e) => e instanceof Date && !isNaN(e.valueOf()), g0 = (e) => e instanceof Error, Hr = (e) => typeof e == "number" && isNaN(e), p0 = (e) => c0(e) || a0(e) || gu(e) || l0(e) || i0(e) || pu(e), _0 = (e) => typeof e == "bigint", h0 = (e) => e === 1 / 0 || e === -1 / 0, v0 = (e) => ArrayBuffer.isView(e) && !(e instanceof DataView), m0 = (e) => e instanceof URL;
m();
var _u = (e) => e.replace(/\./g, "\\."), It = (e) => e.map(String).map(_u).join("."), Be = (e) => {
  const t = [];
  let n = "";
  for (let o = 0; o < e.length; o++) {
    let u = e.charAt(o);
    if (u === "\\" && e.charAt(o + 1) === ".") {
      n += ".", o++;
      continue;
    }
    if (u === ".") {
      t.push(n), n = "";
      continue;
    }
    n += u;
  }
  const r = n;
  return t.push(r), t;
};
m();
function Z(e, t, n, r) {
  return {
    isApplicable: e,
    annotation: t,
    transform: n,
    untransform: r
  };
}
var hu = [
  Z(gu, "undefined", () => null, () => {
  }),
  Z(_0, "bigint", (e) => e.toString(), (e) => typeof BigInt < "u" ? BigInt(e) : (console.error("Please add a BigInt polyfill."), e)),
  Z(d0, "Date", (e) => e.toISOString(), (e) => new Date(e)),
  Z(g0, "Error", (e, t) => {
    const n = {
      name: e.name,
      message: e.message
    };
    return t.allowedErrorProps.forEach((r) => {
      n[r] = e[r];
    }), n;
  }, (e, t) => {
    const n = new Error(e.message);
    return n.name = e.name, n.stack = e.stack, t.allowedErrorProps.forEach((r) => {
      n[r] = e[r];
    }), n;
  }),
  Z(f0, "regexp", (e) => "" + e, (e) => {
    const t = e.slice(1, e.lastIndexOf("/")), n = e.slice(e.lastIndexOf("/") + 1);
    return new RegExp(t, n);
  }),
  Z(
    Ge,
    "set",
    (e) => [...e.values()],
    (e) => new Set(e)
  ),
  Z(qe, "map", (e) => [...e.entries()], (e) => new Map(e)),
  Z((e) => Hr(e) || h0(e), "number", (e) => Hr(e) ? "NaN" : e > 0 ? "Infinity" : "-Infinity", Number),
  Z((e) => e === 0 && 1 / e === -1 / 0, "number", () => "-0", Number),
  Z(m0, "URL", (e) => e.toString(), (e) => new URL(e))
];
function Et(e, t, n, r) {
  return {
    isApplicable: e,
    annotation: t,
    transform: n,
    untransform: r
  };
}
var vu = Et((e, t) => pu(e) ? !!t.symbolRegistry.getIdentifier(e) : !1, (e, t) => {
  const n = t.symbolRegistry.getIdentifier(e);
  return ["symbol", n];
}, (e) => e.description, (e, t, n) => {
  const r = n.symbolRegistry.getValue(t[1]);
  if (!r)
    throw new Error("Trying to deserialize unknown symbol");
  return r;
}), E0 = [
  Int8Array,
  Uint8Array,
  Int16Array,
  Uint16Array,
  Int32Array,
  Uint32Array,
  Float32Array,
  Float64Array,
  Uint8ClampedArray
].reduce((e, t) => (e[t.name] = t, e), {}), mu = Et(v0, (e) => ["typed-array", e.constructor.name], (e) => [...e], (e, t) => {
  const n = E0[t[1]];
  if (!n)
    throw new Error("Trying to deserialize unknown typed array");
  return new n(e);
});
function Eu(e, t) {
  return e != null && e.constructor ? !!t.classRegistry.getIdentifier(e.constructor) : !1;
}
var yu = Et(Eu, (e, t) => {
  const n = t.classRegistry.getIdentifier(e.constructor);
  return ["class", n];
}, (e, t) => {
  const n = t.classRegistry.getAllowedProps(e.constructor);
  if (!n)
    return { ...e };
  const r = {};
  return n.forEach((o) => {
    r[o] = e[o];
  }), r;
}, (e, t, n) => {
  const r = n.classRegistry.getValue(t[1]);
  if (!r)
    throw new Error(`Trying to deserialize unknown class '${t[1]}' - check https://github.com/blitz-js/superjson/issues/116#issuecomment-773996564`);
  return Object.assign(Object.create(r.prototype), e);
}), bu = Et((e, t) => !!t.customTransformerRegistry.findApplicable(e), (e, t) => {
  const n = t.customTransformerRegistry.findApplicable(e);
  return ["custom", n.name];
}, (e, t) => t.customTransformerRegistry.findApplicable(e).serialize(e), (e, t, n) => {
  const r = n.customTransformerRegistry.findByName(t[1]);
  if (!r)
    throw new Error("Trying to deserialize unknown custom value");
  return r.deserialize(e);
}), y0 = [yu, vu, bu, mu], zr = (e, t) => {
  const n = jr(y0, (o) => o.isApplicable(e, t));
  if (n)
    return {
      value: n.transform(e, t),
      type: n.annotation(e, t)
    };
  const r = jr(hu, (o) => o.isApplicable(e, t));
  if (r)
    return {
      value: r.transform(e, t),
      type: r.annotation
    };
}, Tu = {};
hu.forEach((e) => {
  Tu[e.annotation] = e;
});
var b0 = (e, t, n) => {
  if (ue(t))
    switch (t[0]) {
      case "symbol":
        return vu.untransform(e, t, n);
      case "class":
        return yu.untransform(e, t, n);
      case "custom":
        return bu.untransform(e, t, n);
      case "typed-array":
        return mu.untransform(e, t, n);
      default:
        throw new Error("Unknown transformation: " + t);
    }
  else {
    const r = Tu[t];
    if (!r)
      throw new Error("Unknown transformation: " + t);
    return r.untransform(e, n);
  }
};
m();
var Oe = (e, t) => {
  if (t > e.size)
    throw new Error("index out of bounds");
  const n = e.keys();
  for (; t > 0; )
    n.next(), t--;
  return n.next().value;
};
function Au(e) {
  if (st(e, "__proto__"))
    throw new Error("__proto__ is not allowed as a property");
  if (st(e, "prototype"))
    throw new Error("prototype is not allowed as a property");
  if (st(e, "constructor"))
    throw new Error("constructor is not allowed as a property");
}
var T0 = (e, t) => {
  Au(t);
  for (let n = 0; n < t.length; n++) {
    const r = t[n];
    if (Ge(e))
      e = Oe(e, +r);
    else if (qe(e)) {
      const o = +r, u = +t[++n] == 0 ? "key" : "value", s = Oe(e, o);
      switch (u) {
        case "key":
          e = s;
          break;
        case "value":
          e = e.get(s);
          break;
      }
    } else
      e = e[r];
  }
  return e;
}, Gt = (e, t, n) => {
  if (Au(t), t.length === 0)
    return n(e);
  let r = e;
  for (let u = 0; u < t.length - 1; u++) {
    const s = t[u];
    if (ue(r)) {
      const a = +s;
      r = r[a];
    } else if (Ke(r))
      r = r[s];
    else if (Ge(r)) {
      const a = +s;
      r = Oe(r, a);
    } else if (qe(r)) {
      if (u === t.length - 2)
        break;
      const i = +s, c = +t[++u] == 0 ? "key" : "value", f = Oe(r, i);
      switch (c) {
        case "key":
          r = f;
          break;
        case "value":
          r = r.get(f);
          break;
      }
    }
  }
  const o = t[t.length - 1];
  if (ue(r) ? r[+o] = n(r[+o]) : Ke(r) && (r[o] = n(r[o])), Ge(r)) {
    const u = Oe(r, +o), s = n(u);
    u !== s && (r.delete(u), r.add(s));
  }
  if (qe(r)) {
    const u = +t[t.length - 2], s = Oe(r, u);
    switch (+o == 0 ? "key" : "value") {
      case "key": {
        const i = n(s);
        r.set(i, r.get(s)), i !== s && r.delete(s);
        break;
      }
      case "value": {
        r.set(s, n(r.get(s)));
        break;
      }
    }
  }
  return e;
};
function Wt(e, t, n = []) {
  if (!e)
    return;
  if (!ue(e)) {
    Ie(e, (u, s) => Wt(u, t, [...n, ...Be(s)]));
    return;
  }
  const [r, o] = e;
  o && Ie(o, (u, s) => {
    Wt(u, t, [...n, ...Be(s)]);
  }), t(r, n);
}
function A0(e, t, n) {
  return Wt(t, (r, o) => {
    e = Gt(e, o, (u) => b0(u, r, n));
  }), e;
}
function O0(e, t) {
  function n(r, o) {
    const u = T0(e, Be(o));
    r.map(Be).forEach((s) => {
      e = Gt(e, s, () => u);
    });
  }
  if (ue(t)) {
    const [r, o] = t;
    r.forEach((u) => {
      e = Gt(e, Be(u), () => e);
    }), o && Ie(o, n);
  } else
    Ie(t, n);
  return e;
}
var w0 = (e, t) => Ke(e) || ue(e) || qe(e) || Ge(e) || Eu(e, t);
function S0(e, t, n) {
  const r = n.get(e);
  r ? r.push(t) : n.set(e, [t]);
}
function C0(e, t) {
  const n = {};
  let r;
  return e.forEach((o) => {
    if (o.length <= 1)
      return;
    t || (o = o.map((a) => a.map(String)).sort((a, i) => a.length - i.length));
    const [u, ...s] = o;
    u.length === 0 ? r = s.map(It) : n[It(u)] = s.map(It);
  }), r ? qt(n) ? [r] : [r, n] : qt(n) ? void 0 : n;
}
var Ou = (e, t, n, r, o = [], u = [], s = /* @__PURE__ */ new Map()) => {
  var a;
  const i = p0(e);
  if (!i) {
    S0(e, o, t);
    const h = s.get(e);
    if (h)
      return r ? {
        transformedValue: null
      } : h;
  }
  if (!w0(e, n)) {
    const h = zr(e, n), v = h ? {
      transformedValue: h.value,
      annotations: [h.type]
    } : {
      transformedValue: e
    };
    return i || s.set(e, v), v;
  }
  if (st(u, e))
    return {
      transformedValue: null
    };
  const c = zr(e, n), f = (a = c == null ? void 0 : c.value) != null ? a : e, l = ue(f) ? [] : {}, g = {};
  Ie(f, (h, v) => {
    if (v === "__proto__" || v === "constructor" || v === "prototype")
      throw new Error(`Detected property ${v}. This is a prototype pollution risk, please remove it from your object.`);
    const p = Ou(h, t, n, r, [...o, v], [...u, e], s);
    l[v] = p.transformedValue, ue(p.annotations) ? g[v] = p.annotations : Ke(p.annotations) && Ie(p.annotations, (_, y) => {
      g[_u(v) + "." + y] = _;
    });
  });
  const d = qt(g) ? {
    transformedValue: l,
    annotations: c ? [c.type] : void 0
  } : {
    transformedValue: l,
    annotations: c ? [c.type, g] : g
  };
  return i || s.set(e, d), d;
};
m();
m();
function wu(e) {
  return Object.prototype.toString.call(e).slice(8, -1);
}
function Kr(e) {
  return wu(e) === "Array";
}
function D0(e) {
  if (wu(e) !== "Object")
    return !1;
  const t = Object.getPrototypeOf(e);
  return !!t && t.constructor === Object && t === Object.prototype;
}
function I0(e, t, n, r, o) {
  const u = {}.propertyIsEnumerable.call(r, t) ? "enumerable" : "nonenumerable";
  u === "enumerable" && (e[t] = n), o && u === "nonenumerable" && Object.defineProperty(e, t, {
    value: n,
    enumerable: !1,
    writable: !0,
    configurable: !0
  });
}
function Yt(e, t = {}) {
  if (Kr(e))
    return e.map((o) => Yt(o, t));
  if (!D0(e))
    return e;
  const n = Object.getOwnPropertyNames(e), r = Object.getOwnPropertySymbols(e);
  return [...n, ...r].reduce((o, u) => {
    if (Kr(t.props) && !t.props.includes(u))
      return o;
    const s = e[u], a = Yt(s, t);
    return I0(o, u, a, e, t.nonenumerable), o;
  }, {});
}
var I = class {
  constructor({ dedupe: e = !1 } = {}) {
    this.classRegistry = new n0(), this.symbolRegistry = new du((t) => {
      var n;
      return (n = t.description) != null ? n : "";
    }), this.customTransformerRegistry = new u0(), this.allowedErrorProps = [], this.dedupe = e;
  }
  serialize(e) {
    const t = /* @__PURE__ */ new Map(), n = Ou(e, t, this, this.dedupe), r = {
      json: n.transformedValue
    };
    n.annotations && (r.meta = {
      ...r.meta,
      values: n.annotations
    });
    const o = C0(t, this.dedupe);
    return o && (r.meta = {
      ...r.meta,
      referentialEqualities: o
    }), r;
  }
  deserialize(e) {
    const { json: t, meta: n } = e;
    let r = Yt(t);
    return n != null && n.values && (r = A0(r, n.values, this)), n != null && n.referentialEqualities && (r = O0(r, n.referentialEqualities)), r;
  }
  stringify(e) {
    return JSON.stringify(this.serialize(e));
  }
  parse(e) {
    return this.deserialize(JSON.parse(e));
  }
  registerClass(e, t) {
    this.classRegistry.register(e, t);
  }
  registerSymbol(e, t) {
    this.symbolRegistry.register(e, t);
  }
  registerCustom(e, t) {
    this.customTransformerRegistry.register({
      name: t,
      ...e
    });
  }
  allowErrorProps(...e) {
    this.allowedErrorProps.push(...e);
  }
};
I.defaultInstance = new I();
I.serialize = I.defaultInstance.serialize.bind(I.defaultInstance);
I.deserialize = I.defaultInstance.deserialize.bind(I.defaultInstance);
I.stringify = I.defaultInstance.stringify.bind(I.defaultInstance);
I.parse = I.defaultInstance.parse.bind(I.defaultInstance);
I.registerClass = I.defaultInstance.registerClass.bind(I.defaultInstance);
I.registerSymbol = I.defaultInstance.registerSymbol.bind(I.defaultInstance);
I.registerCustom = I.defaultInstance.registerCustom.bind(I.defaultInstance);
I.allowErrorProps = I.defaultInstance.allowErrorProps.bind(I.defaultInstance);
m();
m();
m();
m();
m();
m();
m();
m();
m();
m();
m();
m();
m();
m();
m();
m();
m();
m();
m();
m();
m();
m();
m();
var qr, Gr;
(Gr = (qr = b).__VUE_DEVTOOLS_KIT_MESSAGE_CHANNELS__) != null || (qr.__VUE_DEVTOOLS_KIT_MESSAGE_CHANNELS__ = []);
var Wr, Yr;
(Yr = (Wr = b).__VUE_DEVTOOLS_KIT_RPC_CLIENT__) != null || (Wr.__VUE_DEVTOOLS_KIT_RPC_CLIENT__ = null);
var Xr, Zr;
(Zr = (Xr = b).__VUE_DEVTOOLS_KIT_RPC_SERVER__) != null || (Xr.__VUE_DEVTOOLS_KIT_RPC_SERVER__ = null);
var Jr, Qr;
(Qr = (Jr = b).__VUE_DEVTOOLS_KIT_VITE_RPC_CLIENT__) != null || (Jr.__VUE_DEVTOOLS_KIT_VITE_RPC_CLIENT__ = null);
var eo, to;
(to = (eo = b).__VUE_DEVTOOLS_KIT_VITE_RPC_SERVER__) != null || (eo.__VUE_DEVTOOLS_KIT_VITE_RPC_SERVER__ = null);
var no, ro;
(ro = (no = b).__VUE_DEVTOOLS_KIT_BROADCAST_RPC_SERVER__) != null || (no.__VUE_DEVTOOLS_KIT_BROADCAST_RPC_SERVER__ = null);
m();
m();
m();
m();
m();
m();
m();
const J = "vue-hooks-plus", Pt = "Vue Hooks Plus \u{1F36D}", P0 = "https://raw.githubusercontent.com/InhiblabCore/vue-hooks-plus/c3b984112610ef3fb21140a0beb27b4a228fe0b3/packages/hooks/docs/public/logo.svg";
let oo;
const R0 = /* @__PURE__ */ new Map();
function x0(e) {
  Gd(
    {
      id: J,
      label: Pt,
      packageName: "vue-hooks-plus",
      homepage: "https://inhiblabcore.github.io/docs/hooks",
      logo: P0,
      app: e,
      settings: {
        baseSort: {
          type: "choice",
          component: "button-group",
          label: "Sort Cache Entries",
          options: [
            {
              label: "ASC",
              value: 1
            },
            {
              label: "DESC",
              value: -1
            }
          ],
          defaultValue: 1
        }
      }
    },
    (t) => {
      t.addTimelineLayer({
        id: J,
        label: Pt,
        color: 16767308
      }), t.addInspector({
        id: J,
        label: Pt,
        icon: "api",
        treeFilterPlaceholder: "Search  useRequest",
        actions: [
          {
            icon: "delete",
            tooltip: "Clear useRequest root ",
            action: () => {
              N.reset(oo), t.sendInspectorTree(J), t.sendInspectorState(J);
            }
          }
        ]
      }), N.subscribe((n) => {
        N.update(n.key, { time: n.time, type: n.type }), t.sendInspectorTree(J), t.sendInspectorState(J), t.addTimelineEvent({
          layerId: J,
          event: {
            title: n.type,
            subtitle: `data: ${JSON.stringify(n.data)}`,
            time: t.now(),
            data: {
              ...n
            }
          }
        });
      }), t.on.getInspectorTree((n) => {
        if (n.inspectorId === J) {
          R0.clear();
          const r = t.getSettings(), o = N.getAll();
          let u = [];
          r.baseSort === 1 ? u = Array.from(o.entries()).sort(
            (a, i) => {
              var c, f, l, g;
              return ((f = (c = i[1]) == null ? void 0 : c.time) != null ? f : 0) - ((g = (l = a[1]) == null ? void 0 : l.time) != null ? g : 0);
            }
          ) : u = Array.from(o.entries()).sort(
            (a, i) => {
              var c, f, l, g;
              return ((f = (c = a[1]) == null ? void 0 : c.time) != null ? f : 0) - ((g = (l = i[1]) == null ? void 0 : l.time) != null ? g : 0);
            }
          );
          const s = u.filter((a) => new RegExp(n.filter, "g").test(a[0])).map((a) => {
            var i, c, f;
            return {
              id: a[0],
              label: a[0],
              tags: (i = a[1]) != null && i.type ? [
                {
                  label: `${(c = a[1]) == null ? void 0 : c.type}`,
                  textColor: 16777215,
                  backgroundColor: Ts((f = a[1]) == null ? void 0 : f.type)
                }
              ] : []
            };
          });
          n.rootNodes = [
            {
              id: "vue-hooks-plus-useRequest",
              label: "useRequest",
              tags: [
                {
                  label: "Root",
                  textColor: 16777215,
                  backgroundColor: 4372611
                }
              ],
              children: s != null ? s : []
            }
          ];
        }
      }), t.on.getInspectorState((n) => {
        var o, u, s;
        oo = n.nodeId;
        let r = 0;
        if (n.inspectorId === J) {
          const a = N.getAll();
          if (n.nodeId) {
            const i = a.get(n.nodeId);
            if (!i)
              return;
            n.state = {
              Details: [
                {
                  key: "Key",
                  value: n.nodeId
                },
                {
                  key: "Request Name",
                  value: i.requestName
                }
              ],
              "Data Explorer": Object.keys(i.instance.state).map((c) => ({
                key: c,
                value: w(i.instance.state[c])
              })),
              Option: Object.keys(i.instance.options).map((c) => ({
                key: c,
                value: i.instance.options[c]
              })),
              ["Plugins \u{1F9E9}"]: (s = (u = (o = i.instance.pluginImpls) == null ? void 0 : o.map((c, f) => {
                var g, d, h, v, p, _;
                const l = (h = (d = (g = i == null ? void 0 : i.instance) == null ? void 0 : g.pluginImpls) == null ? void 0 : d[f]) == null ? void 0 : h.name;
                return l || f !== r && r++, {
                  key: l || `plugin ${r}`,
                  value: (_ = (p = (v = i == null ? void 0 : i.instance) == null ? void 0 : v.pluginImpls) == null ? void 0 : p[f]) != null ? _ : null
                };
              })) == null ? void 0 : u.filter((c) => Object.keys(c.value).length !== 0)) != null ? s : []
            };
          }
        }
      });
    }
  );
}
const vg = {
  install(e) {
    process.env.NODE_ENV === "development" && x0(e);
  }
};
function F0(e, t, n) {
  (n == null ? void 0 : n.immediate) && e(), k((o) => {
    if (w(t) === void 0 || typeof w(t) != "number" || w(t) < 0)
      return;
    const u = w(t), s = setTimeout(() => {
      e();
    }, u);
    o(() => {
      clearInterval(s);
    });
  });
}
function mg({ task: e, option: t }) {
  const { delay: n = 0, onError: r, onReady: o, onSuccess: u } = t != null ? t : {};
  if (!(e instanceof Array))
    throw new Error("task must be Array");
  const s = (c) => {
    r == null || r(c);
  }, a = (c) => (f) => {
    s(f), c == null || c({ error: f });
  }, i = () => {
    var c;
    (c = Array(...e.keys())) == null || c.reduce((f, l) => f.then((d) => (d != null && d.error || u == null || u(d), new Promise((h) => {
      var v;
      (v = e == null ? void 0 : e[l]) == null || v.call(e, h, a(h), l);
    }))), Promise.resolve());
  };
  F0(() => {
    o == null || o(), i();
  }, n);
}
function L0(e = !1, t) {
  const n = E(e), r = $(() => {
    const o = t === void 0 ? !e : t;
    return {
      toggle: () => {
        n.value = n.value === e ? o : e;
      },
      set: (c) => n.value = c,
      setLeft: () => n.value = e,
      setRight: () => n.value = o
    };
  });
  return [fo(n), { ...r.value }];
}
function An(e = !1) {
  const [t, { set: n, toggle: r }] = L0(e);
  return [t, {
    set: (u) => n(!!u),
    setTrue: () => n(!0),
    setFalse: () => n(!1),
    toggle: r
  }];
}
/*! js-cookie v3.0.5 | MIT */
function nt(e) {
  for (var t = 1; t < arguments.length; t++) {
    var n = arguments[t];
    for (var r in n)
      e[r] = n[r];
  }
  return e;
}
var k0 = {
  read: function(e) {
    return e[0] === '"' && (e = e.slice(1, -1)), e.replace(/(%[\dA-F]{2})+/gi, decodeURIComponent);
  },
  write: function(e) {
    return encodeURIComponent(e).replace(
      /%(2[346BF]|3[AC-F]|40|5[BDE]|60|7[BCD])/g,
      decodeURIComponent
    );
  }
};
function Xt(e, t) {
  function n(o, u, s) {
    if (!(typeof document > "u")) {
      s = nt({}, t, s), typeof s.expires == "number" && (s.expires = new Date(Date.now() + s.expires * 864e5)), s.expires && (s.expires = s.expires.toUTCString()), o = encodeURIComponent(o).replace(/%(2[346B]|5E|60|7C)/g, decodeURIComponent).replace(/[()]/g, escape);
      var a = "";
      for (var i in s)
        !s[i] || (a += "; " + i, s[i] !== !0 && (a += "=" + s[i].split(";")[0]));
      return document.cookie = o + "=" + e.write(u, o) + a;
    }
  }
  function r(o) {
    if (!(typeof document > "u" || arguments.length && !o)) {
      for (var u = document.cookie ? document.cookie.split("; ") : [], s = {}, a = 0; a < u.length; a++) {
        var i = u[a].split("="), c = i.slice(1).join("=");
        try {
          var f = decodeURIComponent(i[0]);
          if (s[f] = e.read(c, f), o === f)
            break;
        } catch {
        }
      }
      return o ? s[o] : s;
    }
  }
  return Object.create(
    {
      set: n,
      get: r,
      remove: function(o, u) {
        n(
          o,
          "",
          nt({}, u, {
            expires: -1
          })
        );
      },
      withAttributes: function(o) {
        return Xt(this.converter, nt({}, this.attributes, o));
      },
      withConverter: function(o) {
        return Xt(nt({}, this.converter, o), this.attributes);
      }
    },
    {
      attributes: { value: Object.freeze(t) },
      converter: { value: Object.freeze(e) }
    }
  );
}
var Rt = Xt(k0, { path: "/" });
function uo(e) {
  return typeof e == "function";
}
function Eg(e, t = {}) {
  const r = E((() => {
    const u = Rt.get(e);
    return typeof u == "string" ? u : uo(t.defaultValue) ? t.defaultValue() : t.defaultValue;
  })()), o = (u, s = {}) => {
    const { defaultValue: a, ...i } = { ...t, ...s }, c = () => {
      const f = uo(u) ? u(r.value) : u;
      return f === void 0 ? Rt.remove(e) : Rt.set(e, f, i), f;
    };
    r.value = c();
  };
  return [V(r), o];
}
const N0 = (e) => typeof e == "function", U0 = (e) => typeof e == "string", We = (e) => typeof e == "number";
function so(e, t = {}) {
  const { min: n, max: r } = t;
  let o = e;
  return We(r) && (o = Math.min(r, o)), We(n) && (o = Math.max(n, o)), o;
}
function yg(e = 0, t = {}) {
  const { min: n, max: r } = t, o = E(
    so(e, {
      min: n,
      max: r
    })
  ), u = (f) => {
    const l = We(f) ? f : f(o.value);
    return o.value = so(l, {
      max: r,
      min: n
    }), o.value;
  }, s = (f = 1) => {
    u((l) => l + f);
  }, a = (f = 1) => {
    u((l) => l - f);
  }, i = (f) => {
    u(f);
  }, c = () => {
    u(e);
  };
  return [
    V(o),
    {
      inc: s,
      dec: a,
      set: i,
      reset: c
    }
  ];
}
function bg(e, t, n) {
  var c;
  const r = $(() => e.value !== void 0), o = (c = e.value) != null ? c : t, u = E(o), s = E(r.value), a = $(
    () => r.value ? e.value : u.value
  );
  B(r, (f, l) => {
    f !== l && (console.warn(
      `WARN: Component changed from ${s.value ? "controlled" : "uncontrolled"} to ${f ? "controlled" : "uncontrolled"}`
    ), s.value = f);
  });
  function i(f, ...l) {
    if (typeof f == "function") {
      console.warn(
        "Function callbacks are not supported. See: https://github.com/adobe/react-spectrum/issues/2320"
      );
      const g = a.value, d = f(g);
      r.value || (u.value = d), Object.is(g, d) || n == null || n(d, ...l);
    } else {
      const g = !Object.is(a.value, f);
      r.value || (u.value = f), g && (n == null || n(f, ...l));
    }
  }
  return [a, i];
}
function $0(e, t) {
  const n = E(t || { wait: 1e3 }), r = E(), o = () => {
    const { wait: u = 1e3, ...s } = n.value;
    return Qt(e, u, s);
  };
  return r.value = o(), B(
    () => ({ ...n.value }),
    (u, s) => {
      var a;
      (u.wait !== (s == null ? void 0 : s.wait) || u.maxWait !== (s == null ? void 0 : s.maxWait)) && ((a = r.value) == null || a.cancel(), r.value = o());
    },
    { deep: !0 }
  ), _e(() => {
    var u;
    (u = r.value) == null || u.cancel();
  }), {
    run: (...u) => {
      var s;
      return (s = r.value) == null ? void 0 : s.call(r, ...u);
    },
    cancel: () => {
      var u;
      (u = r.value) == null || u.cancel();
    },
    flush: () => {
      var u;
      (u = r.value) == null || u.flush();
    },
    updateOptions: (u) => {
      n.value = u;
    }
  };
}
function Tg(e, t) {
  const n = E(e.value), { run: r } = $0(() => n.value = e.value, t);
  return B(e, () => r(), { deep: !0 }), n;
}
const yt = !!(typeof window < "u" && window.document && window.document.createElement);
function x(e, t) {
  var r, o;
  if (!yt)
    return;
  if (!e)
    return t;
  let n;
  return typeof e == "function" ? n = e() : fe(e) ? n = (o = (r = e.value) == null ? void 0 : r.$el) != null ? o : e.value : n = e, n;
}
function ao(e, t) {
  if (e === t)
    return !0;
  for (let n = 0; n < e.length; n++)
    if (!Object.is(e[n], t[n]))
      return !1;
  return !0;
}
const B0 = (e) => (n, r, o) => {
  const u = E(!1), s = E([]), a = E([]), i = E();
  e(() => {
    var l;
    const f = (Array.isArray(o) ? o : [o]).map((g) => x(g));
    if (!u.value) {
      u.value = !0, s.value = f, a.value = r, i.value = n();
      return;
    }
    (f.length !== s.value.length || !ao(f, s.value) || !ao(r, a.value)) && ((l = i.value) == null || l.call(i), s.value = f, a.value = r, i.value = n());
  }), _e(() => {
    var c;
    (c = i.value) == null || c.call(i), u.value = !1;
  });
}, ye = B0(k), Ag = (e, t, n = {}) => {
  const r = E(n);
  ye(
    () => {
      const o = x(t);
      if (!(o != null && o.addEventListener))
        return;
      const u = (a) => {
        var i, c, f;
        (c = (i = r.value).onDragStart) == null || c.call(i, a), (f = a.dataTransfer) == null || f.setData("custom", JSON.stringify(e));
      }, s = (a) => {
        var i, c;
        (c = (i = r.value).onDragEnd) == null || c.call(i, a);
      };
      return o.setAttribute(
        "draggable",
        `${(n == null ? void 0 : n.draggable) !== void 0 ? n == null ? void 0 : n.draggable : !0}`
      ), o.addEventListener("dragstart", u), o.addEventListener("dragend", s), () => {
        o.removeEventListener("dragstart", u), o.removeEventListener("dragend", s);
      };
    },
    [],
    t
  );
}, Og = (e, t = {}) => {
  const n = E(t), r = E();
  ye(
    () => {
      const o = x(e);
      if (!(o != null && o.addEventListener))
        return;
      const u = (l, g) => {
        const d = l.getData("text/uri-list"), h = l.getData("custom");
        if (h && n.value.onDom) {
          let v = h;
          try {
            v = JSON.parse(h);
          } catch {
            v = h;
          }
          n.value.onDom(v, g);
          return;
        }
        if (d && n.value.onUri) {
          n.value.onUri(d, g);
          return;
        }
        if (l.files && l.files.length && n.value.onFiles) {
          n.value.onFiles(Array.from(l.files), g);
          return;
        }
        l.items && l.items.length && n.value.onText && l.items[0].getAsString((v) => {
          n.value.onText(v, g);
        });
      }, s = (l) => {
        var g, d;
        l.preventDefault(), l.stopPropagation(), r.value = l.target, (d = (g = n.value).onDragEnter) == null || d.call(g, l);
      }, a = (l) => {
        var g, d;
        l.preventDefault(), (d = (g = n.value).onDragOver) == null || d.call(g, l);
      }, i = (l) => {
        var g, d;
        l.target === r.value && ((d = (g = n.value).onDragLeave) == null || d.call(g, l));
      }, c = (l) => {
        var g, d;
        l.preventDefault(), u(l.dataTransfer, l), (d = (g = n.value).onDrop) == null || d.call(g, l);
      }, f = (l) => {
        var g, d;
        u(l.clipboardData, l), (d = (g = n.value).onPaste) == null || d.call(g, l);
      };
      return o.addEventListener("dragenter", s), o.addEventListener("dragover", a), o.addEventListener("dragleave", i), o.addEventListener("drop", c), o.addEventListener("paste", f), () => {
        o.removeEventListener("dragenter", s), o.removeEventListener("dragover", a), o.removeEventListener("dragleave", i), o.removeEventListener("drop", c), o.removeEventListener("paste", f);
      };
    },
    [],
    e
  );
};
function io(e) {
  return typeof e == "function";
}
function Su(e) {
  function t(n, r) {
    let o;
    try {
      o = e();
    } catch (f) {
      console.error(f);
    }
    const u = (f) => r != null && r.serializer ? r == null ? void 0 : r.serializer(f) : JSON.stringify(f), s = (f) => r != null && r.deserializer ? r == null ? void 0 : r.deserializer(f) : JSON.parse(f);
    function a() {
      try {
        const f = o == null ? void 0 : o.getItem(w(n));
        if (f)
          return s(f);
      } catch (f) {
        console.error(f);
      }
      return io(r == null ? void 0 : r.defaultValue) ? r == null ? void 0 : r.defaultValue() : r == null ? void 0 : r.defaultValue;
    }
    const i = E(a());
    k(() => {
      n && (i.value = a());
    });
    const c = (f) => {
      if (typeof f > "u")
        i.value = void 0, o == null || o.removeItem(w(n));
      else if (io(f)) {
        const l = f(i.value);
        try {
          i.value = l, o == null || o.setItem(w(n), u(l));
        } catch (g) {
          console.error(g);
        }
      } else
        try {
          i.value = f, o == null || o.setItem(w(n), u(f));
        } catch (l) {
          console.error(l);
        }
    };
    return [V(i), c];
  }
  return t;
}
const V0 = Su(
  () => yt ? localStorage : void 0
);
function M0(e, t, n) {
  const r = e.map((a) => window.matchMedia(a)), o = () => {
    const a = r.findIndex((i) => i.matches);
    return typeof t[a] < "u" ? t[a] : n;
  }, u = E(o()), s = () => {
    u.value = o();
  };
  return k((a) => {
    r.forEach((i) => {
      i.addListener(s);
    }), a(() => {
      r.forEach(
        (i) => i.removeListener(s)
      );
    });
  }), u;
}
function wg() {
  const [e, t] = V0("dark-mode-enabled"), n = j0(), r = $(() => typeof e.value < "u" ? e.value : n.value);
  return k(() => {
    const o = "[vue-hooks-plus]-dark-mode", u = window.document.body;
    r.value ? u.classList.add(o) : u.classList.remove(o);
  }), [r, t];
}
function j0() {
  return M0(["(prefers-color-scheme: dark)"], [!0], !1);
}
const Le = {}, H0 = (e, t = {}) => {
  const n = document.querySelector(`script[src="${e}"]`);
  if (!n) {
    const r = document.createElement("script");
    return r.src = e, Object.keys(t).forEach((o) => {
      r[o] = t[o];
    }), r.setAttribute("data-status", "loading"), document.body.appendChild(r), {
      ref: r,
      status: "loading"
    };
  }
  return {
    ref: n,
    status: n.getAttribute("data-status") || "ready"
  };
}, z0 = (e, t = {}) => {
  const n = document.querySelector(`link[href="${e}"]`);
  if (!n) {
    const r = document.createElement("link");
    return r.rel = "stylesheet", r.href = e, Object.keys(t).forEach((u) => {
      r[u] = t[u];
    }), "hideFocus" in r && r.relList && (r.rel = "preload", r.as = "style"), r.setAttribute("data-status", "loading"), document.head.appendChild(r), {
      ref: r,
      status: "loading"
    };
  }
  return {
    ref: n,
    status: n.getAttribute("data-status") || "ready"
  };
};
function Sg(e, t) {
  const n = E(e ? "loading" : "unset"), r = E(), o = $(() => w(e));
  return k((u) => {
    var i, c, f, l;
    if (!o.value) {
      n.value = "unset";
      return;
    }
    const s = (c = (i = o.value) == null ? void 0 : i.replace(/[|#].*$/, "")) != null ? c : "";
    if ((t == null ? void 0 : t.type) === "css" || !(t != null && t.type) && /(^css!|\.css$)/.test(s)) {
      const g = z0((f = o.value) != null ? f : "", t == null ? void 0 : t.css);
      r.value = g.ref, n.value = g.status;
    } else if ((t == null ? void 0 : t.type) === "js" || !(t != null && t.type) && /(^js!|\.js$)/.test(s)) {
      const g = H0((l = o.value) != null ? l : "", t == null ? void 0 : t.js);
      r.value = g.ref, n.value = g.status;
    } else
      console.error(
        "Cannot infer the type of external resource, and please provide a type ('js' | 'css'). Refer to the https://ahooks.js.org/hooks/dom/use-external/#options"
      );
    if (!r.value)
      return;
    o.value && Le[o.value] === void 0 ? Le[o.value] = 1 : o.value && (Le[o.value] += 1);
    const a = (g) => {
      var h;
      const d = g.type === "load" ? "ready" : "error";
      (h = r.value) == null || h.setAttribute("data-status", d), n.value = d;
    };
    r.value.addEventListener("load", a), r.value.addEventListener("error", a), u(() => {
      var g, d, h;
      (g = r.value) == null || g.removeEventListener("load", a), (d = r.value) == null || d.removeEventListener("error", a), o.value && (Le[o.value] -= 1), o.value && Le[o.value] === 0 && ((h = r.value) == null || h.remove()), r.value = void 0;
    });
  }), V(n);
}
function se(e, t, n = {}) {
  const r = E(t);
  ye(
    () => {
      const o = x(n.target, window);
      if (!(o != null && o.addEventListener))
        return;
      const u = (s) => r.value(s);
      return o.addEventListener(e, u, {
        capture: n.capture,
        once: n.once,
        passive: n.passive
      }), () => {
        o.removeEventListener(e, u, {
          capture: n.capture
        });
      };
    },
    [e, n.capture, n.once, n.passive],
    n.target
  );
}
class Cu {
  constructor() {
    H(this, "subscriptions", /* @__PURE__ */ new Map());
    H(this, "emitEffectCache", /* @__PURE__ */ new Map());
    H(this, "useSubscription", (t, n) => {
      const r = E();
      k((o) => {
        var a, i;
        r.value = n;
        function u(c) {
          r.value && r.value(c);
        }
        const s = (i = (a = this.subscriptions) == null ? void 0 : a.get(t)) != null ? i : [];
        s.push(u), this.subscriptions.set(t, s), this.emitEffect(t), o(() => {
          this.subscriptions.delete(t);
        });
      });
    });
    H(this, "emit", (t, ...n) => {
      if (typeof t == "string" || typeof t == "number") {
        const r = this.subscriptions.get(t);
        r == null || r.forEach((o) => {
          o == null || o({
            params: Yn(n),
            event: t
          });
        }), this.emitEffectCache.set(t, {
          params: Yn(n),
          event: t
        });
      } else
        throw new TypeError("event must be string or number !");
    });
    H(this, "emitEffect", (t) => {
      const n = this.emitEffectCache.get(t), r = this.subscriptions.get(t);
      n && (r == null || r.forEach((o) => {
        o == null || o({
          ...n
        });
      }));
    });
    H(this, "removeListener", (t) => {
      this.subscriptions.delete(t);
    });
    H(this, "clear", () => {
      this.subscriptions.clear();
    });
    this.clear();
  }
}
const K0 = new Cu();
function Cg(e) {
  const t = E(), n = $(() => e != null ? e : { global: !1 });
  return t.value || (t.value = n.value.global ? t.value = K0 : t.value = new Cu()), k((r) => {
    r(() => {
      var o;
      return (o = t.value) == null ? void 0 : o.clear();
    });
  }), t.value;
}
const q0 = {
  SVG: "image/svg+xml",
  ICO: "image/x-icon",
  GIF: "image/gif",
  PNG: "image/png"
};
function Dg(e) {
  const t = $(() => w(e));
  k(() => {
    var u;
    if (!t.value)
      return;
    const n = (u = t.value) == null ? void 0 : u.split("."), r = n[n.length - 1].toLocaleUpperCase(), o = document.querySelector("link[rel*='icon']") || document.createElement("link");
    o.type = q0[r], o.href = t.value, o.rel = "shortcut icon", document.getElementsByTagName("head")[0].appendChild(o);
  });
}
function Ig(e, t) {
  const [n, { set: r }] = An(!1), { onFocus: o, onBlur: u, onChange: s } = t || {};
  return se(
    "focusin",
    (a) => {
      n.value || (o == null || o(a), s == null || s(!0), r(!0));
    },
    {
      target: e
    }
  ), se(
    "focusout",
    (a) => {
      var i, c;
      n.value && !((c = (i = a.currentTarget) == null ? void 0 : i.contains) != null && c.call(i, a.relatedTarget)) && (u == null || u(a), s == null || s(!1), r(!1));
    },
    {
      target: e
    }
  ), n;
}
function Pg(e, t) {
  return $(() => t(w(e)));
}
function G0(e) {
  ku(
    Nu({
      setup() {
        return e(), () => {
        };
      }
    })
  ).mount(document.createElement("div"));
}
const xt = "VUE_HOOKS_PLUS_USE_REQUEST_DEFAULT_KEY";
function W0(e) {
  return typeof e == "string" || typeof e == "number";
}
function Rg(e, t, n) {
  const r = E(n == null ? void 0 : n.fetchKey), o = E({}), u = E({}), s = (i) => {
    u.value = i;
  };
  return {
    fetchs: u,
    fetchRun: (...i) => {
      var l, g;
      const c = E(), f = (g = (l = r.value) == null ? void 0 : l.call(r, ...i)) != null ? g : xt;
      c.value = f, G0(() => {
        const { data: d, run: h, params: v, loading: p } = zo(e, {
          ...t,
          cacheKey: f,
          manual: !0
        });
        k(() => {
          o.value[f] = {
            key: f,
            data: d == null ? void 0 : d.value,
            params: v.value,
            loading: p.value
          }, s(o.value);
        }), h(...i), B([d, v, p, c], (_) => {
          const [
            y = void 0,
            T = void 0,
            S = !1,
            A = xt
          ] = _, O = W0(A) ? A : xt;
          o.value[O] = {
            key: O,
            data: y,
            params: T,
            loading: S
          }, s(o.value);
        });
      });
    }
  };
}
var Du = { exports: {} };
/*!
* screenfull
* v5.2.0 - 2021-11-03
* (c) Sindre Sorhus; MIT License
*/
(function(e) {
  (function() {
    var t = typeof window < "u" && typeof window.document < "u" ? window.document : {}, n = e.exports, r = function() {
      for (var s, a = [
        [
          "requestFullscreen",
          "exitFullscreen",
          "fullscreenElement",
          "fullscreenEnabled",
          "fullscreenchange",
          "fullscreenerror"
        ],
        [
          "webkitRequestFullscreen",
          "webkitExitFullscreen",
          "webkitFullscreenElement",
          "webkitFullscreenEnabled",
          "webkitfullscreenchange",
          "webkitfullscreenerror"
        ],
        [
          "webkitRequestFullScreen",
          "webkitCancelFullScreen",
          "webkitCurrentFullScreenElement",
          "webkitCancelFullScreen",
          "webkitfullscreenchange",
          "webkitfullscreenerror"
        ],
        [
          "mozRequestFullScreen",
          "mozCancelFullScreen",
          "mozFullScreenElement",
          "mozFullScreenEnabled",
          "mozfullscreenchange",
          "mozfullscreenerror"
        ],
        [
          "msRequestFullscreen",
          "msExitFullscreen",
          "msFullscreenElement",
          "msFullscreenEnabled",
          "MSFullscreenChange",
          "MSFullscreenError"
        ]
      ], i = 0, c = a.length, f = {}; i < c; i++)
        if (s = a[i], s && s[1] in t) {
          for (i = 0; i < s.length; i++)
            f[a[0][i]] = s[i];
          return f;
        }
      return !1;
    }(), o = {
      change: r.fullscreenchange,
      error: r.fullscreenerror
    }, u = {
      request: function(s, a) {
        return new Promise(function(i, c) {
          var f = function() {
            this.off("change", f), i();
          }.bind(this);
          this.on("change", f), s = s || t.documentElement;
          var l = s[r.requestFullscreen](a);
          l instanceof Promise && l.then(f).catch(c);
        }.bind(this));
      },
      exit: function() {
        return new Promise(function(s, a) {
          if (!this.isFullscreen) {
            s();
            return;
          }
          var i = function() {
            this.off("change", i), s();
          }.bind(this);
          this.on("change", i);
          var c = t[r.exitFullscreen]();
          c instanceof Promise && c.then(i).catch(a);
        }.bind(this));
      },
      toggle: function(s, a) {
        return this.isFullscreen ? this.exit() : this.request(s, a);
      },
      onchange: function(s) {
        this.on("change", s);
      },
      onerror: function(s) {
        this.on("error", s);
      },
      on: function(s, a) {
        var i = o[s];
        i && t.addEventListener(i, a, !1);
      },
      off: function(s, a) {
        var i = o[s];
        i && t.removeEventListener(i, a, !1);
      },
      raw: r
    };
    if (!r) {
      n ? e.exports = { isEnabled: !1 } : window.screenfull = { isEnabled: !1 };
      return;
    }
    Object.defineProperties(u, {
      isFullscreen: {
        get: function() {
          return Boolean(t[r.fullscreenElement]);
        }
      },
      element: {
        enumerable: !0,
        get: function() {
          return t[r.fullscreenElement];
        }
      },
      isEnabled: {
        enumerable: !0,
        get: function() {
          return Boolean(t[r.fullscreenEnabled]);
        }
      }
    }), n ? e.exports = u : window.screenfull = u;
  })();
})(Du);
const z = Du.exports, xg = (e, t) => {
  const { onExit: n, onEnter: r, defaultElement: o = document.documentElement } = t || {}, u = E(n), s = E(r), a = E(!1), i = (d) => {
    a.value = d;
  }, c = () => {
    var d, h, v;
    if (z.isEnabled) {
      const p = x(e, o);
      if (!z.element)
        (d = u.value) == null || d.call(u), i(!1), z.off("change", c);
      else {
        const _ = z.element === p;
        _ ? (h = s.value) == null || h.call(s) : (v = u.value) == null || v.call(u), i(_);
      }
    }
  }, f = () => {
    const d = x(e, o);
    if (!!d && z.isEnabled)
      try {
        z.request(d), z.on("change", c);
      } catch (h) {
        console.error(h);
      }
  }, l = () => {
    const d = x(e, o);
    z.isEnabled && z.element === d && z.exit();
  }, g = () => {
    a.value ? l() : f();
  };
  return _e(() => {
    z.isEnabled && z.off("change", c);
  }), [
    V(a),
    {
      enterFullscreen: f,
      exitFullscreen: l,
      toggleFullscreen: g,
      isEnabled: z.isEnabled
    }
  ];
};
function Fg(e, t) {
  const { onEnter: n, onLeave: r, onChange: o } = t || {}, [u, { setTrue: s, setFalse: a }] = An(!1);
  return se(
    "mouseenter",
    () => {
      n == null || n(), s(), o == null || o(!0);
    },
    {
      target: e
    }
  ), se(
    "mouseleave",
    () => {
      r == null || r(), a(), o == null || o(!1);
    },
    {
      target: e
    }
  ), u;
}
const Y0 = (e, t) => !Object.is(e, t);
function Lg(e, t = Y0) {
  const n = E(), r = E();
  return k(() => {
    t(r.value, e.value) && (n.value = r.value, r.value = e.value);
  }), V(n);
}
function kg(e, t, n) {
  const r = n == null ? void 0 : n.immediate, o = E(e), u = E(null), s = () => {
    if (fe(t)) {
      if (typeof t.value != "number" || t.value < 0)
        return;
    } else if (typeof t != "number" || t < 0)
      return;
    r && o.value();
    const c = w(t);
    u.value = setInterval(() => {
      o.value();
    }, c);
  }, a = () => {
    u.value && clearInterval(u.value);
  }, i = () => {
    a(), s();
  };
  return k((c) => {
    s(), c(a);
  }), {
    clear: a,
    restart: i
  };
}
const X0 = (e) => e === document || e === document.documentElement || e === document.body ? Math.max(
  window.pageYOffset,
  document.documentElement.scrollTop,
  document.body.scrollTop
) : e.scrollTop, Z0 = (e) => e.scrollHeight || Math.max(document.documentElement.scrollHeight, document.body.scrollHeight), J0 = (e) => e.clientHeight || Math.max(document.documentElement.clientHeight, document.body.clientHeight), Ng = (e, t = {}) => {
  const {
    target: n,
    isNoMore: r,
    threshold: o = 100,
    reloadDeps: u = [],
    manual: s,
    onBefore: a,
    onSuccess: i,
    onError: c,
    onFinally: f
  } = t, l = E(), [g, { set: d }] = An(), h = (R) => {
    l.value = R;
  }, v = $(() => r ? r(l.value) : !1), { loading: p, run: _, runAsync: y, cancel: T } = zo(
    async (R) => {
      const q = await e(R);
      return R ? l.value = {
        ...q,
        list: [...R.list, ...q.list]
      } : l.value = q, q;
    },
    {
      manual: s,
      onFinally: (R, q, be) => {
        d(!1), f == null || f(q, be);
      },
      onBefore: () => a == null ? void 0 : a(),
      onSuccess: (R) => {
        setTimeout(() => {
          L();
        }), i == null || i(R);
      },
      onError: (R) => c == null ? void 0 : c(R)
    }
  ), S = () => {
    v.value || (d(!0), _(l.value));
  }, A = () => {
    if (!v.value)
      return d(!0), y(l.value);
  }, O = () => _(), F = () => y(), L = () => {
    const R = x(n);
    if (!R)
      return;
    const q = X0(R), be = Z0(R), ae = J0(R);
    be - q <= ae + o && S();
  };
  se(
    "scroll",
    () => {
      p.value || g.value || L();
    },
    { target: n }
  ), B(u, () => {
    _();
  });
  const M = $(() => !g.value && p.value);
  return {
    data: fo(l),
    loading: V(M),
    loadingMore: g,
    noMore: v,
    loadMore: S,
    loadMoreAsync: A,
    reload: O,
    reloadAsync: F,
    mutate: h,
    scrollMethod: L,
    cancel: T
  };
};
function Ug(e, t) {
  const n = E(), r = E();
  return ye(
    () => {
      const o = x(e);
      if (!o)
        return;
      const u = new IntersectionObserver(
        (s) => {
          for (const a of s)
            n.value = a.isIntersecting, r.value = a.intersectionRatio;
        },
        {
          ...t,
          root: x(t == null ? void 0 : t.root)
        }
      );
      return u.observe(o), () => {
        u.disconnect();
      };
    },
    [],
    e
  ), [V(n), V(r)];
}
const Q0 = (e, t = []) => hf(e, t), Iu = (e, t, n) => {
  const r = E(), o = E(0);
  Q0(t, r.value) || (r.value = t, o.value += 1), ye(e, [o], n);
}, eg = {
  0: 48,
  1: 49,
  2: 50,
  3: 51,
  4: 52,
  5: 53,
  6: 54,
  7: 55,
  8: 56,
  9: 57,
  backspace: 8,
  tab: 9,
  enter: 13,
  shift: 16,
  ctrl: 17,
  alt: 18,
  pausebreak: 19,
  capslock: 20,
  esc: 27,
  space: 32,
  pageup: 33,
  pagedown: 34,
  end: 35,
  home: 36,
  leftarrow: 37,
  uparrow: 38,
  rightarrow: 39,
  downarrow: 40,
  insert: 45,
  delete: 46,
  a: 65,
  b: 66,
  c: 67,
  d: 68,
  e: 69,
  f: 70,
  g: 71,
  h: 72,
  i: 73,
  j: 74,
  k: 75,
  l: 76,
  m: 77,
  n: 78,
  o: 79,
  p: 80,
  q: 81,
  r: 82,
  s: 83,
  t: 84,
  u: 85,
  v: 86,
  w: 87,
  x: 88,
  y: 89,
  z: 90,
  leftwindowkey: 91,
  rightwindowkey: 92,
  selectkey: 93,
  numpad0: 96,
  numpad1: 97,
  numpad2: 98,
  numpad3: 99,
  numpad4: 100,
  numpad5: 101,
  numpad6: 102,
  numpad7: 103,
  numpad8: 104,
  numpad9: 105,
  multiply: 106,
  add: 107,
  subtract: 109,
  decimalpoint: 110,
  divide: 111,
  f1: 112,
  f2: 113,
  f3: 114,
  f4: 115,
  f5: 116,
  f6: 117,
  f7: 118,
  f8: 119,
  f9: 120,
  f10: 121,
  f11: 122,
  f12: 123,
  numlock: 144,
  scrolllock: 145,
  semicolon: 186,
  equalsign: 187,
  comma: 188,
  dash: 189,
  period: 190,
  forwardslash: 191,
  graveaccent: 192,
  openbracket: 219,
  backslash: 220,
  closebracket: 221,
  singlequote: 222
}, Zt = {
  ctrl: (e) => e.ctrlKey,
  shift: (e) => e.shiftKey,
  alt: (e) => e.altKey,
  meta: (e) => e.metaKey
};
function tg(e) {
  const t = Object.keys(Zt).reduce((n, r) => Zt[r](e) ? n + 1 : n, 0);
  return [16, 17, 18, 91, 92].includes(e.keyCode) ? t : t + 1;
}
function lo(e, t, n) {
  if (!e.key)
    return !1;
  if (We(t))
    return e.keyCode === t;
  const r = t.split(".");
  let o = 0;
  for (const u of r) {
    const s = Zt[u], a = eg[u.toLowerCase()];
    (s && s(e) || a && a === e.keyCode) && o++;
  }
  return n ? o === r.length && tg(e) === r.length : o === r.length;
}
function ng(e, t) {
  return N0(e) ? e : U0(e) || We(e) ? (n) => lo(n, e, t) : Array.isArray(e) ? (n) => e.some((r) => lo(n, r, t)) : e ? () => !0 : () => !1;
}
const rg = ["keydown"];
function $g(e, t, n) {
  const { events: r = rg, target: o, exactMatch: u = !1 } = n || {}, s = E(t), a = E(e);
  Iu(
    () => {
      var f;
      const i = x(o, window);
      if (!i)
        return;
      const c = (l) => {
        var d;
        if (ng(a.value, u)(l))
          return (d = s.value) == null ? void 0 : d.call(s, l);
      };
      for (const l of r)
        (f = i == null ? void 0 : i.addEventListener) == null || f.call(i, l, c);
      return () => {
        var l;
        for (const g of r)
          (l = i == null ? void 0 : i.removeEventListener) == null || l.call(
            i,
            g,
            c
          );
      };
    },
    [r],
    o
  );
}
function Bg(e) {
  let t = !1;
  return async (...n) => {
    if (!t) {
      t = !0;
      try {
        const r = await e(...n);
        return t = !1, r;
      } catch (r) {
        throw t = !1, r;
      }
    }
  };
}
const og = () => "PointerEvent" in window ? {
  mouseDown: "pointerdown",
  mouseUp: "pointerup",
  mouseMove: "pointermove"
} : "ontouchstart" in window || navigator.maxTouchPoints > 0 ? {
  mouseDown: "touchstart",
  mouseUp: "touchend",
  mouseMove: "touchmove"
} : {
  mouseDown: "mousedown",
  mouseUp: "mouseup",
  mouseMove: "mousemove"
}, Vg = (e, t = {}) => {
  var c, f;
  const o = E(!1), u = E(0);
  let s, a;
  const i = {
    capture: (c = t == null ? void 0 : t.modifiers) == null ? void 0 : c.capture,
    once: (f = t == null ? void 0 : t.modifiers) == null ? void 0 : f.once
  };
  return ye(
    () => {
      var _;
      const l = x(e);
      if (!l)
        return;
      const { mouseDown: g, mouseUp: d, mouseMove: h } = og();
      function v() {
        a && (clearTimeout(a), a = void 0), s && (clearInterval(s), s = void 0), u.value = 0, o.value = !1;
      }
      function p(y) {
        var T, S, A;
        ((T = t.modifiers) == null ? void 0 : T.self) && y.target !== l || (v(), (S = t.modifiers) != null && S.prevent && y.preventDefault(), (A = t.modifiers) != null && A.stop && y.stopPropagation(), a = setTimeout(() => {
          o.value = !0, s = setInterval(() => {
            u.value += t.minUpdateTime || 100;
          }, t.minUpdateTime || 100);
        }, t.delay || 500));
      }
      return l.addEventListener(g, p, i), l.addEventListener(d, v, i), ((_ = t.cancelOnMove) == null || _) && l.addEventListener(h, v, i), () => {
        var y;
        l.removeEventListener(g, p), l.removeEventListener(d, v), ((y = t.cancelOnMove) == null || y) && l.removeEventListener(h, v);
      };
    },
    [],
    e
  ), { isPressing: V(o), pressingTime: V(u) };
}, ug = {
  screenX: NaN,
  screenY: NaN,
  clientX: NaN,
  clientY: NaN,
  pageX: NaN,
  pageY: NaN,
  elementX: NaN,
  elementY: NaN,
  elementH: NaN,
  elementW: NaN,
  elementPosX: NaN,
  elementPosY: NaN
};
function Mg(e) {
  const t = E(ug);
  return se(
    "mousemove",
    (n) => {
      const { screenX: r, screenY: o, clientX: u, clientY: s, pageX: a, pageY: i } = n, c = {
        screenX: r,
        screenY: o,
        clientX: u,
        clientY: s,
        pageX: a,
        pageY: i,
        elementX: NaN,
        elementY: NaN,
        elementH: NaN,
        elementW: NaN,
        elementPosX: NaN,
        elementPosY: NaN
      }, f = x(e);
      if (f) {
        const { left: l, top: g, width: d, height: h } = f.getBoundingClientRect();
        c.elementPosX = l + window.pageXOffset, c.elementPosY = g + window.pageYOffset, c.elementX = a - c.elementPosX, c.elementY = i - c.elementPosY, c.elementW = d, c.elementH = h;
      }
      t.value = c;
    },
    {
      target: document
    }
  ), V(t);
}
function jg(e) {
  const t = () => e ? new Map(e) : /* @__PURE__ */ new Map(), n = E(t());
  return [n, go({
    set: (o, u) => {
      n.value.set(o, u);
    },
    get: (o) => n.value.get(o),
    remove: (o) => {
      n.value.delete(o);
    },
    has: (o) => n.value.has(o),
    clear: () => n.value.clear(),
    setAll: (o) => {
      n.value = new Map(o);
    },
    reset: () => n.value = t()
  })];
}
function Pu() {
  const e = navigator;
  return typeof e != "object" ? null : e.connection || e.mozConnection || e.webkitConnection;
}
function co() {
  const e = Pu();
  return e ? {
    rtt: e.rtt,
    type: e.type,
    saveData: e.saveData,
    downlink: e.downlink,
    downlinkMax: e.downlinkMax,
    effectiveType: e.effectiveType
  } : {};
}
function Hg() {
  const e = E({
    since: void 0,
    online: navigator == null ? void 0 : navigator.onLine,
    ...co()
  });
  return k((t) => {
    const n = () => {
      e.value = {
        ...e.value,
        online: !0,
        since: new Date()
      };
    }, r = () => {
      e.value = {
        ...e.value,
        online: !1,
        since: new Date()
      };
    }, o = () => {
      e.value = {
        ...e.value,
        ...co()
      };
    };
    window.addEventListener("online", n), window.addEventListener("offline", r);
    const u = Pu();
    u == null || u.addEventListener("change", o), t(() => {
      window.removeEventListener("online", n), window.removeEventListener("offline", r), u == null || u.removeEventListener("change", o);
    });
  }), V(e);
}
const zg = Su(
  () => yt ? sessionStorage : void 0
);
function Kg(e) {
  const t = () => e === void 0 ? /* @__PURE__ */ new Set() : new Set(e), n = E(t()), r = {
    add: (o) => {
      n.value.add(o);
    },
    remove: (o) => {
      n.value.delete(o);
    },
    has: (o) => n.value.has(o),
    clear: () => n.value.clear(),
    reset: () => {
      n.value = t();
    }
  };
  return [V(n), go(r)];
}
function qg(e) {
  const n = E((() => w(e))()), r = (o, u = !1) => {
    const s = w(o);
    u ? n.value = s : n.value = s ? mf(n.value, s) : n.value;
  };
  return [V(n), r];
}
function sg(e = () => {
}) {
  const t = () => {
    $u(() => {
      e();
    });
  };
  return gt(() => {
    window.addEventListener("resize", t, !1);
  }), Uu(() => {
    window.removeEventListener("resize", t);
  }), null;
}
function ag(e) {
  const t = dt({
    width: 0,
    height: 0
  }), n = () => {
    var o, u;
    const r = x(e);
    t.width = (o = r == null ? void 0 : r.clientWidth) != null ? o : 0, t.height = (u = r == null ? void 0 : r.clientHeight) != null ? u : 0;
  };
  return sg(n), gt(() => {
    setTimeout(() => {
      n();
    }, 120);
  }), { ...Jt(t) };
}
function Gg(e, t = () => !0) {
  const n = E(), r = E(t);
  return ye(
    () => {
      const o = x(e, document);
      if (!o)
        return;
      const u = () => {
        let s;
        o === document ? document.scrollingElement ? s = {
          left: document.scrollingElement.scrollLeft,
          top: document.scrollingElement.scrollTop
        } : s = {
          left: Math.max(
            window.pageYOffset,
            document.documentElement.scrollTop,
            document.body.scrollTop
          ),
          top: Math.max(
            window.pageXOffset,
            document.documentElement.scrollLeft,
            document.body.scrollLeft
          )
        } : s = {
          left: o.scrollLeft,
          top: o.scrollTop
        }, r.value(s) && (n.value = s);
      };
      return u(), o.addEventListener("scroll", u), () => {
        o.removeEventListener("scroll", u);
      };
    },
    [],
    e
  ), V(n);
}
function ig(e, t) {
  var u, s, a;
  let n;
  const r = E(), o = $(() => {
    var c;
    const i = {};
    return w(t == null ? void 0 : t.wait) !== void 0 && (i.wait = w(t == null ? void 0 : t.wait)), w(t == null ? void 0 : t.leading) !== void 0 && (i.leading = w(t == null ? void 0 : t.leading)), w(t == null ? void 0 : t.trailing) !== void 0 && (i.trailing = w(t == null ? void 0 : t.trailing)), {
      ...i,
      wait: (c = i == null ? void 0 : i.wait) != null ? c : 1e3
    };
  });
  return B(o, (i) => {
    const { wait: c = 1e3, ...f } = i;
    n && (n.cancel(), n.flush());
    const l = jo(
      (...g) => e([...g]),
      c,
      f
    );
    n = l, r.value = l;
  }, {
    immediate: !0,
    deep: !0
  }), _e(() => {
    var i;
    (i = r.value) == null || i.cancel();
  }), {
    run: (u = r.value) != null ? u : () => {
    },
    cancel: (s = r.value) == null ? void 0 : s.cancel,
    flush: (a = r.value) == null ? void 0 : a.flush
  };
}
function Wg(e, t) {
  const n = E();
  n.value = e.value;
  const { run: r } = ig(() => {
    n.value = e.value;
  }, t);
  return B(
    e,
    () => {
      r == null || r();
    },
    {
      immediate: !0
    }
  ), n;
}
const lg = {
  restoreOnUnmount: !1
};
function Yg(e, t = lg) {
  const n = E(yt ? document.title : "");
  fe(e) ? B(e, () => {
    document.title = e.value;
  }) : document.title = e, gt(() => {
    document.title = w(e);
  }), _e(() => {
    t.restoreOnUnmount && (document.title = w(n));
  });
}
const cg = (e, t) => e ? e.map((n, r) => e[r] !== (t == null ? void 0 : t[r]) ? r : -1).filter((n) => n >= 0) : t ? t.map((n, r) => r) : [], Xg = (e, t) => {
  B(t != null ? t : [], (n, r) => {
    const o = cg(r, n);
    e(o);
  });
};
function Zg() {
  const e = E({}), t = () => {
    e.value = Object.assign({}, { ...e.value });
  };
  return {
    update: V(e),
    setUpdate: t
  };
}
const Jg = (e, t) => {
  const n = E(), { wrapperTarget: r, itemHeight: o, overscan: u = 5 } = t, s = ag(n), a = E([]), i = E(!1), c = (p, _) => {
    if (typeof o == "number")
      return Math.ceil(p / o);
    let y = 0, T = 0;
    for (let S = _; S < e.value.length && (y += o(S, e.value[S]), T = S, !(y >= p)); S++)
      ;
    return T - _;
  }, f = (p) => {
    if (typeof o == "number")
      return Math.floor(p / o) + 1;
    let _ = 0, y = 0;
    for (let T = 0; T < e.value.length; T++)
      if (_ += o(T, e.value[T]), _ >= p) {
        y = T;
        break;
      }
    return y + 1;
  }, l = (p) => {
    var y, T;
    return typeof o == "number" ? p * o : (T = (y = e.value) == null ? void 0 : y.slice(0, p)) == null ? void 0 : T.reduce((S, A, O) => S + (o == null ? void 0 : o(O, e == null ? void 0 : e.value[p])), 0);
  }, g = $(() => typeof o == "number" ? e.value.length * o : e.value.reduce(
    (p, _, y) => p + (o == null ? void 0 : o(y, e == null ? void 0 : e.value[y])),
    0
  )), d = () => {
    const p = x(n), _ = x(r);
    if (p && _) {
      const { scrollTop: y, clientHeight: T } = p, S = f(y), A = c(T, S), O = Math.max(0, S - u), F = Math.min(e.value.length, S + A + u), L = l(O);
      _.style.height = g.value - L + "px", _.style.marginTop = L + "px", a.value = e.value.slice(O, F).map((M, R) => ({
        data: M,
        index: R + O
      }));
    }
  };
  B([s == null ? void 0 : s.width, s == null ? void 0 : s.height, e], () => {
    d();
  });
  const h = (p) => {
    const _ = x(n);
    _ && (i.value = !0, _.scrollTop = l(p), d());
  }, v = dt({
    ref: (p) => {
      n.value = p;
    },
    onScroll: (p) => {
      if (i.value) {
        i.value = !1;
        return;
      }
      p.preventDefault(), d();
    }
  });
  return [a, v, h];
};
function Qg(e, t) {
  const n = $(() => Object.keys(t).map((r) => r));
  B(
    Object.keys(t).map((r) => t[r]),
    (r, o) => {
      const u = {};
      n.value.forEach((s, a) => {
        o[a] !== r[a] && (u[s] = {
          from: o[a],
          to: r[a]
        });
      }), Object.keys(u).length && console.log("[why-did-you-update]", e, u);
    },
    {
      deep: !0
    }
  );
}
function ep(e, t = {}) {
  const {
    reconnectLimit: n = 3,
    reconnectInterval: r = 3 * 1e3,
    manual: o = E(!1),
    onOpen: u,
    onClose: s,
    onMessage: a,
    onError: i,
    protocols: c
  } = t, f = E(0), l = E(), g = E(), d = E(!1), h = E(), v = E(3), p = () => {
    var A;
    f.value < n && ((A = g.value) == null ? void 0 : A.readyState) !== 1 && (l.value && clearTimeout(l.value), l.value = setTimeout(() => {
      _(), f.value++;
    }, r));
  }, _ = () => {
    l.value && clearTimeout(l.value), g.value && g.value.close();
    const A = new WebSocket(w(e), c);
    v.value = 0, A.onerror = (O) => {
      d.value || (p(), i == null || i(O, A), v.value = A.readyState || 3);
    }, A.onopen = (O) => {
      d.value || (u == null || u(O, A), f.value = 0, v.value = A.readyState || 1);
    }, A.onmessage = (O) => {
      d.value || (a == null || a(O, A), h.value = O);
    }, A.onclose = (O) => {
      d.value || (p(), s == null || s(O, A), v.value = A.readyState || 3);
    }, g.value = A;
  }, y = (A) => {
    var O;
    if (v.value === 1)
      (O = g.value) == null || O.send(A);
    else
      throw new Error("WebSocket disconnected");
  }, T = () => {
    f.value = 0, _();
  }, S = () => {
    var A;
    l.value && clearTimeout(l.value), f.value = n, (A = g.value) == null || A.close();
  };
  return fe(e) && fe(o) ? B(
    [e, o],
    (A) => {
      const [O, F] = A;
      F || T();
    },
    {
      immediate: !0
    }
  ) : fe(o) ? B(
    o,
    (A) => {
      A || T();
    },
    {
      immediate: !0
    }
  ) : o || T(), _e(() => {
    d.value = !0, S();
  }), {
    latestMessage: h,
    sendMessage: y,
    connect: T,
    disconnect: S,
    readyState: v,
    webSocketIns: g.value
  };
}
function fg(e, t, n) {
  const { box: r = "content-box", ...o } = n != null ? n : {}, u = E(window && "ResizeObserver" in window);
  let s;
  const a = $(
    () => Array.isArray(e) ? e.map((l) => x(l)) : [x(e)]
  ), i = () => {
    s && (s.disconnect(), s = null);
  }, c = B(
    a,
    (l) => {
      i(), u.value && window && (s = new ResizeObserver(t), l.forEach((g) => {
        g && s.observe(g, {
          box: r,
          ...o
        });
      }));
    },
    {
      flush: "post",
      immediate: !0
    }
  ), f = () => {
    i(), c();
  };
  return Bu() && ft(f), {
    isSupported: u,
    stop: f
  };
}
function dg(e) {
  return ["width", "height", "top", "left", "bottom", "right"].includes(e);
}
function tp(e, t) {
  const { reset: n = !0, windowResize: r = !0, windowScroll: o = !0, immediate: u = !0 } = t != null ? t : {}, s = dt({
    width: 0,
    height: 0,
    top: 0,
    left: 0,
    bottom: 0,
    right: 0
  }), a = () => {
    const i = x(e);
    if (!i) {
      n && Object.keys(s).forEach((c) => {
        dg(c) && (s[c] = 0);
      });
      return;
    }
    if (i) {
      const { width: c, height: f, top: l, left: g, bottom: d, right: h } = i.getBoundingClientRect();
      s.width = c, s.height = f, s.top = l, s.left = g, s.bottom = d, s.right = h;
    }
  };
  return r && se("resize", a, {
    passive: !0
  }), o && se("scroll", a, {
    capture: !0,
    passive: !0
  }), fg(e, a), B(() => x(e), a), gt(() => {
    u && a();
  }), {
    ...Jt(s)
  };
}
const np = (e, t, n = {}) => {
  Iu(
    () => {
      const r = x(t);
      if (!r)
        return;
      const o = new MutationObserver(e);
      return o.observe(r, n), () => {
        o == null || o.disconnect();
      };
    },
    [n],
    t
  );
};
export {
  _g as clearUseRequestCache,
  mg as useAsyncOrder,
  An as useBoolean,
  bg as useControlledState,
  Eg as useCookieState,
  yg as useCounter,
  wg as useDarkMode,
  Tg as useDebounce,
  $0 as useDebounceFn,
  Ag as useDrag,
  Og as useDrop,
  tp as useElementBounding,
  Cg as useEventEmitter,
  se as useEventListener,
  Sg as useExternal,
  Dg as useFavicon,
  Rg as useFetchs,
  Ig as useFocusWithin,
  Pg as useFormatResult,
  xg as useFullscreen,
  Fg as useHover,
  Ug as useInViewport,
  Ng as useInfiniteScroll,
  kg as useInterval,
  $g as useKeyPress,
  V0 as useLocalStorageState,
  Bg as useLockFn,
  Vg as useLongPress,
  jg as useMap,
  M0 as useMedia,
  Mg as useMouse,
  np as useMutationObserver,
  Hg as useNetwork,
  Lg as usePrevious,
  zo as useRequest,
  vg as useRequestDevToolsPlugin,
  hg as useRequestProvider,
  fg as useResizeObserver,
  Gg as useScroll,
  zg as useSessionStorageState,
  Kg as useSet,
  qg as useSetState,
  ag as useSize,
  Wg as useThrottle,
  ig as useThrottleFn,
  F0 as useTimeout,
  Yg as useTitle,
  L0 as useToggle,
  Xg as useTrackedEffect,
  Zg as useUpdate,
  Jg as useVirtualList,
  ep as useWebSocket,
  Qg as useWhyDidYouUpdate,
  sg as useWinResize
};
