// vue-hooks-plus/lib/index.js
"use strict";
Object.defineProperties(exports, { __esModule: { value: true }, [Symbol.toStringTag]: { value: "Module" } });
const useRequest = require("./useRequest");
const useAsyncOrder = require("./useAsyncOrder");
const useBoolean = require("./useBoolean");
const useCookieState = require("./useCookieState");
const useCounter = require("./useCounter");
const useControlledState = require("./useControlledState");
const useDebounce = require("./useDebounce");
const useDebounceFn = require("./useDebounceFn");
const useDrag = require("./useDrag");
const useDrop = require("./useDrop");
const useDarkMode = require("./useDarkMode");
const useExternal = require("./useExternal");
const useEventListener = require("./useEventListener");
const useEventEmitter = require("./useEventEmitter");
const useFavicon = require("./useFavicon");
const useFocusWithin = require("./useFocusWithin");
const useFormatResult = require("./useFormatResult");
const useFetchs = require("./useFetchs");
const useFullscreen = require("./useFullscreen");
const useHover = require("./useHover");
const usePrevious = require("./usePrevious");
const useInterval = require("./useInterval");
const useInfiniteScroll = require("./useInfiniteScroll");
const useInViewport = require("./useInViewport");
const useKeyPress = require("./useKeyPress");
const useLocalStorageState = require("./useLocalStorageState");
const useLockFn = require("./useLockFn");
const useLongPress = require("./useLongPress");
const useMouse = require("./useMouse");
const useMap = require("./useMap");
const useMedia = require("./useMedia");
const useNetwork = require("./useNetwork");
const useSessionStorageState = require("./useSessionStorageState");
const useSet = require("./useSet");
const useSetState = require("./useSetState");
const useSize = require("./useSize");
const useScroll = require("./useScroll");
const useToggle = require("./useToggle");
const useThrottle = require("./useThrottle");
const useThrottleFn = require("./useThrottleFn");
const useTimeout = require("./useTimeout");
const useTitle = require("./useTitle");
const useTrackedEffect = require("./useTrackedEffect");
const useUpdate = require("./useUpdate");
const useVirtualList = require("./useVirtualList");
const useWhyDidYouUpdate = require("./useWhyDidYouUpdate");
const useWinResize = require("./useWinResize");
const useWebSocket = require("./useWebSocket");
const useElementBounding = require("./useElementBounding");
const useResizeObserver = require("./useResizeObserver");
const useMutationObserver = require("./useMutationObserver");
const _interopDefaultLegacy = (e) => e && typeof e === "object" && "default" in e ? e : { default: e };
const useRequest__default = /* @__PURE__ */ _interopDefaultLegacy(useRequest);
const useAsyncOrder__default = /* @__PURE__ */ _interopDefaultLegacy(useAsyncOrder);
const useBoolean__default = /* @__PURE__ */ _interopDefaultLegacy(useBoolean);
const useCookieState__default = /* @__PURE__ */ _interopDefaultLegacy(useCookieState);
const useCounter__default = /* @__PURE__ */ _interopDefaultLegacy(useCounter);
const useDebounce__default = /* @__PURE__ */ _interopDefaultLegacy(useDebounce);
const useDebounceFn__default = /* @__PURE__ */ _interopDefaultLegacy(useDebounceFn);
const useDrag__default = /* @__PURE__ */ _interopDefaultLegacy(useDrag);
const useDrop__default = /* @__PURE__ */ _interopDefaultLegacy(useDrop);
const useDarkMode__default = /* @__PURE__ */ _interopDefaultLegacy(useDarkMode);
const useExternal__default = /* @__PURE__ */ _interopDefaultLegacy(useExternal);
const useEventListener__default = /* @__PURE__ */ _interopDefaultLegacy(useEventListener);
const useEventEmitter__default = /* @__PURE__ */ _interopDefaultLegacy(useEventEmitter);
const useFavicon__default = /* @__PURE__ */ _interopDefaultLegacy(useFavicon);
const useFocusWithin__default = /* @__PURE__ */ _interopDefaultLegacy(useFocusWithin);
const useFormatResult__default = /* @__PURE__ */ _interopDefaultLegacy(useFormatResult);
const useFetchs__default = /* @__PURE__ */ _interopDefaultLegacy(useFetchs);
const useFullscreen__default = /* @__PURE__ */ _interopDefaultLegacy(useFullscreen);
const useHover__default = /* @__PURE__ */ _interopDefaultLegacy(useHover);
const usePrevious__default = /* @__PURE__ */ _interopDefaultLegacy(usePrevious);
const useInterval__default = /* @__PURE__ */ _interopDefaultLegacy(useInterval);
const useInfiniteScroll__default = /* @__PURE__ */ _interopDefaultLegacy(useInfiniteScroll);
const useInViewport__default = /* @__PURE__ */ _interopDefaultLegacy(useInViewport);
const useKeyPress__default = /* @__PURE__ */ _interopDefaultLegacy(useKeyPress);
const useLocalStorageState__default = /* @__PURE__ */ _interopDefaultLegacy(useLocalStorageState);
const useLockFn__default = /* @__PURE__ */ _interopDefaultLegacy(useLockFn);
const useLongPress__default = /* @__PURE__ */ _interopDefaultLegacy(useLongPress);
const useMouse__default = /* @__PURE__ */ _interopDefaultLegacy(useMouse);
const useMap__default = /* @__PURE__ */ _interopDefaultLegacy(useMap);
const useMedia__default = /* @__PURE__ */ _interopDefaultLegacy(useMedia);
const useNetwork__default = /* @__PURE__ */ _interopDefaultLegacy(useNetwork);
const useSessionStorageState__default = /* @__PURE__ */ _interopDefaultLegacy(useSessionStorageState);
const useSet__default = /* @__PURE__ */ _interopDefaultLegacy(useSet);
const useSetState__default = /* @__PURE__ */ _interopDefaultLegacy(useSetState);
const useSize__default = /* @__PURE__ */ _interopDefaultLegacy(useSize);
const useScroll__default = /* @__PURE__ */ _interopDefaultLegacy(useScroll);
const useToggle__default = /* @__PURE__ */ _interopDefaultLegacy(useToggle);
const useThrottle__default = /* @__PURE__ */ _interopDefaultLegacy(useThrottle);
const useThrottleFn__default = /* @__PURE__ */ _interopDefaultLegacy(useThrottleFn);
const useTimeout__default = /* @__PURE__ */ _interopDefaultLegacy(useTimeout);
const useTitle__default = /* @__PURE__ */ _interopDefaultLegacy(useTitle);
const useTrackedEffect__default = /* @__PURE__ */ _interopDefaultLegacy(useTrackedEffect);
const useUpdate__default = /* @__PURE__ */ _interopDefaultLegacy(useUpdate);
const useVirtualList__default = /* @__PURE__ */ _interopDefaultLegacy(useVirtualList);
const useWhyDidYouUpdate__default = /* @__PURE__ */ _interopDefaultLegacy(useWhyDidYouUpdate);
const useWinResize__default = /* @__PURE__ */ _interopDefaultLegacy(useWinResize);
const useWebSocket__default = /* @__PURE__ */ _interopDefaultLegacy(useWebSocket);
const useElementBounding__default = /* @__PURE__ */ _interopDefaultLegacy(useElementBounding);
const useResizeObserver__default = /* @__PURE__ */ _interopDefaultLegacy(useResizeObserver);
const useMutationObserver__default = /* @__PURE__ */ _interopDefaultLegacy(useMutationObserver);
Object.defineProperty(exports, "clearUseRequestCache", {
  enumerable: true,
  get: () => useRequest.clearUseRequestCache
});
Object.defineProperty(exports, "useRequest", {
  enumerable: true,
  get: () => useRequest__default.default
});
Object.defineProperty(exports, "useRequestDevToolsPlugin", {
  enumerable: true,
  get: () => useRequest.useRequestDevToolsPlugin
});
Object.defineProperty(exports, "useRequestProvider", {
  enumerable: true,
  get: () => useRequest.useRequestProvider
});
Object.defineProperty(exports, "useAsyncOrder", {
  enumerable: true,
  get: () => useAsyncOrder__default.default
});
Object.defineProperty(exports, "useBoolean", {
  enumerable: true,
  get: () => useBoolean__default.default
});
Object.defineProperty(exports, "useCookieState", {
  enumerable: true,
  get: () => useCookieState__default.default
});
Object.defineProperty(exports, "useCounter", {
  enumerable: true,
  get: () => useCounter__default.default
});
Object.defineProperty(exports, "useControlledState", {
  enumerable: true,
  get: () => useControlledState.useControlledState
});
Object.defineProperty(exports, "useDebounce", {
  enumerable: true,
  get: () => useDebounce__default.default
});
Object.defineProperty(exports, "useDebounceFn", {
  enumerable: true,
  get: () => useDebounceFn__default.default
});
Object.defineProperty(exports, "useDrag", {
  enumerable: true,
  get: () => useDrag__default.default
});
Object.defineProperty(exports, "useDrop", {
  enumerable: true,
  get: () => useDrop__default.default
});
Object.defineProperty(exports, "useDarkMode", {
  enumerable: true,
  get: () => useDarkMode__default.default
});
Object.defineProperty(exports, "useExternal", {
  enumerable: true,
  get: () => useExternal__default.default
});
Object.defineProperty(exports, "useEventListener", {
  enumerable: true,
  get: () => useEventListener__default.default
});
Object.defineProperty(exports, "useEventEmitter", {
  enumerable: true,
  get: () => useEventEmitter__default.default
});
Object.defineProperty(exports, "useFavicon", {
  enumerable: true,
  get: () => useFavicon__default.default
});
Object.defineProperty(exports, "useFocusWithin", {
  enumerable: true,
  get: () => useFocusWithin__default.default
});
Object.defineProperty(exports, "useFormatResult", {
  enumerable: true,
  get: () => useFormatResult__default.default
});
Object.defineProperty(exports, "useFetchs", {
  enumerable: true,
  get: () => useFetchs__default.default
});
Object.defineProperty(exports, "useFullscreen", {
  enumerable: true,
  get: () => useFullscreen__default.default
});
Object.defineProperty(exports, "useHover", {
  enumerable: true,
  get: () => useHover__default.default
});
Object.defineProperty(exports, "usePrevious", {
  enumerable: true,
  get: () => usePrevious__default.default
});
Object.defineProperty(exports, "useInterval", {
  enumerable: true,
  get: () => useInterval__default.default
});
Object.defineProperty(exports, "useInfiniteScroll", {
  enumerable: true,
  get: () => useInfiniteScroll__default.default
});
Object.defineProperty(exports, "useInViewport", {
  enumerable: true,
  get: () => useInViewport__default.default
});
Object.defineProperty(exports, "useKeyPress", {
  enumerable: true,
  get: () => useKeyPress__default.default
});
Object.defineProperty(exports, "useLocalStorageState", {
  enumerable: true,
  get: () => useLocalStorageState__default.default
});
Object.defineProperty(exports, "useLockFn", {
  enumerable: true,
  get: () => useLockFn__default.default
});
Object.defineProperty(exports, "useLongPress", {
  enumerable: true,
  get: () => useLongPress__default.default
});
Object.defineProperty(exports, "useMouse", {
  enumerable: true,
  get: () => useMouse__default.default
});
Object.defineProperty(exports, "useMap", {
  enumerable: true,
  get: () => useMap__default.default
});
Object.defineProperty(exports, "useMedia", {
  enumerable: true,
  get: () => useMedia__default.default
});
Object.defineProperty(exports, "useNetwork", {
  enumerable: true,
  get: () => useNetwork__default.default
});
Object.defineProperty(exports, "useSessionStorageState", {
  enumerable: true,
  get: () => useSessionStorageState__default.default
});
Object.defineProperty(exports, "useSet", {
  enumerable: true,
  get: () => useSet__default.default
});
Object.defineProperty(exports, "useSetState", {
  enumerable: true,
  get: () => useSetState__default.default
});
Object.defineProperty(exports, "useSize", {
  enumerable: true,
  get: () => useSize__default.default
});
Object.defineProperty(exports, "useScroll", {
  enumerable: true,
  get: () => useScroll__default.default
});
Object.defineProperty(exports, "useToggle", {
  enumerable: true,
  get: () => useToggle__default.default
});
Object.defineProperty(exports, "useThrottle", {
  enumerable: true,
  get: () => useThrottle__default.default
});
Object.defineProperty(exports, "useThrottleFn", {
  enumerable: true,
  get: () => useThrottleFn__default.default
});
Object.defineProperty(exports, "useTimeout", {
  enumerable: true,
  get: () => useTimeout__default.default
});
Object.defineProperty(exports, "useTitle", {
  enumerable: true,
  get: () => useTitle__default.default
});
Object.defineProperty(exports, "useTrackedEffect", {
  enumerable: true,
  get: () => useTrackedEffect__default.default
});
Object.defineProperty(exports, "useUpdate", {
  enumerable: true,
  get: () => useUpdate__default.default
});
Object.defineProperty(exports, "useVirtualList", {
  enumerable: true,
  get: () => useVirtualList__default.default
});
Object.defineProperty(exports, "useWhyDidYouUpdate", {
  enumerable: true,
  get: () => useWhyDidYouUpdate__default.default
});
Object.defineProperty(exports, "useWinResize", {
  enumerable: true,
  get: () => useWinResize__default.default
});
Object.defineProperty(exports, "useWebSocket", {
  enumerable: true,
  get: () => useWebSocket__default.default
});
Object.defineProperty(exports, "useElementBounding", {
  enumerable: true,
  get: () => useElementBounding__default.default
});
Object.defineProperty(exports, "useResizeObserver", {
  enumerable: true,
  get: () => useResizeObserver__default.default
});
Object.defineProperty(exports, "useMutationObserver", {
  enumerable: true,
  get: () => useMutationObserver__default.default
});
